---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r setup, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  # comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)
```

# gwasrapidd <img src="man/figures/logo.svg" align="right" height=140/>

[![](https://img.shields.io/badge/doi-10.1093/bioinformatics/btz605-blue.svg)](https://doi.org/10.1093/bioinformatics/btz605)
[![](https://img.shields.io/badge/Altmetric-38-yellow.svg)](https://www.altmetric.com/details/64505748)
[![R build status](https://github.com/ramiromagno/gwasrapidd/workflows/R-CMD-check/badge.svg)](https://github.com/ramiromagno/gwasrapidd/actions)
[![Codecov test coverage](https://codecov.io/gh/ramiromagno/gwasrapidd/branch/master/graph/badge.svg)](https://codecov.io/gh/ramiromagno/gwasrapidd?branch=master)
[![License: MIT](https://img.shields.io/badge/License-MIT-yellow.svg)](https://opensource.org/licenses/MIT)
[![lifecycle](https://img.shields.io/badge/lifecycle-maturing-blue.svg)](https://www.tidyverse.org/lifecycle/#maturing)
[![](https://img.shields.io/badge/devel%20version-0.99.11-blue.svg)](https://github.com/ramiromagno/gwasrapidd)

The goal of gwasrapidd is to provide programmatic access to the [NHGRI-EBI Catalog](https://www.ebi.ac.uk/gwas) of published genome-wide association studies.

Get started by reading the [documentation](https://rmagno.eu/gwasrapidd/articles/gwasrapidd.html).


## Installation

You can install the current version of gwasrapidd with:

``` r
# install.packages("remotes")
remotes::install_github("ramiromagno/gwasrapidd")
```

## Cheatsheet

<a href="https://github.com/rstudio/cheatsheets/blob/master/gwasrapidd.pdf"><img src="https://raw.githubusercontent.com/rstudio/cheatsheets/master/pngs/gwasrapidd.png" width="420" height="300"/></a>

## Example

Get studies related to triple-negative breast cancer:

```{r example1}
library(gwasrapidd)
studies <- get_studies(efo_trait = 'triple-negative breast cancer')
studies@studies[1:4]
```

Find associated variants with study `r studies@studies$study_id[1]`:

```{r example2}
variants <- get_variants(study_id = 'GCST002305')
variants@variants[c('variant_id', 'functional_class')]
```

## Citing this work

gwasrapidd was published in Bioinformatics in 2019: https://doi.org/10.1093/bioinformatics/btz605.

To generate a citation for this publication from within R:

```{r citation}
citation('gwasrapidd')
```

## Contributors

Please note that the gwasrapidd project is released with a [Contributor Code of Conduct](.github/CODE_OF_CONDUCT.md). By contributing to this project, you agree to abide by its terms.

## Similar projects

- Bioconductor R package *gwascat* by Vincent J Carey: [https://www.bioconductor.org/packages/release/bioc/html/gwascat.html](https://www.bioconductor.org/packages/release/bioc/html/gwascat.html)
- Web application *PhenoScanner V2* by Mihir A. Kamat, James R. Staley, and others: [http://www.phenoscanner.medschl.cam.ac.uk/](http://www.phenoscanner.medschl.cam.ac.uk/)
- Web application *GWEHS: Genome-Wide Effect sizes and Heritability Screener* by Eugenio LÃ³pez-Cortegano and Armando Caballero: [http://gwehs.uvigo.es/](http://gwehs.uvigo.es/)

## Acknowledgements

This work would have not been possible without the precious help from the [GWAS Catalog team](http://www.ensembl.info/2018/06/28/ten-years-of-the-gwas-catalog-past-present-and-future/), particularly [Daniel Suveges](https://www.ebi.ac.uk/about/people/daniel-suveges).

We also thank [Hadley](http://hadley.nz/)'s team for all the brilliant software that has influenced so profoundly the development of this package.
