<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>An S4 class to represent a set of GWAS Catalog associations — associations-class • gwasrapidd</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



  <link href="../extra.css" rel="stylesheet">
  

<meta property="og:title" content="An S4 class to represent a set of GWAS Catalog associations — associations-class" />
<meta property="og:description" content="The association object consists of six slots, each a table
(tibble), that combined form a relational database of a
subset of GWAS Catalog associations. Each association is an observation (row)
in the associations table --- main table. All tables have the column
association_id as primary key." />
<meta property="og:image" content="https://rmagno.eu/gwasrapidd/logo.svg" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gwasrapidd</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.8</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/gwasrapidd.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/bmi_variants.html">Variants associated with Body Mass Index (BMI)</a>
    </li>
    <li>
      <a href="../articles/faq.html">Frequently asked questions</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/ramiromagno/gwasrapidd/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>An S4 class to represent a set of GWAS Catalog associations</h1>
    <small class="dont-index">Source: <a href='https://github.com/ramiromagno/gwasrapidd/blob/master/R/class-associations.R'><code>R/class-associations.R</code></a></small>
    <div class="hidden name"><code>associations-class.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>The association object consists of six slots, each a table
(<code><a href='https://tibble.tidyverse.org/reference/tibble.html'>tibble</a></code>), that combined form a relational database of a
subset of GWAS Catalog associations. Each association is an observation (row)
in the <code>associations</code> table --- main table. All tables have the column
<code>association_id</code> as primary key.</p>
    </div>



    <h2 class="hasAnchor" id="slots"><a class="anchor" href="#slots"></a>Slots</h2>

    

<dl'>
<dt><code>associations</code></dt><dd><p>A <code><a href='https://tibble.tidyverse.org/reference/tibble.html'>tibble</a></code> listing associations.
  Columns:
<dl'>
<dt>association_id</dt><dd><p>GWAS Catalog association accession identifier, e.g.,
<code>"20250"</code>.</p></dd>
<dt>pvalue</dt><dd><p>Reported p-value for strongest variant risk or effect allele.</p></dd>
<dt>pvalue_description</dt><dd><p>Information describing context of p-value.</p></dd>
<dt>pvalue_mantissa</dt><dd><p>Mantissa of p-value.</p></dd>
<dt>pvalue_exponent</dt><dd><p>Exponent of p-value.</p></dd>
<dt>multiple_snp_haplotype</dt><dd><p>Whether the association is for a multi-SNP
haplotype.</p></dd>
<dt>snp_interaction</dt><dd><p>Whether the association is for a SNP-SNP interaction.</p></dd>
<dt>snp_type</dt><dd><p>Whether the SNP has previously been reported. Either
<code>'known'</code> or <code>'novel'</code>.</p></dd>
<dt>standard_error</dt><dd><p>Standard error of the effect size.</p></dd>
<dt>range</dt><dd><p>Reported 95% confidence interval associated with strongest SNP
risk allele, along with unit in the case of beta coefficients. If 95% CIs
have not been not reported, these are estimated using the standard error,
when available.</p></dd>
<dt>or_per_copy_number</dt><dd><p>Reported odds ratio (OR) associated
with strongest SNP risk allele. Note that all ORs included in the
Catalog are &gt;1.</p></dd>
<dt>beta_number</dt><dd><p>Beta coefficient associated with strongest SNP risk
allele.</p></dd>
<dt>beta_unit</dt><dd><p>Beta coefficient unit.</p></dd>
<dt>beta_direction</dt><dd><p>Beta coefficient direction, either <code>'decrease'</code> or
<code>'increase'</code>.</p></dd>
<dt>beta_description</dt><dd><p>Additional beta coefficient comment.</p></dd>
<dt>last_mapping_date</dt><dd><p>Last time this association was mapped to Ensembl.</p></dd>
<dt>last_update_date</dt><dd><p>Last time this association was updated.</p></dd></dl></p></dd>

<dt><code>loci</code></dt><dd><p>A <code><a href='https://tibble.tidyverse.org/reference/tibble.html'>tibble</a></code> listing loci. Columns:
<dl'>
<dt>association_id</dt><dd><p>GWAS Catalog association accession identifier, e.g.,
<code>"20250"</code>.</p></dd>
<dt>locus_id</dt><dd><p>A locus identifier referring to a single variant locus or to
a multi-loci entity such as a multi-SNP haplotype.</p></dd>
<dt>haplotype_snp_count</dt><dd><p>Number of variants per locus. Most loci are
single-SNP loci, i.e., there is a one to one relationship between a variant
and a <code>locus_id</code> (<code>haplotype_snp_count == NA</code>). There are however
cases of associations involving multiple loci at once, such as SNP-SNP
interactions and multi-SNP haplotypes. This is signalled in the columns:
<code>multiple_snp_haplotype</code> and <code>snp_interaction</code> with value
<code>TRUE</code>.</p></dd>
<dt>description</dt><dd><p>Description of the locus identifier, e.g.,
<code>'Single variant'</code>, <code>SNP x SNP interaction</code>, or <code>3-SNP
Haplotype</code>.</p></dd></dl></p></dd>

<dt><code>risk_alleles</code></dt><dd><p>A <code><a href='https://tibble.tidyverse.org/reference/tibble.html'>tibble</a></code> listing risk alleles.
  Columns:
<dl'>
<dt>association_id</dt><dd><p>GWAS Catalog association accession identifier, e.g.,
<code>"20250"</code>.</p></dd>
<dt>locus_id</dt><dd><p>A locus identifier referring to a single variant locus or to
a multi-loci entity such as a multi-SNP haplotype.</p></dd>
<dt>variant_id</dt><dd><p>Variant identifier, e.g., <code>'rs1333048'</code>.</p></dd>
<dt>risk_allele</dt><dd><p>Risk allele or effect allele.</p></dd>
<dt>risk_frequency</dt><dd><p>Reported risk/effect allele frequency associated with
strongest SNP in controls (if not available among all controls, among the
control group with the largest sample size). If the associated locus is a
haplotype the haplotype frequency will be extracted.</p></dd>
<dt>genome_wide</dt><dd><p>Whether this variant allele has been part of a genome-wide
study or not.</p></dd>
<dt>limited_list</dt><dd><p>Undocumented.</p></dd></dl></p></dd>

<dt><code>genes</code></dt><dd><p>A <code><a href='https://tibble.tidyverse.org/reference/tibble.html'>tibble</a></code> listing author reported genes.
  Columns:
<dl'>
<dt>association_id</dt><dd><p>GWAS Catalog association accession identifier, e.g.,
<code>"20250"</code>.</p></dd>
<dt>locus_id</dt><dd><p>A locus identifier referring to a single variant locus or to
a multi-loci entity such as a multi-SNP haplotype.</p></dd>
<dt>gene_name</dt><dd><p>Gene symbol according to
<a href='https://www.genenames.org/'>HUGO Gene Nomenclature (HGNC)</a>.</p></dd> 
</dl></p></dd>

<dt><code>ensembl_ids</code></dt><dd><p>A <code><a href='https://tibble.tidyverse.org/reference/tibble.html'>tibble</a></code> listing Ensembl gene
  identifiers. Columns:
<dl'>
<dt>association_id</dt><dd><p>GWAS Catalog association accession identifier, e.g.,
<code>"20250"</code>.</p></dd>
<dt>locus_id</dt><dd><p>A locus identifier referring to a single variant locus or to
a multi-loci entity such as a multi-SNP haplotype.</p></dd>
<dt>gene_name</dt><dd><p>Gene symbol according to
<a href='https://www.genenames.org/'>HUGO Gene Nomenclature (HGNC)</a>.</p></dd>
<dt>entrez_id</dt><dd><p>The Entrez identifier of a gene, see ref.
<a href='https://doi.org/10.1093%2Fnar%2Fgkq1237'>10.1093/nar/gkq1237</a> for
more information.</p></dd>
<dt>ensembl_id</dt><dd><p>The Ensembl identifier of an Ensembl gene, see Section
<a href='https://tinyurl.com/yxufd22b'>Gene
annotation in Ensembl</a> for more information.</p></dd> 
</dl></p></dd>

<dt><code>entrez_ids</code></dt><dd><p>A <code><a href='https://tibble.tidyverse.org/reference/tibble.html'>tibble</a></code> listing Entrez gene
  identifiers. Columns:
<dl'>
<dt>association_id</dt><dd><p>GWAS Catalog association accession identifier, e.g.,
<code>"20250"</code>.</p></dd>
<dt>locus_id</dt><dd><p>A locus identifier referring to a single variant locus or to
a multi-loci entity such as a multi-SNP haplotype.</p></dd>
<dt>gene_name</dt><dd><p>Gene symbol according to
<a href='https://www.genenames.org/'>HUGO Gene Nomenclature (HGNC)</a>.</p></dd>
<dt>entrez_id</dt><dd><p>The Entrez identifier of a gene, see ref.
<a href='https://doi.org/10.1093%2Fnar%2Fgkq1237'>10.1093/nar/gkq1237</a> for
more information.</p></dd></dl></p></dd>

</dl>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Ramiro Magno, Ana-Teresa Maia, Centre for Biomedical Research, University of Algarve.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


