<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en-US"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>An S4 class to represent a set of GWAS Catalog associations — associations-class • gwasrapidd</title><!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png"><link rel="icon" type="”image/svg+xml”" href="../favicon.svg"><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png"><link rel="icon" sizes="any" href="../favicon.ico"><link rel="manifest" href="../site.webmanifest"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/Nunito-0.4.10/font.css" rel="stylesheet"><link href="../deps/Inconsolata-0.4.10/font.css" rel="stylesheet"><link href="../deps/Source_Sans_Pro-0.4.10/font.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="An S4 class to represent a set of GWAS Catalog associations — associations-class"><meta name="description" content="The association object consists of six slots, each a table
(tibble), that combined form a relational database of a
subset of GWAS Catalog associations. Each association is an observation (row)
in the associations table — main table. All tables have the column
association_id as primary key."><meta property="og:description" content="The association object consists of six slots, each a table
(tibble), that combined form a relational database of a
subset of GWAS Catalog associations. Each association is an observation (row)
in the associations table — main table. All tables have the column
association_id as primary key."><meta property="og:image" content="https://rmagno.eu/gwasrapidd/logo.svg"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-none" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">gwasrapidd</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.18</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/gwasrapidd.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/bmi_variants.html">Variants associated with Body Mass Index (BMI)</a></li>
    <li><a class="dropdown-item" href="../articles/faq.html">Frequently asked questions</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/ramiromagno/gwasrapidd/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>An S4 class to represent a set of GWAS Catalog associations</h1>
      <small class="dont-index">Source: <a href="https://github.com/ramiromagno/gwasrapidd/blob/master/R/class-associations.R" class="external-link"><code>R/class-associations.R</code></a></small>
      <div class="d-none name"><code>associations-class.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>The association object consists of six slots, each a table
(<code><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></code>), that combined form a relational database of a
subset of GWAS Catalog associations. Each association is an observation (row)
in the <code>associations</code> table — main table. All tables have the column
<code>association_id</code> as primary key.</p>
    </div>


    <div class="section level2">
    <h2 id="slots">Slots<a class="anchor" aria-label="anchor" href="#slots"></a></h2>


<dl><dt><code>associations</code></dt>
<dd><p>A <code><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></code> listing associations.
  Columns:</p><dl><dt>association_id</dt>
<dd><p>GWAS Catalog association accession identifier, e.g.,
<code>"20250"</code>.</p></dd>

<dt>pvalue</dt>
<dd><p>Reported p-value for strongest variant risk or effect allele.</p></dd>

<dt>pvalue_description</dt>
<dd><p>Information describing context of p-value.</p></dd>

<dt>pvalue_mantissa</dt>
<dd><p>Mantissa of p-value.</p></dd>

<dt>pvalue_exponent</dt>
<dd><p>Exponent of p-value.</p></dd>

<dt>multiple_snp_haplotype</dt>
<dd><p>Whether the association is for a multi-SNP
haplotype.</p></dd>

<dt>snp_interaction</dt>
<dd><p>Whether the association is for a SNP-SNP interaction.</p></dd>

<dt>snp_type</dt>
<dd><p>Whether the SNP has previously been reported. Either
<code>'known'</code> or <code>'novel'</code>.</p></dd>

<dt>risk_frequency</dt>
<dd><p>Reported risk/effect allele frequency associated with
strongest SNP in controls.</p></dd>

<dt>standard_error</dt>
<dd><p>Standard error of the effect size.</p></dd>

<dt>range</dt>
<dd><p>Reported 95% confidence interval associated with strongest SNP
risk allele, along with unit in the case of beta coefficients. If 95% CIs
have not been not reported, these are estimated using the standard error,
when available.</p></dd>

<dt>or_per_copy_number</dt>
<dd><p>Reported odds ratio (OR) associated
with strongest SNP risk allele. Note that all ORs included in the
Catalog are &gt;1.</p></dd>

<dt>beta_number</dt>
<dd><p>Beta coefficient associated with strongest SNP risk
allele.</p></dd>

<dt>beta_unit</dt>
<dd><p>Beta coefficient unit.</p></dd>

<dt>beta_direction</dt>
<dd><p>Beta coefficient direction, either <code>'decrease'</code> or
<code>'increase'</code>.</p></dd>

<dt>beta_description</dt>
<dd><p>Additional beta coefficient comment.</p></dd>

<dt>last_mapping_date</dt>
<dd><p>Last time this association was mapped to Ensembl.</p></dd>

<dt>last_update_date</dt>
<dd><p>Last time this association was updated.</p></dd>


</dl></dd>


<dt><code>loci</code></dt>
<dd><p>A <code><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></code> listing loci. Columns:</p><dl><dt>association_id</dt>
<dd><p>GWAS Catalog association accession identifier, e.g.,
<code>"20250"</code>.</p></dd>

<dt>locus_id</dt>
<dd><p>A locus identifier referring to a single variant locus or to
a multi-loci entity such as a multi-SNP haplotype.</p></dd>

<dt>haplotype_snp_count</dt>
<dd><p>Number of variants per locus. Most loci are
single-SNP loci, i.e., there is a one to one relationship between a variant
and a <code>locus_id</code> (<code>haplotype_snp_count == NA</code>). There are however
cases of associations involving multiple loci at once, such as SNP-SNP
interactions and multi-SNP haplotypes. This is signalled in the columns:
<code>multiple_snp_haplotype</code> and <code>snp_interaction</code> with value
<code>TRUE</code>.</p></dd>

<dt>description</dt>
<dd><p>Description of the locus identifier, e.g.,
<code>'Single variant'</code>, <code>SNP x SNP interaction</code>, or <code>3-SNP
Haplotype</code>.</p></dd>


</dl></dd>


<dt><code>risk_alleles</code></dt>
<dd><p>A <code><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></code> listing risk alleles.
  Columns:</p><dl><dt>association_id</dt>
<dd><p>GWAS Catalog association accession identifier, e.g.,
<code>"20250"</code>.</p></dd>

<dt>locus_id</dt>
<dd><p>A locus identifier referring to a single variant locus or to
a multi-loci entity such as a multi-SNP haplotype.</p></dd>

<dt>variant_id</dt>
<dd><p>Variant identifier, e.g., <code>'rs1333048'</code>.</p></dd>

<dt>risk_allele</dt>
<dd><p>Risk allele or effect allele.</p></dd>

<dt>risk_frequency</dt>
<dd><p>Reported risk/effect allele frequency associated with
strongest SNP in controls (if not available among all controls, among the
control group with the largest sample size). If the associated locus is a
haplotype the haplotype frequency will be extracted.</p></dd>

<dt>genome_wide</dt>
<dd><p>Whether this variant allele has been part of a genome-wide
study or not.</p></dd>

<dt>limited_list</dt>
<dd><p>Undocumented.</p></dd>


</dl></dd>


<dt><code>genes</code></dt>
<dd><p>A <code><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></code> listing author reported genes.
  Columns:</p><dl><dt>association_id</dt>
<dd><p>GWAS Catalog association accession identifier, e.g.,
<code>"20250"</code>.</p></dd>

<dt>locus_id</dt>
<dd><p>A locus identifier referring to a single variant locus or to
a multi-loci entity such as a multi-SNP haplotype.</p></dd>

<dt>gene_name</dt>
<dd><p>Gene symbol according to
<a href="https://www.genenames.org/" class="external-link">HUGO Gene Nomenclature (HGNC)</a>.</p></dd>

</dl></dd>


<dt><code>ensembl_ids</code></dt>
<dd><p>A <code><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></code> listing Ensembl gene
  identifiers. Columns:</p><dl><dt>association_id</dt>
<dd><p>GWAS Catalog association accession identifier, e.g.,
<code>"20250"</code>.</p></dd>

<dt>locus_id</dt>
<dd><p>A locus identifier referring to a single variant locus or to
a multi-loci entity such as a multi-SNP haplotype.</p></dd>

<dt>gene_name</dt>
<dd><p>Gene symbol according to
<a href="https://www.genenames.org/" class="external-link">HUGO Gene Nomenclature (HGNC)</a>.</p></dd>

<dt>ensembl_id</dt>
<dd><p>The Ensembl identifier of an Ensembl gene, see Section
<a href="https://www.ensembl.org/info/genome/genebuild/index.html" class="external-link">Gene
annotation in Ensembl</a> for more information.</p></dd>

</dl></dd>


<dt><code>entrez_ids</code></dt>
<dd><p>A <code><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></code> listing Entrez gene
  identifiers. Columns:</p><dl><dt>association_id</dt>
<dd><p>GWAS Catalog association accession identifier, e.g.,
<code>"20250"</code>.</p></dd>

<dt>locus_id</dt>
<dd><p>A locus identifier referring to a single variant locus or to
a multi-loci entity such as a multi-SNP haplotype.</p></dd>

<dt>gene_name</dt>
<dd><p>Gene symbol according to
<a href="https://www.genenames.org/" class="external-link">HUGO Gene Nomenclature (HGNC)</a>.</p></dd>

<dt>entrez_id</dt>
<dd><p>The Entrez identifier of a gene, see ref.
<a href="https://doi.org/10.1093/nar/gkq1237" class="external-link">doi:10.1093/nar/gkq1237</a>
 for
more information.</p></dd>


</dl></dd>


</dl></div>

  </main></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://www.pattern.institute/team/rmagno/" class="external-link">Ramiro Magno</a>, Ana-Teresa Maia, CINTESIS, <a href="https://www.pattern.institute" class="external-link">Pattern Institute</a>.</p>
</div>

<div class="pkgdown-footer-right">

</div>

    </footer></div>





  </body></html>

