<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Bind GWAS Catalog objects — bind • gwasrapidd</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



  <link href="../extra.css" rel="stylesheet">
  

<meta property="og:title" content="Bind GWAS Catalog objects — bind" />
<meta property="og:description" content="Binds together GWAS Catalog objects of the same class. Note that
bind() preserves duplicates whereas
union does not." />
<meta property="og:image" content="https://rmagno.eu/gwasrapidd/logo.svg" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gwasrapidd</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.8</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/gwasrapidd.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/faq.html">Frequently asked questions</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/ramiromagno/gwasrapidd/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Bind GWAS Catalog objects</h1>
    <small class="dont-index">Source: <a href='https://github.com/ramiromagno/gwasrapidd/blob/master/R/generics.R'><code>R/generics.R</code></a></small>
    <div class="hidden name"><code>bind.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Binds together GWAS Catalog objects of the same class. Note that
<code>bind()</code> preserves duplicates whereas
<code><a href='setop.html'>union</a></code> does not.</p>
    </div>

    <pre class="usage"><span class='fu'>bind</span>(<span class='no'>x</span>, <span class='no'>...</span>)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>x</th>
      <td><p>An object of class: <a href='studies-class.html'>studies</a>,
<a href='associations-class.html'>associations</a>, <a href='variants-class.html'>variants</a>, or
<a href='traits-class.html'>traits</a>.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Objects of the same class as <code>x</code>.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>An object of the same class as <code>x</code>.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'># Join two studies objects.</span>
<span class='fu'>bind</span>(<span class='no'>studies_ex01</span>, <span class='no'>studies_ex02</span>)</div><div class='output co'>#&gt; An object of class "studies"
#&gt; Slot "studies":
#&gt; # A tibble: 4 x 13
#&gt;   study_id reported_trait initial_sample_… replication_sam… gxe   gxg  
#&gt;   &lt;chr&gt;    &lt;chr&gt;          &lt;chr&gt;            &lt;chr&gt;            &lt;lgl&gt; &lt;lgl&gt;
#&gt; 1 GCST001… Breast size    16,175 European… &lt;NA&gt;             FALSE FALSE
#&gt; 2 GCST003… Breast size    33,790 European… &lt;NA&gt;             FALSE FALSE
#&gt; 3 GCST001… Breast size    16,175 European… &lt;NA&gt;             FALSE FALSE
#&gt; 4 GCST006… Breast size    5,609 Japanese … &lt;NA&gt;             FALSE FALSE
#&gt; # … with 7 more variables: snp_count &lt;int&gt;, qualifier &lt;chr&gt;, imputed &lt;lgl&gt;,
#&gt; #   pooled &lt;lgl&gt;, study_design_comment &lt;chr&gt;, full_pvalue_set &lt;lgl&gt;,
#&gt; #   user_requested &lt;lgl&gt;
#&gt; 
#&gt; Slot "genotyping_techs":
#&gt; # A tibble: 4 x 2
#&gt;   study_id   genotyping_technology       
#&gt;   &lt;chr&gt;      &lt;chr&gt;                       
#&gt; 1 GCST001585 Genome-wide genotyping array
#&gt; 2 GCST003985 Genome-wide genotyping array
#&gt; 3 GCST001585 Genome-wide genotyping array
#&gt; 4 GCST006655 Genome-wide genotyping array
#&gt; 
#&gt; Slot "platforms":
#&gt; # A tibble: 4 x 2
#&gt;   study_id   manufacturer
#&gt;   &lt;chr&gt;      &lt;chr&gt;       
#&gt; 1 GCST001585 Illumina    
#&gt; 2 GCST003985 Illumina    
#&gt; 3 GCST001585 Illumina    
#&gt; 4 GCST006655 Affymetrix  
#&gt; 
#&gt; Slot "ancestries":
#&gt; # A tibble: 4 x 4
#&gt;   study_id   ancestry_id type    number_of_individuals
#&gt;   &lt;chr&gt;            &lt;int&gt; &lt;chr&gt;                   &lt;int&gt;
#&gt; 1 GCST001585           1 initial                 16175
#&gt; 2 GCST003985           1 initial                 33790
#&gt; 3 GCST001585           1 initial                 16175
#&gt; 4 GCST006655           1 initial                  5609
#&gt; 
#&gt; Slot "ancestral_groups":
#&gt; # A tibble: 4 x 3
#&gt;   study_id   ancestry_id ancestral_group
#&gt;   &lt;chr&gt;            &lt;int&gt; &lt;chr&gt;          
#&gt; 1 GCST001585           1 European       
#&gt; 2 GCST003985           1 European       
#&gt; 3 GCST001585           1 European       
#&gt; 4 GCST006655           1 East Asian     
#&gt; 
#&gt; Slot "countries_of_origin":
#&gt; # A tibble: 4 x 5
#&gt;   study_id   ancestry_id country_name major_area region
#&gt;   &lt;chr&gt;            &lt;int&gt; &lt;chr&gt;        &lt;chr&gt;      &lt;chr&gt; 
#&gt; 1 GCST001585           1 &lt;NA&gt;         &lt;NA&gt;       &lt;NA&gt;  
#&gt; 2 GCST003985           1 &lt;NA&gt;         &lt;NA&gt;       &lt;NA&gt;  
#&gt; 3 GCST001585           1 &lt;NA&gt;         &lt;NA&gt;       &lt;NA&gt;  
#&gt; 4 GCST006655           1 &lt;NA&gt;         &lt;NA&gt;       &lt;NA&gt;  
#&gt; 
#&gt; Slot "countries_of_recruitment":
#&gt; # A tibble: 4 x 5
#&gt;   study_id   ancestry_id country_name major_area region
#&gt;   &lt;chr&gt;            &lt;int&gt; &lt;chr&gt;        &lt;chr&gt;      &lt;chr&gt; 
#&gt; 1 GCST001585           1 &lt;NA&gt;         &lt;NA&gt;       &lt;NA&gt;  
#&gt; 2 GCST003985           1 &lt;NA&gt;         &lt;NA&gt;       &lt;NA&gt;  
#&gt; 3 GCST001585           1 &lt;NA&gt;         &lt;NA&gt;       &lt;NA&gt;  
#&gt; 4 GCST006655           1 &lt;NA&gt;         &lt;NA&gt;       &lt;NA&gt;  
#&gt; 
#&gt; Slot "publications":
#&gt; # A tibble: 4 x 7
#&gt;   study_id pubmed_id publication_date publication title author_fullname
#&gt;   &lt;chr&gt;        &lt;int&gt; &lt;date&gt;           &lt;chr&gt;       &lt;chr&gt; &lt;chr&gt;          
#&gt; 1 GCST001…  22747683 2012-06-29       BMC Med Ge… Gene… Eriksson N     
#&gt; 2 GCST003…  27182965 2016-05-15       Nat Genet   Dete… Pickrell JK    
#&gt; 3 GCST001…  22747683 2012-06-29       BMC Med Ge… Gene… Eriksson N     
#&gt; 4 GCST006…  29855537 2018-05-30       Sci Rep     Japa… Hirata T       
#&gt; # … with 1 more variable: author_orcid &lt;chr&gt;
#&gt; </div><div class='input'>
<span class='co'># Join two associations objects.</span>
<span class='fu'>bind</span>(<span class='no'>associations_ex01</span>, <span class='no'>associations_ex02</span>)</div><div class='output co'>#&gt; An object of class "associations"
#&gt; Slot "associations":
#&gt; # A tibble: 7 x 17
#&gt;   association_id   pvalue pvalue_descript… pvalue_mantissa pvalue_exponent
#&gt;   &lt;chr&gt;             &lt;dbl&gt; &lt;chr&gt;                      &lt;int&gt;           &lt;int&gt;
#&gt; 1 22509          3.00e- 9 &lt;NA&gt;                           3              -9
#&gt; 2 22505          7.00e- 6 &lt;NA&gt;                           7              -6
#&gt; 3 19537565       7.00e-31 &lt;NA&gt;                           7             -31
#&gt; 4 19537593       1.00e-11 &lt;NA&gt;                           1             -11
#&gt; 5 19537593       1.00e-11 &lt;NA&gt;                           1             -11
#&gt; 6 31665940       8.00e- 6 &lt;NA&gt;                           8              -6
#&gt; 7 34944736       6.00e- 9 &lt;NA&gt;                           6              -9
#&gt; # … with 12 more variables: multiple_snp_haplotype &lt;lgl&gt;,
#&gt; #   snp_interaction &lt;lgl&gt;, snp_type &lt;chr&gt;, standard_error &lt;dbl&gt;, range &lt;chr&gt;,
#&gt; #   or_per_copy_number &lt;dbl&gt;, beta_number &lt;dbl&gt;, beta_unit &lt;chr&gt;,
#&gt; #   beta_direction &lt;chr&gt;, beta_description &lt;chr&gt;, last_mapping_date &lt;dttm&gt;,
#&gt; #   last_update_date &lt;dttm&gt;
#&gt; 
#&gt; Slot "loci":
#&gt; # A tibble: 7 x 4
#&gt;   association_id locus_id haplotype_snp_count description   
#&gt;   &lt;chr&gt;             &lt;int&gt;               &lt;int&gt; &lt;chr&gt;         
#&gt; 1 22509                 1                  NA Single variant
#&gt; 2 22505                 1                  NA Single variant
#&gt; 3 19537565              1                  NA Single variant
#&gt; 4 19537593              1                  NA Single variant
#&gt; 5 19537593              1                  NA Single variant
#&gt; 6 31665940              1                  NA Single variant
#&gt; 7 34944736              1                  NA Single variant
#&gt; 
#&gt; Slot "risk_alleles":
#&gt; # A tibble: 7 x 7
#&gt;   association_id locus_id variant_id risk_allele risk_frequency genome_wide
#&gt;   &lt;chr&gt;             &lt;int&gt; &lt;chr&gt;      &lt;chr&gt;                &lt;dbl&gt; &lt;lgl&gt;      
#&gt; 1 22509                 1 rs7089814  C                   NA     NA         
#&gt; 2 22505                 1 chr22:458… T                   NA     NA         
#&gt; 3 19537565              1 rs10110651 &lt;NA&gt;                NA     FALSE      
#&gt; 4 19537593              1 rs9397437  &lt;NA&gt;                NA     FALSE      
#&gt; 5 19537593              1 rs9397437  &lt;NA&gt;                NA     FALSE      
#&gt; 6 31665940              1 rs985648   G                    0.248 FALSE      
#&gt; 7 34944736              1 rs1469924… TTCTTTCTTTC          0.381 FALSE      
#&gt; # … with 1 more variable: limited_list &lt;lgl&gt;
#&gt; 
#&gt; Slot "genes":
#&gt; # A tibble: 13 x 3
#&gt;    association_id locus_id gene_name   
#&gt;    &lt;chr&gt;             &lt;int&gt; &lt;chr&gt;       
#&gt;  1 22509                 1 ADO         
#&gt;  2 22509                 1 ZNF365      
#&gt;  3 22509                 1 ALDH7A1P4   
#&gt;  4 22505                 1 TBC1D22A    
#&gt;  5 22505                 1 LOC100128818
#&gt;  6 19537565              1 KCNU1       
#&gt;  7 19537565              1 ZNF703      
#&gt;  8 19537593              1 CCDC170     
#&gt;  9 19537593              1 ESR1        
#&gt; 10 19537593              1 CCDC170     
#&gt; 11 19537593              1 ESR1        
#&gt; 12 31665940              1 &lt;NA&gt;        
#&gt; 13 34944736              1 KCNU1       
#&gt; 
#&gt; Slot "ensembl_ids":
#&gt; # A tibble: 13 x 4
#&gt;    association_id locus_id gene_name    ensembl_id     
#&gt;    &lt;chr&gt;             &lt;int&gt; &lt;chr&gt;        &lt;chr&gt;          
#&gt;  1 22509                 1 ADO          ENSG00000181915
#&gt;  2 22509                 1 ZNF365       ENSG00000138311
#&gt;  3 22509                 1 ALDH7A1P4    ENSG00000234489
#&gt;  4 22505                 1 TBC1D22A     ENSG00000054611
#&gt;  5 22505                 1 LOC100128818 &lt;NA&gt;           
#&gt;  6 19537565              1 KCNU1        ENSG00000215262
#&gt;  7 19537565              1 ZNF703       ENSG00000183779
#&gt;  8 19537593              1 CCDC170      ENSG00000120262
#&gt;  9 19537593              1 ESR1         ENSG00000091831
#&gt; 10 19537593              1 CCDC170      ENSG00000120262
#&gt; 11 19537593              1 ESR1         ENSG00000091831
#&gt; 12 31665940              1 &lt;NA&gt;         &lt;NA&gt;           
#&gt; 13 34944736              1 KCNU1        ENSG00000215262
#&gt; 
#&gt; Slot "entrez_ids":
#&gt; # A tibble: 13 x 4
#&gt;    association_id locus_id gene_name    entrez_id
#&gt;    &lt;chr&gt;             &lt;int&gt; &lt;chr&gt;        &lt;chr&gt;    
#&gt;  1 22509                 1 ADO          84890    
#&gt;  2 22509                 1 ZNF365       22891    
#&gt;  3 22509                 1 ALDH7A1P4    544      
#&gt;  4 22505                 1 TBC1D22A     25771    
#&gt;  5 22505                 1 LOC100128818 &lt;NA&gt;     
#&gt;  6 19537565              1 KCNU1        157855   
#&gt;  7 19537565              1 ZNF703       80139    
#&gt;  8 19537593              1 CCDC170      80129    
#&gt;  9 19537593              1 ESR1         2099     
#&gt; 10 19537593              1 CCDC170      80129    
#&gt; 11 19537593              1 ESR1         2099     
#&gt; 12 31665940              1 &lt;NA&gt;         &lt;NA&gt;     
#&gt; 13 34944736              1 KCNU1        157855   
#&gt; </div><div class='input'>
<span class='co'># Join two variants objects.</span>
<span class='fu'>bind</span>(<span class='no'>variants_ex01</span>, <span class='no'>variants_ex02</span>)</div><div class='output co'>#&gt; An object of class "variants"
#&gt; Slot "variants":
#&gt; # A tibble: 7 x 7
#&gt;   variant_id merged functional_class chromosome_name chromosome_posi…
#&gt;   &lt;chr&gt;       &lt;int&gt; &lt;chr&gt;            &lt;chr&gt;                      &lt;int&gt;
#&gt; 1 rs1469924…      0 regulatory_regi… 8                       36984538
#&gt; 2 rs56261590      0 intron_variant   4                       73189781
#&gt; 3 rs4725504       0 intergenic_vari… 7                      153608338
#&gt; 4 rs56261590      0 intron_variant   4                       73189781
#&gt; 5 rs4725504       0 intergenic_vari… 7                      153608338
#&gt; 6 rs11099757      0 intron_variant   4                      150143216
#&gt; 7 rs16871509      0 intergenic_vari… 5                       71726395
#&gt; # … with 2 more variables: chromosome_region &lt;chr&gt;, last_update_date &lt;dttm&gt;
#&gt; 
#&gt; Slot "genomic_contexts":
#&gt; # A tibble: 44 x 11
#&gt;    variant_id gene_name chromosome_name chromosome_posi… distance
#&gt;    &lt;chr&gt;      &lt;chr&gt;     &lt;chr&gt;                      &lt;int&gt;    &lt;int&gt;
#&gt;  1 rs1469924… LOC10537… 8                       36984538    81555
#&gt;  2 rs1469924… KCNU1     8                       36984538    48413
#&gt;  3 rs1469924… LOC10537… 8                       36984538     2533
#&gt;  4 rs1469924… TPT1P8    8                       36984538    95231
#&gt;  5 rs1469924… RPL26P25  8                       36984538     3392
#&gt;  6 rs1469924… KCNU1     8                       36984538    48413
#&gt;  7 rs1469924… AC090453… 8                       36984538     3439
#&gt;  8 rs1469924… TPT1P8    8                       36984538    95008
#&gt;  9 rs1469924… AC092818… 8                       36984538    82903
#&gt; 10 rs1469924… LOC10537… 8                       36984538    59331
#&gt; # … with 34 more rows, and 6 more variables: is_closest_gene &lt;lgl&gt;,
#&gt; #   is_intergenic &lt;lgl&gt;, is_upstream &lt;lgl&gt;, is_downstream &lt;lgl&gt;, source &lt;chr&gt;,
#&gt; #   mapping_method &lt;chr&gt;
#&gt; 
#&gt; Slot "ensembl_ids":
#&gt; # A tibble: 23 x 3
#&gt;    variant_id  gene_name  ensembl_id     
#&gt;    &lt;chr&gt;       &lt;chr&gt;      &lt;chr&gt;          
#&gt;  1 rs146992477 KCNU1      ENSG00000215262
#&gt;  2 rs146992477 TPT1P8     ENSG00000219491
#&gt;  3 rs146992477 AC090453.1 ENSG00000243503
#&gt;  4 rs146992477 AC092818.1 ENSG00000254038
#&gt;  5 rs56261590  ANKRD17    ENSG00000132466
#&gt;  6 rs56261590  HMGA1P2    ENSG00000248641
#&gt;  7 rs56261590  AC053527.1 ENSG00000250220
#&gt;  8 rs56261590  RF00334    ENSG00000221093
#&gt;  9 rs4725504   DPP6       ENSG00000130226
#&gt; 10 rs4725504   PAXBP1P1   ENSG00000233489
#&gt; # … with 13 more rows
#&gt; 
#&gt; Slot "entrez_ids":
#&gt; # A tibble: 24 x 3
#&gt;    variant_id  gene_name    entrez_id
#&gt;    &lt;chr&gt;       &lt;chr&gt;        &lt;chr&gt;    
#&gt;  1 rs146992477 LOC105379377 105379377
#&gt;  2 rs146992477 KCNU1        157855   
#&gt;  3 rs146992477 LOC105379376 105379376
#&gt;  4 rs146992477 TPT1P8       59347    
#&gt;  5 rs146992477 RPL26P25     100270966
#&gt;  6 rs146992477 LOC105379375 105379375
#&gt;  7 rs56261590  ANKRD17      26057    
#&gt;  8 rs56261590  HMGA1P2      171559   
#&gt;  9 rs56261590  LOC102724832 102724832
#&gt; 10 rs4725504   DPP6         1804     
#&gt; # … with 14 more rows
#&gt; </div><div class='input'>
<span class='co'># Join two traits objects.</span>
<span class='fu'>bind</span>(<span class='no'>traits_ex01</span>, <span class='no'>traits_ex02</span>)</div><div class='output co'>#&gt; An object of class "traits"
#&gt; Slot "traits":
#&gt; # A tibble: 7 x 3
#&gt;   efo_id      trait                      uri                                 
#&gt;   &lt;chr&gt;       &lt;chr&gt;                      &lt;chr&gt;                               
#&gt; 1 EFO_0004884 breast size                http://www.ebi.ac.uk/efo/EFO_0004884
#&gt; 2 EFO_0004343 waist-hip ratio            http://www.ebi.ac.uk/efo/EFO_0004343
#&gt; 3 EFO_0005299 non-word reading           http://www.ebi.ac.uk/efo/EFO_0005299
#&gt; 4 EFO_0007845 lip morphology measurement http://www.ebi.ac.uk/efo/EFO_0007845
#&gt; 5 EFO_0004699 gambling                   http://www.ebi.ac.uk/efo/EFO_0004699
#&gt; 6 EFO_0004884 breast size                http://www.ebi.ac.uk/efo/EFO_0004884
#&gt; 7 EFO_0004875 mathematical ability       http://www.ebi.ac.uk/efo/EFO_0004875
#&gt; </div><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Ramiro Magno, Ana-Teresa Maia, Centre for Biomedical Research, University of Algarve.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


