<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Get GWAS Catalog variants — get_variants • gwasrapidd</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



  <link href="../extra.css" rel="stylesheet">
  

<meta property="og:title" content="Get GWAS Catalog variants — get_variants" />
<meta property="og:description" content="Retrieves variants via the NHGRI-EBI GWAS Catalog REST API. The REST
API is queried multiple times with the criteria passed as arguments (see
below). By default all variants that match the criteria supplied in the
arguments are retrieved: this corresponds to the default option
set_operation set to 'union'. If you rather have only the
variants that match simultaneously all criteria provided, then set
set_operation to 'intersection'." />
<meta property="og:image" content="https://rmagno.eu/gwasrapidd/logo.svg" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gwasrapidd</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.11</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/gwasrapidd.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/bmi_variants.html">Variants associated with Body Mass Index (BMI)</a>
    </li>
    <li>
      <a href="../articles/faq.html">Frequently asked questions</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/ramiromagno/gwasrapidd/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Get GWAS Catalog variants</h1>
    <small class="dont-index">Source: <a href='https://github.com/ramiromagno/gwasrapidd/blob/master/R/get_variants.R'><code>R/get_variants.R</code></a></small>
    <div class="hidden name"><code>get_variants.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Retrieves variants via the NHGRI-EBI GWAS Catalog REST API. The REST
API is queried multiple times with the criteria passed as arguments (see
below). By default all variants that match the criteria supplied in the
arguments are retrieved: this corresponds to the default option
<code>set_operation</code> set to <code>'union'</code>. If you rather have only the
variants that match simultaneously all criteria provided, then set
<code>set_operation</code> to <code>'intersection'</code>.</p>
    </div>

    <pre class="usage"><span class='fu'>get_variants</span><span class='op'>(</span>
  study_id <span class='op'>=</span> <span class='cn'>NULL</span>,
  association_id <span class='op'>=</span> <span class='cn'>NULL</span>,
  variant_id <span class='op'>=</span> <span class='cn'>NULL</span>,
  efo_id <span class='op'>=</span> <span class='cn'>NULL</span>,
  pubmed_id <span class='op'>=</span> <span class='cn'>NULL</span>,
  genomic_range <span class='op'>=</span> <span class='cn'>NULL</span>,
  cytogenetic_band <span class='op'>=</span> <span class='cn'>NULL</span>,
  gene_name <span class='op'>=</span> <span class='cn'>NULL</span>,
  efo_trait <span class='op'>=</span> <span class='cn'>NULL</span>,
  reported_trait <span class='op'>=</span> <span class='cn'>NULL</span>,
  set_operation <span class='op'>=</span> <span class='st'>"union"</span>,
  interactive <span class='op'>=</span> <span class='cn'>TRUE</span>,
  std_chromosomes_only <span class='op'>=</span> <span class='cn'>TRUE</span>,
  verbose <span class='op'>=</span> <span class='cn'>FALSE</span>,
  warnings <span class='op'>=</span> <span class='cn'>TRUE</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>study_id</th>
      <td><p>A character vector of GWAS Catalog study accession
identifiers.</p></td>
    </tr>
    <tr>
      <th>association_id</th>
      <td><p>A character vector of GWAS Catalog association
identifiers.</p></td>
    </tr>
    <tr>
      <th>variant_id</th>
      <td><p>A character vector of GWAS Catalog variant identifiers.</p></td>
    </tr>
    <tr>
      <th>efo_id</th>
      <td><p>A character vector of <a href='https://www.ebi.ac.uk/efo/'>EFO</a>
identifiers.</p></td>
    </tr>
    <tr>
      <th>pubmed_id</th>
      <td><p>An integer vector of
<a href='https://pubmed.ncbi.nlm.nih.gov/'>PubMed</a> identifiers.</p></td>
    </tr>
    <tr>
      <th>genomic_range</th>
      <td><p>A named list of three vectors:</p><dl>
<dt>chromosome</dt><dd><p>A character vector of chromosome names of the
form 1--22, X or Y.</p></dd> <dt>start</dt><dd><p>A numeric vector of start positions,
starting at 1.</p></dd> <dt>end</dt><dd><p>A numeric vector of end positions.</p></dd> 
</dl><p>The three vectors need to be of the same length so that <code>chromosome</code>
names, <code>start</code> and <code>end</code> positions can be matched by position.</p></td>
    </tr>
    <tr>
      <th>cytogenetic_band</th>
      <td><p>A character vector of cytogenetic bands of the form
<code>'1p36.11'</code>.</p></td>
    </tr>
    <tr>
      <th>gene_name</th>
      <td><p>Gene symbol according to
<a href='https://www.genenames.org/'>HUGO Gene Nomenclature (HGNC)</a>.</p></td>
    </tr>
    <tr>
      <th>efo_trait</th>
      <td><p>A character vector of
<a href='https://www.ebi.ac.uk/efo/'>EFO</a> trait descriptions, e.g.,
<code>'uric acid measurement'</code>.</p></td>
    </tr>
    <tr>
      <th>reported_trait</th>
      <td><p>A character vector of phenotypic traits as
reported by the original authors of the study.</p></td>
    </tr>
    <tr>
      <th>set_operation</th>
      <td><p>Either <code>'union'</code> or <code>'intersection'</code>. This
tells how variants retrieved by different criteria  should be combined:
<code>'union'</code> binds together all results removing duplicates and
<code>'intersection'</code> only keeps same variants found with different
criteria.</p></td>
    </tr>
    <tr>
      <th>interactive</th>
      <td><p>A logical. If all variants are requested, whether to ask
interactively if we really want to proceed.</p></td>
    </tr>
    <tr>
      <th>std_chromosomes_only</th>
      <td><p>Whether to return only variants mapped to
standard chromosomes: 1 thru 22, X, Y, and MT.</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>Whether the function should be
verbose about the different queries or not.</p></td>
    </tr>
    <tr>
      <th>warnings</th>
      <td><p>Whether to print warnings.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A <a href='variants-class.html'>variants</a> object.</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>Please note that all search criteria are vectorised, thus allowing for batch
mode search, e.g., one can search by multiple variant identifiers at once by
passing a vector of identifiers to <code>variant_id</code>.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><span class='r-in'><span class='co'># Get variants by study identifier</span></span>
<span class='r-in'><span class='fu'>get_variants</span><span class='op'>(</span>study_id <span class='op'>=</span> <span class='st'>'GCST001085'</span><span class='op'>)</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> An object of class "variants"</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> Slot "variants":</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #949494;'># A tibble: 18 × 7</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>    variant_id merged functional_class          chromosome_name chromosome_posit…</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>    <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>       <span style='color: #949494; font-style: italic;'>&lt;int&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>                     <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>                       <span style='color: #949494; font-style: italic;'>&lt;int&gt;</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 1</span> rs10496288      0 intergenic_variant        <span style='color: #BB0000;'>NA</span>                             <span style='color: #BB0000;'>NA</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 2</span> rs7827545       0 intron_variant            <span style='color: #BB0000;'>NA</span>                             <span style='color: #BB0000;'>NA</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 3</span> rs10785581      0 intron_variant            12                       45<span style='text-decoration: underline;'>537</span>190</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 4</span> rs200759        0 intron_variant            20                       15<span style='text-decoration: underline;'>625</span>776</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 5</span> rs7960483       0 intron_variant            <span style='color: #BB0000;'>NA</span>                             <span style='color: #BB0000;'>NA</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 6</span> rs1372662       0 intron_variant            8                       134<span style='text-decoration: underline;'>554</span>803</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 7</span> rs10496289      0 intergenic_variant        2                        83<span style='text-decoration: underline;'>066</span>256</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 8</span> rs6452524       0 intron_variant            <span style='color: #BB0000;'>NA</span>                             <span style='color: #BB0000;'>NA</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 9</span> rs200752        0 intron_variant            <span style='color: #BB0000;'>NA</span>                             <span style='color: #BB0000;'>NA</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'>10</span> rs3798440       0 intron_variant            <span style='color: #BB0000;'>NA</span>                             <span style='color: #BB0000;'>NA</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'>11</span> rs7735940       0 regulatory_region_variant <span style='color: #BB0000;'>NA</span>                             <span style='color: #BB0000;'>NA</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'>12</span> rs9350602       0 intron_variant            6                        75<span style='text-decoration: underline;'>850</span>781</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'>13</span> rs10188442      0 intron_variant            2                       132<span style='text-decoration: underline;'>431</span>666</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'>14</span> rs12522034      0 intergenic_variant        5                        36<span style='text-decoration: underline;'>425</span>491</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'>15</span> rs13420028      0 intron_variant            <span style='color: #BB0000;'>NA</span>                             <span style='color: #BB0000;'>NA</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'>16</span> rs2469997       0 intergenic_variant        8                       119<span style='text-decoration: underline;'>341</span>027</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'>17</span> rs6887846       0 intron_variant            5                        83<span style='text-decoration: underline;'>140</span>542</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'>18</span> rs6469823       1 intergenic_variant        8                       119<span style='text-decoration: underline;'>341</span>744</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #949494;'># … with 2 more variables: chromosome_region &lt;chr&gt;, last_update_date &lt;dttm&gt;</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> Slot "genomic_contexts":</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #949494;'># A tibble: 60 × 12</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>    variant_id gene_name chromosome_name chromosome_posi… distance is_mapped_gene</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>    <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>      <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>     <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>                      <span style='color: #949494; font-style: italic;'>&lt;int&gt;</span>    <span style='color: #949494; font-style: italic;'>&lt;int&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;lgl&gt;</span>         </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 1</span> rs10785581 LOC10536… 12                      45<span style='text-decoration: underline;'>537</span>190    <span style='text-decoration: underline;'>58</span>636 FALSE         </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 2</span> rs10785581 ANO6      12                      45<span style='text-decoration: underline;'>537</span>190    <span style='text-decoration: underline;'>96</span>786 FALSE         </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 3</span> rs10785581 ANO6      12                      45<span style='text-decoration: underline;'>537</span>190    <span style='text-decoration: underline;'>54</span>910 TRUE          </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 4</span> rs10785581 U6        12                      45<span style='text-decoration: underline;'>537</span>190    <span style='text-decoration: underline;'>17</span>400 TRUE          </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 5</span> rs10785581 LOC10012… 12                      45<span style='text-decoration: underline;'>537</span>190    <span style='text-decoration: underline;'>91</span>755 FALSE         </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 6</span> rs10785581 LOC10536… 12                      45<span style='text-decoration: underline;'>537</span>190        0 FALSE         </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 7</span> rs200759   MACROD2   20                      15<span style='text-decoration: underline;'>625</span>776        0 FALSE         </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 8</span> rs200759   LOC10798… 20                      15<span style='text-decoration: underline;'>625</span>776     <span style='text-decoration: underline;'>1</span>048 FALSE         </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 9</span> rs200759   ENSAP1    20                      15<span style='text-decoration: underline;'>625</span>776     <span style='text-decoration: underline;'>5</span>991 FALSE         </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'>10</span> rs200759   ENSAP1    20                      15<span style='text-decoration: underline;'>625</span>776     <span style='text-decoration: underline;'>5</span>957 FALSE         </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #949494;'># … with 50 more rows, and 6 more variables: is_closest_gene &lt;lgl&gt;,</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #949494;'>#   is_intergenic &lt;lgl&gt;, is_upstream &lt;lgl&gt;, is_downstream &lt;lgl&gt;, source &lt;chr&gt;,</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #949494;'>#   mapping_method &lt;chr&gt;</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> Slot "ensembl_ids":</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #949494;'># A tibble: 34 × 3</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>    variant_id gene_name ensembl_id     </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>    <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>      <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>     <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>          </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 1</span> rs10785581 ANO6      ENSG00000177119</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 2</span> rs10785581 U6        ENSG00000264557</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 3</span> rs10785581 U6        ENSG00000265920</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 4</span> rs10785581 U6        ENSG00000263869</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 5</span> rs10785581 U6        ENSG00000265351</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 6</span> rs10785581 U6        ENSG00000266660</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 7</span> rs10785581 U6        ENSG00000265048</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 8</span> rs10785581 U6        ENSG00000266435</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 9</span> rs200759   MACROD2   ENSG00000172264</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'>10</span> rs200759   ENSAP1    ENSG00000224274</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #949494;'># … with 24 more rows</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> Slot "entrez_ids":</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #949494;'># A tibble: 32 × 3</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>    variant_id gene_name    entrez_id</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>    <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>      <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>        <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>    </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 1</span> rs10785581 LOC105369744 105369744</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 2</span> rs10785581 ANO6         196527   </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 3</span> rs10785581 LOC100128248 100128248</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 4</span> rs10785581 LOC105369743 105369743</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 5</span> rs200759   MACROD2      140733   </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 6</span> rs200759   LOC107985394 107985394</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 7</span> rs200759   ENSAP1       170511   </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 8</span> rs1372662  ZFAT-AS1     594840   </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 9</span> rs1372662  ZFAT         57623    </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'>10</span> rs9350602  MYO6         4646     </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #949494;'># … with 22 more rows</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-in'></span>
<span class='r-in'><span class='co'># Get variants by association identifier</span></span>
<span class='r-in'><span class='fu'>get_variants</span><span class='op'>(</span>association_id <span class='op'>=</span> <span class='st'>'25389945'</span><span class='op'>)</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> An object of class "variants"</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> Slot "variants":</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #949494;'># A tibble: 27 × 7</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>    variant_id merged functional_class          chromosome_name chromosome_posit…</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>    <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>       <span style='color: #949494; font-style: italic;'>&lt;int&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>                     <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>                       <span style='color: #949494; font-style: italic;'>&lt;int&gt;</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 1</span> rs9486815       0 intron_variant            6                       108<span style='text-decoration: underline;'>126</span>842</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 2</span> rs4245535       0 intron_variant            <span style='color: #BB0000;'>NA</span>                             <span style='color: #BB0000;'>NA</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 3</span> rs17069173      0 intergenic_variant        <span style='color: #BB0000;'>NA</span>                             <span style='color: #BB0000;'>NA</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 4</span> rs9374021       0 intron_variant            <span style='color: #BB0000;'>NA</span>                             <span style='color: #BB0000;'>NA</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 5</span> rs9374007       0 intergenic_variant        <span style='color: #BB0000;'>NA</span>                             <span style='color: #BB0000;'>NA</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 6</span> rs9386694       0 intergenic_variant        <span style='color: #BB0000;'>NA</span>                             <span style='color: #BB0000;'>NA</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 7</span> rs9374002       0 non_coding_transcript_ex… <span style='color: #BB0000;'>NA</span>                             <span style='color: #BB0000;'>NA</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 8</span> rs218289        0 intron_variant            <span style='color: #BB0000;'>NA</span>                             <span style='color: #BB0000;'>NA</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 9</span> rs1064346       0 3_prime_UTR_variant       <span style='color: #BB0000;'>NA</span>                             <span style='color: #BB0000;'>NA</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'>10</span> rs9374013       0 intron_variant            <span style='color: #BB0000;'>NA</span>                             <span style='color: #BB0000;'>NA</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #949494;'># … with 17 more rows, and 2 more variables: chromosome_region &lt;chr&gt;,</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #949494;'>#   last_update_date &lt;dttm&gt;</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> Slot "genomic_contexts":</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #949494;'># A tibble: 10 × 12</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>    variant_id gene_name chromosome_name chromosome_posi… distance is_mapped_gene</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>    <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>      <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>     <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>                      <span style='color: #949494; font-style: italic;'>&lt;int&gt;</span>    <span style='color: #949494; font-style: italic;'>&lt;int&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;lgl&gt;</span>         </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 1</span> rs9486815  SNX3      6                      108<span style='text-decoration: underline;'>126</span>842    <span style='text-decoration: underline;'>84</span>380 FALSE         </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 2</span> rs9486815  SNX3      6                      108<span style='text-decoration: underline;'>126</span>842    <span style='text-decoration: underline;'>84</span>380 FALSE         </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 3</span> rs9486815  NR2E1     6                      108<span style='text-decoration: underline;'>126</span>842    <span style='text-decoration: underline;'>39</span>180 FALSE         </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 4</span> rs9486815  Y_RNA     6                      108<span style='text-decoration: underline;'>126</span>842    <span style='text-decoration: underline;'>65</span>912 FALSE         </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 5</span> rs9486815  OSTM1     6                      108<span style='text-decoration: underline;'>126</span>842        0 TRUE          </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 6</span> rs9486815  NR2E1     6                      108<span style='text-decoration: underline;'>126</span>842    <span style='text-decoration: underline;'>39</span>180 FALSE         </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 7</span> rs9486815  OSTM1-AS1 6                      108<span style='text-decoration: underline;'>126</span>842        0 FALSE         </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 8</span> rs9486815  LOC10537… 6                      108<span style='text-decoration: underline;'>126</span>842    <span style='text-decoration: underline;'>86</span>455 FALSE         </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 9</span> rs9486815  OSTM1     6                      108<span style='text-decoration: underline;'>126</span>842    <span style='text-decoration: underline;'>52</span>101 FALSE         </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'>10</span> rs9486815  OSTM1-AS1 6                      108<span style='text-decoration: underline;'>126</span>842        0 TRUE          </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #949494;'># … with 6 more variables: is_closest_gene &lt;lgl&gt;, is_intergenic &lt;lgl&gt;,</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #949494;'>#   is_upstream &lt;lgl&gt;, is_downstream &lt;lgl&gt;, source &lt;chr&gt;, mapping_method &lt;chr&gt;</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> Slot "ensembl_ids":</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #949494;'># A tibble: 5 × 3</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   variant_id gene_name ensembl_id     </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>      <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>     <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>          </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'>1</span> rs9486815  SNX3      ENSG00000112335</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'>2</span> rs9486815  NR2E1     ENSG00000112333</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'>3</span> rs9486815  Y_RNA     ENSG00000202533</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'>4</span> rs9486815  OSTM1     ENSG00000081087</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'>5</span> rs9486815  OSTM1-AS1 ENSG00000225174</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> Slot "entrez_ids":</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #949494;'># A tibble: 5 × 3</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   variant_id gene_name    entrez_id</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>      <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>        <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>    </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'>1</span> rs9486815  SNX3         8724     </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'>2</span> rs9486815  NR2E1        7101     </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'>3</span> rs9486815  OSTM1        28962    </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'>4</span> rs9486815  OSTM1-AS1    100287366</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'>5</span> rs9486815  LOC105377929 105377929</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-in'></span>
<span class='r-in'><span class='co'># Get a variant by its identifier</span></span>
<span class='r-in'><span class='fu'>get_variants</span><span class='op'>(</span>variant_id <span class='op'>=</span> <span class='st'>'rs3798440'</span><span class='op'>)</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> An object of class "variants"</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> Slot "variants":</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #949494;'># A tibble: 1 × 7</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   variant_id merged functional_class chromosome_name chromosome_position</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>       <span style='color: #949494; font-style: italic;'>&lt;int&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>            <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>                         <span style='color: #949494; font-style: italic;'>&lt;int&gt;</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'>1</span> rs3798440       0 intron_variant   <span style='color: #BB0000;'>NA</span>                               <span style='color: #BB0000;'>NA</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #949494;'># … with 2 more variables: chromosome_region &lt;chr&gt;, last_update_date &lt;dttm&gt;</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> Slot "genomic_contexts":</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #949494;'># A tibble: 0 × 12</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #949494;'># … with 12 variables: variant_id &lt;chr&gt;, gene_name &lt;chr&gt;,</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #949494;'>#   chromosome_name &lt;chr&gt;, chromosome_position &lt;int&gt;, distance &lt;int&gt;,</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #949494;'>#   is_mapped_gene &lt;lgl&gt;, is_closest_gene &lt;lgl&gt;, is_intergenic &lt;lgl&gt;,</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #949494;'>#   is_upstream &lt;lgl&gt;, is_downstream &lt;lgl&gt;, source &lt;chr&gt;, mapping_method &lt;chr&gt;</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> Slot "ensembl_ids":</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #949494;'># A tibble: 0 × 3</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #949494;'># … with 3 variables: variant_id &lt;chr&gt;, gene_name &lt;chr&gt;, ensembl_id &lt;chr&gt;</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> Slot "entrez_ids":</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #949494;'># A tibble: 0 × 3</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #949494;'># … with 3 variables: variant_id &lt;chr&gt;, gene_name &lt;chr&gt;, entrez_id &lt;chr&gt;</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-in'></span>
<span class='r-in'><span class='co'># Get variants by EFO trait identifier</span></span>
<span class='r-in'><span class='fu'>get_variants</span><span class='op'>(</span>efo_id <span class='op'>=</span> <span class='st'>'EFO_0005537'</span><span class='op'>)</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> An object of class "variants"</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> Slot "variants":</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #949494;'># A tibble: 8 × 7</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   variant_id merged functional_class    chromosome_name chromosome_position</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>   <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>       <span style='color: #949494; font-style: italic;'>&lt;int&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>               <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>                         <span style='color: #949494; font-style: italic;'>&lt;int&gt;</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'>1</span> rs2464195       0 missense_variant    12                        120<span style='text-decoration: underline;'>997</span>672</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'>2</span> rs17215231      0 5_prime_UTR_variant 6                          33<span style='text-decoration: underline;'>272</span>092</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'>3</span> rs10069690      0 intron_variant      5                           1<span style='text-decoration: underline;'>279</span>675</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'>4</span> rs2363956       0 missense_variant    19                         17<span style='text-decoration: underline;'>283</span>315</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'>5</span> rs4245739       0 3_prime_UTR_variant 1                         204<span style='text-decoration: underline;'>549</span>714</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'>6</span> rs10771399      0 intergenic_variant  12                         28<span style='text-decoration: underline;'>002</span>147</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'>7</span> rs3757318       0 intron_variant      6                         151<span style='text-decoration: underline;'>592</span>978</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'>8</span> rs78378222      0 3_prime_UTR_variant 17                          7<span style='text-decoration: underline;'>668</span>434</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #949494;'># … with 2 more variables: chromosome_region &lt;chr&gt;, last_update_date &lt;dttm&gt;</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> Slot "genomic_contexts":</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #949494;'># A tibble: 359 × 12</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>    variant_id gene_name chromosome_name chromosome_posi… distance is_mapped_gene</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>    <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>      <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>     <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>                      <span style='color: #949494; font-style: italic;'>&lt;int&gt;</span>    <span style='color: #949494; font-style: italic;'>&lt;int&gt;</span> <span style='color: #949494; font-style: italic;'>&lt;lgl&gt;</span>         </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 1</span> rs2464195  OASL2P    12                     120<span style='text-decoration: underline;'>997</span>672    <span style='text-decoration: underline;'>56</span>060 FALSE         </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 2</span> rs2464195  CLIC1P1   12                     120<span style='text-decoration: underline;'>997</span>672    <span style='text-decoration: underline;'>82</span>549 FALSE         </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 3</span> rs2464195  OASL      12                     120<span style='text-decoration: underline;'>997</span>672    <span style='text-decoration: underline;'>21</span>210 FALSE         </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 4</span> rs2464195  XLOC_009… 12                     120<span style='text-decoration: underline;'>997</span>672    <span style='text-decoration: underline;'>89</span>598 FALSE         </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 5</span> rs2464195  HNF1A     12                     120<span style='text-decoration: underline;'>997</span>672        0 TRUE          </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 6</span> rs2464195  RPL12P33  12                     120<span style='text-decoration: underline;'>997</span>672    <span style='text-decoration: underline;'>80</span>436 FALSE         </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 7</span> rs2464195  LOC10537… 12                     120<span style='text-decoration: underline;'>997</span>672    <span style='text-decoration: underline;'>71</span>187 FALSE         </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 8</span> rs2464195  C12orf43  12                     120<span style='text-decoration: underline;'>997</span>672     <span style='text-decoration: underline;'>2</span>814 FALSE         </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 9</span> rs2464195  RPL12P33  12                     120<span style='text-decoration: underline;'>997</span>672    <span style='text-decoration: underline;'>80</span>387 FALSE         </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'>10</span> rs2464195  C12orf43  12                     120<span style='text-decoration: underline;'>997</span>672     <span style='text-decoration: underline;'>2</span>814 FALSE         </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #949494;'># … with 349 more rows, and 6 more variables: is_closest_gene &lt;lgl&gt;,</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #949494;'>#   is_intergenic &lt;lgl&gt;, is_upstream &lt;lgl&gt;, is_downstream &lt;lgl&gt;, source &lt;chr&gt;,</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #949494;'>#   mapping_method &lt;chr&gt;</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> Slot "ensembl_ids":</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #949494;'># A tibble: 154 × 3</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>    variant_id gene_name ensembl_id     </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>    <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>      <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>     <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>          </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 1</span> rs2464195  OASL2P    ENSG00000283542</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 2</span> rs2464195  CLIC1P1   ENSG00000231313</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 3</span> rs2464195  OASL      ENSG00000135114</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 4</span> rs2464195  HNF1A     ENSG00000135100</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 5</span> rs2464195  RPL12P33  ENSG00000244585</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 6</span> rs2464195  C12orf43  ENSG00000157895</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 7</span> rs2464195  SPPL3     ENSG00000157837</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 8</span> rs2464195  HNF1A-AS1 ENSG00000241388</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 9</span> rs17215231 ZBTB22    ENSG00000236104</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'>10</span> rs17215231 ZBTB22    ENSG00000206280</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #949494;'># … with 144 more rows</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> Slot "entrez_ids":</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #949494;'># A tibble: 89 × 3</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>    variant_id gene_name    entrez_id</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span>    <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>      <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>        <span style='color: #949494; font-style: italic;'>&lt;chr&gt;</span>    </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 1</span> rs2464195  OASL2P       111216278</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 2</span> rs2464195  CLIC1P1      390363   </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 3</span> rs2464195  OASL         8638     </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 4</span> rs2464195  XLOC_009911  105500240</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 5</span> rs2464195  HNF1A        6927     </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 6</span> rs2464195  RPL12P33     643550   </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 7</span> rs2464195  LOC105378258 105378258</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 8</span> rs2464195  C12orf43     64897    </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'> 9</span> rs2464195  SPPL3        121665   </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #BCBCBC;'>10</span> rs2464195  HNF1A-AS1    283460   </span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> <span style='color: #949494;'># … with 79 more rows</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> </span>
<span class='r-in'></span>
</pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by <a href='https://rmagno.eu/'>Ramiro Magno</a>, Ana-Teresa Maia, <a href='https://cintesis.eu/'><img src='reference/figures/cintesis.svg' height=24></a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


