<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Retrieves variants via the NHGRI-EBI GWAS Catalog REST API. The REST
API is queried multiple times with the criteria passed as arguments (see
below). By default all variants that match the criteria supplied in the
arguments are retrieved: this corresponds to the default option
set_operation set to 'union'. If you rather have only the
variants that match simultaneously all criteria provided, then set
set_operation to 'intersection'."><title>Get GWAS Catalog variants — get_variants • gwasrapidd</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Get GWAS Catalog variants — get_variants"><meta property="og:description" content="Retrieves variants via the NHGRI-EBI GWAS Catalog REST API. The REST
API is queried multiple times with the criteria passed as arguments (see
below). By default all variants that match the criteria supplied in the
arguments are retrieved: this corresponds to the default option
set_operation set to 'union'. If you rather have only the
variants that match simultaneously all criteria provided, then set
set_operation to 'intersection'."><meta property="og:image" content="https://rmagno.eu/gwasrapidd/logo.svg"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">gwasrapidd</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.13</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/gwasrapidd.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/bmi_variants.html">Variants associated with Body Mass Index (BMI)</a>
    <a class="dropdown-item" href="../articles/faq.html">Frequently asked questions</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/ramiromagno/gwasrapidd/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>Get GWAS Catalog variants</h1>
      <small class="dont-index">Source: <a href="https://github.com/ramiromagno/gwasrapidd/blob/HEAD/R/get_variants.R" class="external-link"><code>R/get_variants.R</code></a></small>
      <div class="d-none name"><code>get_variants.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Retrieves variants via the NHGRI-EBI GWAS Catalog REST API. The REST
API is queried multiple times with the criteria passed as arguments (see
below). By default all variants that match the criteria supplied in the
arguments are retrieved: this corresponds to the default option
<code>set_operation</code> set to <code>'union'</code>. If you rather have only the
variants that match simultaneously all criteria provided, then set
<code>set_operation</code> to <code>'intersection'</code>.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">get_variants</span><span class="op">(</span></span>
<span>  study_id <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  association_id <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  variant_id <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  efo_id <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  pubmed_id <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  genomic_range <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  cytogenetic_band <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  gene_name <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  efo_trait <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  reported_trait <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  set_operation <span class="op">=</span> <span class="st">"union"</span>,</span>
<span>  interactive <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  std_chromosomes_only <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  warnings <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>study_id</dt>
<dd><p>A character vector of GWAS Catalog study accession
identifiers.</p></dd>


<dt>association_id</dt>
<dd><p>A character vector of GWAS Catalog association
identifiers.</p></dd>


<dt>variant_id</dt>
<dd><p>A character vector of GWAS Catalog variant identifiers.</p></dd>


<dt>efo_id</dt>
<dd><p>A character vector of <a href="https://www.ebi.ac.uk/efo/" class="external-link">EFO</a>
identifiers.</p></dd>


<dt>pubmed_id</dt>
<dd><p>An integer vector of
<a href="https://pubmed.ncbi.nlm.nih.gov/" class="external-link">PubMed</a> identifiers.</p></dd>


<dt>genomic_range</dt>
<dd><p>A named list of three vectors:</p><dl><dt>chromosome</dt>
<dd><p>A character vector of chromosome names of the
form 1--22, X or Y.</p></dd>
 <dt>start</dt>
<dd><p>A numeric vector of start positions,
starting at 1.</p></dd>
 <dt>end</dt>
<dd><p>A numeric vector of end positions.</p></dd>
 
</dl><p>The three vectors need to be of the same length so that <code>chromosome</code>
names, <code>start</code> and <code>end</code> positions can be matched by position.</p></dd>


<dt>cytogenetic_band</dt>
<dd><p>A character vector of cytogenetic bands of the form
<code>'1p36.11'</code>.</p></dd>


<dt>gene_name</dt>
<dd><p>Gene symbol according to
<a href="https://www.genenames.org/" class="external-link">HUGO Gene Nomenclature (HGNC)</a>.</p></dd>


<dt>efo_trait</dt>
<dd><p>A character vector of
<a href="https://www.ebi.ac.uk/efo/" class="external-link">EFO</a> trait descriptions, e.g.,
<code>'uric acid measurement'</code>.</p></dd>


<dt>reported_trait</dt>
<dd><p>A character vector of phenotypic traits as
reported by the original authors of the study.</p></dd>


<dt>set_operation</dt>
<dd><p>Either <code>'union'</code> or <code>'intersection'</code>. This
tells how variants retrieved by different criteria  should be combined:
<code>'union'</code> binds together all results removing duplicates and
<code>'intersection'</code> only keeps same variants found with different
criteria.</p></dd>


<dt>interactive</dt>
<dd><p>A logical. If all variants are requested, whether to ask
interactively if we really want to proceed.</p></dd>


<dt>std_chromosomes_only</dt>
<dd><p>Whether to return only variants mapped to
standard chromosomes: 1 thru 22, X, Y, and MT.</p></dd>


<dt>verbose</dt>
<dd><p>Whether the function should be
verbose about the different queries or not.</p></dd>


<dt>warnings</dt>
<dd><p>Whether to print warnings.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>A <a href="variants-class.html">variants</a> object.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>Please note that all search criteria are vectorised, thus allowing for batch
mode search, e.g., one can search by multiple variant identifiers at once by
passing a vector of identifiers to <code>variant_id</code>.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># Get variants by study identifier</span></span></span>
<span class="r-in"><span><span class="fu">get_variants</span><span class="op">(</span>study_id <span class="op">=</span> <span class="st">'GCST001085'</span>, warnings <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> An object of class "variants"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Slot "variants":</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 18 × 7</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    variant_id merged functional_cl…¹ chrom…² chrom…³ chrom…⁴ last_update_date   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dttm&gt;</span>             </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 1</span> rs10496288      0 intergenic_var… <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>   <span style="color: #949494;"> </span>  <span style="color: #BB0000;">NA</span>      2021-06-24 <span style="color: #949494;">19:41:17</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 2</span> rs7827545       0 intron_variant  <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>   <span style="color: #949494;"> </span>  <span style="color: #BB0000;">NA</span>      2021-06-24 <span style="color: #949494;">20:19:57</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 3</span> rs10785581      0 intron_variant  12       4.55<span style="color: #949494;">e</span>7 12q12   2021-06-24 <span style="color: #949494;">19:41:19</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 4</span> rs200759        0 intron_variant  20       1.56<span style="color: #949494;">e</span>7 20p12.1 2021-06-24 <span style="color: #949494;">19:41:19</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 5</span> rs7960483       0 intron_variant  <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>   <span style="color: #949494;"> </span>  <span style="color: #BB0000;">NA</span>      2021-06-24 <span style="color: #949494;">19:41:17</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 6</span> rs1372662       0 intron_variant  8        1.35<span style="color: #949494;">e</span>8 8q24.22 2021-06-24 <span style="color: #949494;">20:19:58</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 7</span> rs10496289      0 intergenic_var… 2        8.31<span style="color: #949494;">e</span>7 2p12    2021-06-24 <span style="color: #949494;">19:41:17</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 8</span> rs6452524       0 intron_variant  <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>   <span style="color: #949494;"> </span>  <span style="color: #BB0000;">NA</span>      2021-06-24 <span style="color: #949494;">19:41:17</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 9</span> rs200752        0 intron_variant  <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>   <span style="color: #949494;"> </span>  <span style="color: #BB0000;">NA</span>      2021-06-24 <span style="color: #949494;">19:41:17</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">10</span> rs3798440       0 intron_variant  <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>   <span style="color: #949494;"> </span>  <span style="color: #BB0000;">NA</span>      2021-06-24 <span style="color: #949494;">19:46:29</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">11</span> rs7735940       0 regulatory_reg… <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>   <span style="color: #949494;"> </span>  <span style="color: #BB0000;">NA</span>      2021-06-24 <span style="color: #949494;">19:41:11</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">12</span> rs9350602       0 intron_variant  6        7.59<span style="color: #949494;">e</span>7 6q14.1  2021-06-24 <span style="color: #949494;">19:46:30</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">13</span> rs10188442      0 intron_variant  2        1.32<span style="color: #949494;">e</span>8 2q21.2  2021-06-24 <span style="color: #949494;">20:19:59</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">14</span> rs12522034      0 intergenic_var… 5        3.64<span style="color: #949494;">e</span>7 5p13.2  2021-06-24 <span style="color: #949494;">19:41:12</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">15</span> rs13420028      0 intron_variant  <span style="color: #BB0000;">NA</span>      <span style="color: #BB0000;">NA</span>   <span style="color: #949494;"> </span>  <span style="color: #BB0000;">NA</span>      2021-06-24 <span style="color: #949494;">20:19:57</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">16</span> rs2469997       0 intergenic_var… 8        1.19<span style="color: #949494;">e</span>8 8q24.12 2021-06-25 <span style="color: #949494;">04:40:13</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">17</span> rs6887846       0 intron_variant  5        8.31<span style="color: #949494;">e</span>7 5q14.2  2021-06-24 <span style="color: #949494;">19:41:19</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">18</span> rs6469823       1 intergenic_var… 8        1.19<span style="color: #949494;">e</span>8 8q24.12 2021-06-24 <span style="color: #949494;">19:41:14</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># … with abbreviated variable names ¹​functional_class, ²​chromosome_name,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   ³​chromosome_position, ⁴​chromosome_region</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Slot "genomic_contexts":</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 60 × 12</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    variant_id gene_name  chrom…¹ chrom…² dista…³ is_ma…⁴ is_cl…⁵ is_in…⁶ is_up…⁷</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 1</span> rs10785581 LOC105369… 12       4.55<span style="color: #949494;">e</span>7   <span style="text-decoration: underline;">58</span>636 FALSE   TRUE    TRUE    FALSE  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 2</span> rs10785581 ANO6       12       4.55<span style="color: #949494;">e</span>7   <span style="text-decoration: underline;">96</span>786 FALSE   FALSE   TRUE    TRUE   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 3</span> rs10785581 ANO6       12       4.55<span style="color: #949494;">e</span>7   <span style="text-decoration: underline;">54</span>910 TRUE    TRUE    TRUE    TRUE   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 4</span> rs10785581 U6         12       4.55<span style="color: #949494;">e</span>7   <span style="text-decoration: underline;">17</span>400 TRUE    TRUE    TRUE    FALSE  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 5</span> rs10785581 LOC100128… 12       4.55<span style="color: #949494;">e</span>7   <span style="text-decoration: underline;">91</span>755 FALSE   TRUE    TRUE    TRUE   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 6</span> rs10785581 LOC105369… 12       4.55<span style="color: #949494;">e</span>7       0 FALSE   FALSE   FALSE   FALSE  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 7</span> rs200759   MACROD2    20       1.56<span style="color: #949494;">e</span>7       0 FALSE   FALSE   FALSE   FALSE  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 8</span> rs200759   LOC107985… 20       1.56<span style="color: #949494;">e</span>7    <span style="text-decoration: underline;">1</span>048 FALSE   TRUE    TRUE    FALSE  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 9</span> rs200759   ENSAP1     20       1.56<span style="color: #949494;">e</span>7    <span style="text-decoration: underline;">5</span>991 FALSE   TRUE    TRUE    TRUE   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">10</span> rs200759   ENSAP1     20       1.56<span style="color: #949494;">e</span>7    <span style="text-decoration: underline;">5</span>957 FALSE   TRUE    TRUE    TRUE   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># … with 50 more rows, 3 more variables: is_downstream &lt;lgl&gt;, source &lt;chr&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   mapping_method &lt;chr&gt;, and abbreviated variable names ¹​chromosome_name,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   ²​chromosome_position, ³​distance, ⁴​is_mapped_gene, ⁵​is_closest_gene,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   ⁶​is_intergenic, ⁷​is_upstream</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ Use `print(n = ...)` to see more rows, and `colnames()` to see all variable names</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Slot "ensembl_ids":</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 28 × 3</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    variant_id gene_name ensembl_id     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 1</span> rs10785581 ANO6      ENSG00000177119</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 2</span> rs10785581 U6        ENSG00000277091</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 3</span> rs200759   MACROD2   ENSG00000172264</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 4</span> rs200759   ENSAP1    ENSG00000224274</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 5</span> rs1372662  ZFAT-AS1  ENSG00000248492</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 6</span> rs1372662  ZFAT      ENSG00000066827</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 7</span> rs9350602  MYO6      ENSG00000196586</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 8</span> rs9350602  RNU6-155P ENSG00000252156</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 9</span> rs9350602  Y_RNA     ENSG00000252254</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">10</span> rs9350602  RNA5SP209 ENSG00000223169</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># … with 18 more rows</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ Use `print(n = ...)` to see more rows</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Slot "entrez_ids":</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 32 × 3</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    variant_id gene_name    entrez_id</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 1</span> rs10785581 LOC105369744 105369744</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 2</span> rs10785581 ANO6         196527   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 3</span> rs10785581 LOC100128248 100128248</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 4</span> rs10785581 LOC105369743 105369743</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 5</span> rs200759   MACROD2      140733   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 6</span> rs200759   LOC107985394 107985394</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 7</span> rs200759   ENSAP1       170511   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 8</span> rs1372662  ZFAT-AS1     594840   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 9</span> rs1372662  ZFAT         57623    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">10</span> rs9350602  MYO6         4646     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># … with 22 more rows</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ Use `print(n = ...)` to see more rows</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Get a variant by its identifier</span></span></span>
<span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span><span class="fu">get_variants</span><span class="op">(</span>variant_id <span class="op">=</span> <span class="st">'rs3798440'</span>, warnings <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by <a href="https://rmagno.eu/" class="external-link">Ramiro Magno</a>, Ana-Teresa Maia, <a href="https://cintesis.eu/" class="external-link"><img src="https://rmagno.eu/gwasrapidd/reference/figures/cintesis.svg" height="24"></a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer></div>

  

  

  </body></html>

