<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Get GWAS Catalog studies — get_studies • gwasrapidd</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



  <link href="../extra.css" rel="stylesheet">
  

<meta property="og:title" content="Get GWAS Catalog studies — get_studies" />
<meta property="og:description" content="Retrieves studies via the NHGRI-EBI GWAS Catalog REST API. The REST
API is queried multiple times with the criteria passed as arguments (see
below). By default all studies that match the criteria supplied in the
arguments are retrieved: this corresponds to the default option
set_operation set to 'union'. If you rather have only the
studies that match simultaneously all criteria provided, then set
set_operation to 'intersection'." />
<meta property="og:image" content="https://rmagno.eu/gwasrapidd/logo.svg" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gwasrapidd</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.8</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/gwasrapidd.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/faq.html">Frequently asked questions</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/ramiromagno/gwasrapidd/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Get GWAS Catalog studies</h1>
    <small class="dont-index">Source: <a href='https://github.com/ramiromagno/gwasrapidd/blob/master/R/get_studies.R'><code>R/get_studies.R</code></a></small>
    <div class="hidden name"><code>get_studies.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Retrieves studies via the NHGRI-EBI GWAS Catalog REST API. The REST
API is queried multiple times with the criteria passed as arguments (see
below). By default all studies that match the criteria supplied in the
arguments are retrieved: this corresponds to the default option
<code>set_operation</code> set to <code>'union'</code>. If you rather have only the
studies that match simultaneously all criteria provided, then set
<code>set_operation</code> to <code>'intersection'</code>.</p>
    </div>

    <pre class="usage"><span class='fu'>get_studies</span>(
  <span class='kw'>study_id</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>association_id</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>variant_id</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>efo_id</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>pubmed_id</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>user_requested</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>full_pvalue_set</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>efo_uri</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>efo_trait</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>reported_trait</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>set_operation</span> <span class='kw'>=</span> <span class='st'>"union"</span>,
  <span class='kw'>interactive</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>verbose</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>warnings</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>
)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>study_id</th>
      <td><p>A character vector of GWAS Catalog study accession
identifiers.</p></td>
    </tr>
    <tr>
      <th>association_id</th>
      <td><p>A character vector of GWAS Catalog association
identifiers.</p></td>
    </tr>
    <tr>
      <th>variant_id</th>
      <td><p>A character vector of GWAS Catalog variant identifiers.</p></td>
    </tr>
    <tr>
      <th>efo_id</th>
      <td><p>A character vector of <a href='https://www.ebi.ac.uk/efo/'>EFO</a>
identifiers.</p></td>
    </tr>
    <tr>
      <th>pubmed_id</th>
      <td><p>An integer vector of
<a href='https://en.wikipedia.org/wiki/PubMed'>PubMed</a> identifiers.</p></td>
    </tr>
    <tr>
      <th>user_requested</th>
      <td><p>A <code>logical</code> (scalar!) indicating to retrieve either studies
requested by users of the Catalog (<code>TRUE</code>) or otherwise
(<code>FALSE</code>).</p></td>
    </tr>
    <tr>
      <th>full_pvalue_set</th>
      <td><p>A <code>logical</code> (scalar!) indicating to retrieve studies with full
summary statistics (<code>TRUE</code>) or studies without it (<code>FALSE</code>).</p></td>
    </tr>
    <tr>
      <th>efo_uri</th>
      <td><p>A character vector of <a href='https://www.ebi.ac.uk/efo/'>EFO</a> URIs.</p></td>
    </tr>
    <tr>
      <th>efo_trait</th>
      <td><p>A character vector of
<a href='https://www.ebi.ac.uk/efo/'>EFO</a> trait descriptions, e.g.,
<code>'uric acid measurement'</code>.</p></td>
    </tr>
    <tr>
      <th>reported_trait</th>
      <td><p>A character vector of phenotypic traits as
reported by the original authors of the study.</p></td>
    </tr>
    <tr>
      <th>set_operation</th>
      <td><p>Either <code>'union'</code> or <code>'intersection'</code>. This
tells how studies retrieved by different criteria  should be combined:
<code>'union'</code> binds together all results removing duplicates and
<code>'intersection'</code> only keeps same studies found with different
criteria.</p></td>
    </tr>
    <tr>
      <th>interactive</th>
      <td><p>A logical. If all studies are requested, whether to ask
interactively if we really want to proceed.</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>Whether the function should be verbose about the different
queries or not.</p></td>
    </tr>
    <tr>
      <th>warnings</th>
      <td><p>Whether to print warnings.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A <a href='studies-class.html'>studies</a> object.</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>Please note that all search criteria are vectorised, thus allowing for batch
mode search, e.g., one can search by multiple variant identifiers at once by
passing a vector of identifiers to <code>variant_id</code>.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'># Get a study by its accession identifier</span>
<span class='fu'>get_studies</span>(<span class='kw'>study_id</span> <span class='kw'>=</span> <span class='st'>'GCST001085'</span>)</div><div class='output co'>#&gt; An object of class "studies"
#&gt; Slot "studies":
#&gt; # A tibble: 1 x 13
#&gt;   study_id reported_trait initial_sample_… replication_sam… gxe   gxg  
#&gt;   &lt;chr&gt;    &lt;chr&gt;          &lt;chr&gt;            &lt;chr&gt;            &lt;lgl&gt; &lt;lgl&gt;
#&gt; 1 GCST001… Hypertension … ~2,000 European… &lt;NA&gt;             FALSE TRUE 
#&gt; # … with 7 more variables: snp_count &lt;int&gt;, qualifier &lt;chr&gt;, imputed &lt;lgl&gt;,
#&gt; #   pooled &lt;lgl&gt;, study_design_comment &lt;chr&gt;, full_pvalue_set &lt;lgl&gt;,
#&gt; #   user_requested &lt;lgl&gt;
#&gt; 
#&gt; Slot "genotyping_techs":
#&gt; # A tibble: 1 x 2
#&gt;   study_id   genotyping_technology       
#&gt;   &lt;chr&gt;      &lt;chr&gt;                       
#&gt; 1 GCST001085 Genome-wide genotyping array
#&gt; 
#&gt; Slot "platforms":
#&gt; # A tibble: 1 x 2
#&gt;   study_id   manufacturer
#&gt;   &lt;chr&gt;      &lt;chr&gt;       
#&gt; 1 GCST001085 Affymetrix  
#&gt; 
#&gt; Slot "ancestries":
#&gt; # A tibble: 1 x 4
#&gt;   study_id   ancestry_id type    number_of_individuals
#&gt;   &lt;chr&gt;            &lt;int&gt; &lt;chr&gt;                   &lt;int&gt;
#&gt; 1 GCST001085           1 initial                  5000
#&gt; 
#&gt; Slot "ancestral_groups":
#&gt; # A tibble: 1 x 3
#&gt;   study_id   ancestry_id ancestral_group
#&gt;   &lt;chr&gt;            &lt;int&gt; &lt;chr&gt;          
#&gt; 1 GCST001085           1 European       
#&gt; 
#&gt; Slot "countries_of_origin":
#&gt; # A tibble: 1 x 5
#&gt;   study_id   ancestry_id country_name major_area region
#&gt;   &lt;chr&gt;            &lt;int&gt; &lt;chr&gt;        &lt;chr&gt;      &lt;chr&gt; 
#&gt; 1 GCST001085           1 &lt;NA&gt;         &lt;NA&gt;       &lt;NA&gt;  
#&gt; 
#&gt; Slot "countries_of_recruitment":
#&gt; # A tibble: 1 x 5
#&gt;   study_id   ancestry_id country_name major_area region
#&gt;   &lt;chr&gt;            &lt;int&gt; &lt;chr&gt;        &lt;chr&gt;      &lt;chr&gt; 
#&gt; 1 GCST001085           1 &lt;NA&gt;         &lt;NA&gt;       &lt;NA&gt;  
#&gt; 
#&gt; Slot "publications":
#&gt; # A tibble: 1 x 7
#&gt;   study_id pubmed_id publication_date publication title author_fullname
#&gt;   &lt;chr&gt;        &lt;int&gt; &lt;date&gt;           &lt;chr&gt;       &lt;chr&gt; &lt;chr&gt;          
#&gt; 1 GCST001…  21626137 2011-05-27       Hum Genet   Two-… Slavin TP      
#&gt; # … with 1 more variable: author_orcid &lt;chr&gt;
#&gt; </div><div class='input'>
<span class='co'># Get a study by association identifier</span>
<span class='fu'>get_studies</span>(<span class='kw'>association_id</span> <span class='kw'>=</span> <span class='st'>'25389945'</span>)</div><div class='output co'>#&gt; An object of class "studies"
#&gt; Slot "studies":
#&gt; # A tibble: 1 x 13
#&gt;   study_id reported_trait initial_sample_… replication_sam… gxe   gxg  
#&gt;   &lt;chr&gt;    &lt;chr&gt;          &lt;chr&gt;            &lt;chr&gt;            &lt;lgl&gt; &lt;lgl&gt;
#&gt; 1 GCST005… Major depress… 2,605 European … 8,508 European … FALSE FALSE
#&gt; # … with 7 more variables: snp_count &lt;int&gt;, qualifier &lt;chr&gt;, imputed &lt;lgl&gt;,
#&gt; #   pooled &lt;lgl&gt;, study_design_comment &lt;chr&gt;, full_pvalue_set &lt;lgl&gt;,
#&gt; #   user_requested &lt;lgl&gt;
#&gt; 
#&gt; Slot "genotyping_techs":
#&gt; # A tibble: 1 x 2
#&gt;   study_id   genotyping_technology       
#&gt;   &lt;chr&gt;      &lt;chr&gt;                       
#&gt; 1 GCST005108 Genome-wide genotyping array
#&gt; 
#&gt; Slot "platforms":
#&gt; # A tibble: 1 x 2
#&gt;   study_id   manufacturer
#&gt;   &lt;chr&gt;      &lt;chr&gt;       
#&gt; 1 GCST005108 Illumina    
#&gt; 
#&gt; Slot "ancestries":
#&gt; # A tibble: 2 x 4
#&gt;   study_id   ancestry_id type        number_of_individuals
#&gt;   &lt;chr&gt;            &lt;int&gt; &lt;chr&gt;                       &lt;int&gt;
#&gt; 1 GCST005108           1 replication                 25035
#&gt; 2 GCST005108           2 initial                     18773
#&gt; 
#&gt; Slot "ancestral_groups":
#&gt; # A tibble: 2 x 3
#&gt;   study_id   ancestry_id ancestral_group
#&gt;   &lt;chr&gt;            &lt;int&gt; &lt;chr&gt;          
#&gt; 1 GCST005108           1 European       
#&gt; 2 GCST005108           2 European       
#&gt; 
#&gt; Slot "countries_of_origin":
#&gt; # A tibble: 2 x 5
#&gt;   study_id   ancestry_id country_name major_area region
#&gt;   &lt;chr&gt;            &lt;int&gt; &lt;chr&gt;        &lt;chr&gt;      &lt;chr&gt; 
#&gt; 1 GCST005108           1 &lt;NA&gt;         &lt;NA&gt;       &lt;NA&gt;  
#&gt; 2 GCST005108           2 &lt;NA&gt;         &lt;NA&gt;       &lt;NA&gt;  
#&gt; 
#&gt; Slot "countries_of_recruitment":
#&gt; # A tibble: 2 x 5
#&gt;   study_id   ancestry_id country_name major_area region         
#&gt;   &lt;chr&gt;            &lt;int&gt; &lt;chr&gt;        &lt;chr&gt;      &lt;chr&gt;          
#&gt; 1 GCST005108           1 U.K.         Europe     Northern Europe
#&gt; 2 GCST005108           2 U.K.         Europe     Northern Europe
#&gt; 
#&gt; Slot "publications":
#&gt; # A tibble: 1 x 7
#&gt;   study_id pubmed_id publication_date publication title author_fullname
#&gt;   &lt;chr&gt;        &lt;int&gt; &lt;date&gt;           &lt;chr&gt;       &lt;chr&gt; &lt;chr&gt;          
#&gt; 1 GCST005…  29187746 2017-11-30       Transl Psy… Geno… Howard DM      
#&gt; # … with 1 more variable: author_orcid &lt;chr&gt;
#&gt; </div><div class='input'>
<span class='co'># Get studies by variant identifier</span>
<span class='fu'>get_studies</span>(<span class='kw'>variant_id</span> <span class='kw'>=</span> <span class='st'>'rs3798440'</span>)</div><div class='output co'>#&gt; An object of class "studies"
#&gt; Slot "studies":
#&gt; # A tibble: 1 x 13
#&gt;   study_id reported_trait initial_sample_… replication_sam… gxe   gxg  
#&gt;   &lt;chr&gt;    &lt;chr&gt;          &lt;chr&gt;            &lt;chr&gt;            &lt;lgl&gt; &lt;lgl&gt;
#&gt; 1 GCST001… Hypertension … ~2,000 European… &lt;NA&gt;             FALSE TRUE 
#&gt; # … with 7 more variables: snp_count &lt;int&gt;, qualifier &lt;chr&gt;, imputed &lt;lgl&gt;,
#&gt; #   pooled &lt;lgl&gt;, study_design_comment &lt;chr&gt;, full_pvalue_set &lt;lgl&gt;,
#&gt; #   user_requested &lt;lgl&gt;
#&gt; 
#&gt; Slot "genotyping_techs":
#&gt; # A tibble: 1 x 2
#&gt;   study_id   genotyping_technology       
#&gt;   &lt;chr&gt;      &lt;chr&gt;                       
#&gt; 1 GCST001085 Genome-wide genotyping array
#&gt; 
#&gt; Slot "platforms":
#&gt; # A tibble: 1 x 2
#&gt;   study_id   manufacturer
#&gt;   &lt;chr&gt;      &lt;chr&gt;       
#&gt; 1 GCST001085 Affymetrix  
#&gt; 
#&gt; Slot "ancestries":
#&gt; # A tibble: 1 x 4
#&gt;   study_id   ancestry_id type    number_of_individuals
#&gt;   &lt;chr&gt;            &lt;int&gt; &lt;chr&gt;                   &lt;int&gt;
#&gt; 1 GCST001085           1 initial                  5000
#&gt; 
#&gt; Slot "ancestral_groups":
#&gt; # A tibble: 1 x 3
#&gt;   study_id   ancestry_id ancestral_group
#&gt;   &lt;chr&gt;            &lt;int&gt; &lt;chr&gt;          
#&gt; 1 GCST001085           1 European       
#&gt; 
#&gt; Slot "countries_of_origin":
#&gt; # A tibble: 1 x 5
#&gt;   study_id   ancestry_id country_name major_area region
#&gt;   &lt;chr&gt;            &lt;int&gt; &lt;chr&gt;        &lt;chr&gt;      &lt;chr&gt; 
#&gt; 1 GCST001085           1 &lt;NA&gt;         &lt;NA&gt;       &lt;NA&gt;  
#&gt; 
#&gt; Slot "countries_of_recruitment":
#&gt; # A tibble: 1 x 5
#&gt;   study_id   ancestry_id country_name major_area region
#&gt;   &lt;chr&gt;            &lt;int&gt; &lt;chr&gt;        &lt;chr&gt;      &lt;chr&gt; 
#&gt; 1 GCST001085           1 &lt;NA&gt;         &lt;NA&gt;       &lt;NA&gt;  
#&gt; 
#&gt; Slot "publications":
#&gt; # A tibble: 1 x 7
#&gt;   study_id pubmed_id publication_date publication title author_fullname
#&gt;   &lt;chr&gt;        &lt;int&gt; &lt;date&gt;           &lt;chr&gt;       &lt;chr&gt; &lt;chr&gt;          
#&gt; 1 GCST001…  21626137 2011-05-27       Hum Genet   Two-… Slavin TP      
#&gt; # … with 1 more variable: author_orcid &lt;chr&gt;
#&gt; </div><div class='input'>
<span class='co'># Get studies by EFO trait identifier</span>
<span class='fu'>get_studies</span>(<span class='kw'>efo_id</span> <span class='kw'>=</span> <span class='st'>'EFO_0005537'</span>)</div><div class='output co'>#&gt; An object of class "studies"
#&gt; Slot "studies":
#&gt; # A tibble: 1 x 13
#&gt;   study_id reported_trait initial_sample_… replication_sam… gxe   gxg  
#&gt;   &lt;chr&gt;    &lt;chr&gt;          &lt;chr&gt;            &lt;chr&gt;            &lt;lgl&gt; &lt;lgl&gt;
#&gt; 1 GCST002… Breast cancer… 1,529 European … 2,148 European … FALSE FALSE
#&gt; # … with 7 more variables: snp_count &lt;int&gt;, qualifier &lt;chr&gt;, imputed &lt;lgl&gt;,
#&gt; #   pooled &lt;lgl&gt;, study_design_comment &lt;chr&gt;, full_pvalue_set &lt;lgl&gt;,
#&gt; #   user_requested &lt;lgl&gt;
#&gt; 
#&gt; Slot "genotyping_techs":
#&gt; # A tibble: 1 x 2
#&gt;   study_id   genotyping_technology       
#&gt;   &lt;chr&gt;      &lt;chr&gt;                       
#&gt; 1 GCST002305 Genome-wide genotyping array
#&gt; 
#&gt; Slot "platforms":
#&gt; # A tibble: 1 x 2
#&gt;   study_id   manufacturer
#&gt;   &lt;chr&gt;      &lt;chr&gt;       
#&gt; 1 GCST002305 Illumina    
#&gt; 
#&gt; Slot "ancestries":
#&gt; # A tibble: 2 x 4
#&gt;   study_id   ancestry_id type        number_of_individuals
#&gt;   &lt;chr&gt;            &lt;int&gt; &lt;chr&gt;                       &lt;int&gt;
#&gt; 1 GCST002305           1 replication                  3457
#&gt; 2 GCST002305           2 initial                      4928
#&gt; 
#&gt; Slot "ancestral_groups":
#&gt; # A tibble: 2 x 3
#&gt;   study_id   ancestry_id ancestral_group
#&gt;   &lt;chr&gt;            &lt;int&gt; &lt;chr&gt;          
#&gt; 1 GCST002305           1 European       
#&gt; 2 GCST002305           2 European       
#&gt; 
#&gt; Slot "countries_of_origin":
#&gt; # A tibble: 0 x 5
#&gt; # … with 5 variables: study_id &lt;chr&gt;, ancestry_id &lt;int&gt;, country_name &lt;chr&gt;,
#&gt; #   major_area &lt;chr&gt;, region &lt;chr&gt;
#&gt; 
#&gt; Slot "countries_of_recruitment":
#&gt; # A tibble: 10 x 5
#&gt;    study_id   ancestry_id country_name major_area       region               
#&gt;    &lt;chr&gt;            &lt;int&gt; &lt;chr&gt;        &lt;chr&gt;            &lt;chr&gt;                
#&gt;  1 GCST002305           1 U.S.         Northern America &lt;NA&gt;                 
#&gt;  2 GCST002305           1 Greece       Europe           Southern Europe      
#&gt;  3 GCST002305           1 Germany      Europe           Western Europe       
#&gt;  4 GCST002305           1 U.K.         Europe           Northern Europe      
#&gt;  5 GCST002305           1 Norway       Europe           Northern Europe      
#&gt;  6 GCST002305           2 Finland      Europe           Northern Europe      
#&gt;  7 GCST002305           2 U.S.         Northern America &lt;NA&gt;                 
#&gt;  8 GCST002305           2 Australia    Oceania          Australia/New Zealand
#&gt;  9 GCST002305           2 Germany      Europe           Western Europe       
#&gt; 10 GCST002305           2 U.K.         Europe           Northern Europe      
#&gt; 
#&gt; Slot "publications":
#&gt; # A tibble: 1 x 7
#&gt;   study_id pubmed_id publication_date publication title author_fullname
#&gt;   &lt;chr&gt;        &lt;int&gt; &lt;date&gt;           &lt;chr&gt;       &lt;chr&gt; &lt;chr&gt;          
#&gt; 1 GCST002…  24325915 2013-12-09       Carcinogen… Geno… Purrington KS  
#&gt; # … with 1 more variable: author_orcid &lt;chr&gt;
#&gt; </div><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Ramiro Magno, Ana-Teresa Maia, Centre for Biomedical Research, University of Algarve.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


