<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en-US"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Get GWAS Catalog variants — get_variants • gwasrapidd</title><!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png"><link rel="icon" type="”image/svg+xml”" href="../favicon.svg"><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png"><link rel="icon" sizes="any" href="../favicon.ico"><link rel="manifest" href="../site.webmanifest"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/Nunito-0.4.10/font.css" rel="stylesheet"><link href="../deps/Inconsolata-0.4.10/font.css" rel="stylesheet"><link href="../deps/Source_Sans_Pro-0.4.10/font.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Get GWAS Catalog variants — get_variants"><meta name="description" content="Retrieves variants via the NHGRI-EBI GWAS Catalog REST API. The REST
API is queried multiple times with the criteria passed as arguments (see
below). By default all variants that match the criteria supplied in the
arguments are retrieved: this corresponds to the default option
set_operation set to 'union'. If you rather have only the
variants that match simultaneously all criteria provided, then set
set_operation to 'intersection'."><meta property="og:description" content="Retrieves variants via the NHGRI-EBI GWAS Catalog REST API. The REST
API is queried multiple times with the criteria passed as arguments (see
below). By default all variants that match the criteria supplied in the
arguments are retrieved: this corresponds to the default option
set_operation set to 'union'. If you rather have only the
variants that match simultaneously all criteria provided, then set
set_operation to 'intersection'."><meta property="og:image" content="https://rmagno.eu/gwasrapidd/logo.svg"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-none" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">gwasrapidd</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.18</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/gwasrapidd.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/bmi_variants.html">Variants associated with Body Mass Index (BMI)</a></li>
    <li><a class="dropdown-item" href="../articles/faq.html">Frequently asked questions</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/ramiromagno/gwasrapidd/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>Get GWAS Catalog variants</h1>
      <small class="dont-index">Source: <a href="https://github.com/ramiromagno/gwasrapidd/blob/master/R/get_variants.R" class="external-link"><code>R/get_variants.R</code></a></small>
      <div class="d-none name"><code>get_variants.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Retrieves variants via the NHGRI-EBI GWAS Catalog REST API. The REST
API is queried multiple times with the criteria passed as arguments (see
below). By default all variants that match the criteria supplied in the
arguments are retrieved: this corresponds to the default option
<code>set_operation</code> set to <code>'union'</code>. If you rather have only the
variants that match simultaneously all criteria provided, then set
<code>set_operation</code> to <code>'intersection'</code>.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">get_variants</span><span class="op">(</span></span>
<span>  study_id <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  association_id <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  variant_id <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  efo_id <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  pubmed_id <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  genomic_range <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  cytogenetic_band <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  gene_name <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  efo_trait <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  reported_trait <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  set_operation <span class="op">=</span> <span class="st">"union"</span>,</span>
<span>  interactive <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  std_chromosomes_only <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  warnings <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-study-id">study_id<a class="anchor" aria-label="anchor" href="#arg-study-id"></a></dt>
<dd><p>A character vector of GWAS Catalog study accession
identifiers.</p></dd>


<dt id="arg-association-id">association_id<a class="anchor" aria-label="anchor" href="#arg-association-id"></a></dt>
<dd><p>A character vector of GWAS Catalog association
identifiers.</p></dd>


<dt id="arg-variant-id">variant_id<a class="anchor" aria-label="anchor" href="#arg-variant-id"></a></dt>
<dd><p>A character vector of GWAS Catalog variant identifiers.</p></dd>


<dt id="arg-efo-id">efo_id<a class="anchor" aria-label="anchor" href="#arg-efo-id"></a></dt>
<dd><p>A character vector of <a href="https://www.ebi.ac.uk/efo/" class="external-link">EFO</a>
identifiers.</p></dd>


<dt id="arg-pubmed-id">pubmed_id<a class="anchor" aria-label="anchor" href="#arg-pubmed-id"></a></dt>
<dd><p>An integer vector of
<a href="https://pubmed.ncbi.nlm.nih.gov/" class="external-link">PubMed</a> identifiers.</p></dd>


<dt id="arg-genomic-range">genomic_range<a class="anchor" aria-label="anchor" href="#arg-genomic-range"></a></dt>
<dd><p>A named list of three vectors:</p><dl><dt>chromosome</dt>
<dd><p>A character vector of chromosome names of the
form 1–22, X or Y.</p></dd>
 <dt>start</dt>
<dd><p>A numeric vector of start positions,
starting at 1.</p></dd>
 <dt>end</dt>
<dd><p>A numeric vector of end positions.</p></dd>

</dl><p>The three vectors need to be of the same length so that <code>chromosome</code>
names, <code>start</code> and <code>end</code> positions can be matched by position.</p></dd>


<dt id="arg-cytogenetic-band">cytogenetic_band<a class="anchor" aria-label="anchor" href="#arg-cytogenetic-band"></a></dt>
<dd><p>A character vector of cytogenetic bands of the form
<code>'1p36.11'</code>.</p></dd>


<dt id="arg-gene-name">gene_name<a class="anchor" aria-label="anchor" href="#arg-gene-name"></a></dt>
<dd><p>Gene symbol according to
<a href="https://www.genenames.org/" class="external-link">HUGO Gene Nomenclature (HGNC)</a>.</p></dd>


<dt id="arg-efo-trait">efo_trait<a class="anchor" aria-label="anchor" href="#arg-efo-trait"></a></dt>
<dd><p>A character vector of
<a href="https://www.ebi.ac.uk/efo/" class="external-link">EFO</a> trait descriptions, e.g.,
<code>'uric acid measurement'</code>.</p></dd>


<dt id="arg-reported-trait">reported_trait<a class="anchor" aria-label="anchor" href="#arg-reported-trait"></a></dt>
<dd><p>A character vector of phenotypic traits as
reported by the original authors of the study.</p></dd>


<dt id="arg-set-operation">set_operation<a class="anchor" aria-label="anchor" href="#arg-set-operation"></a></dt>
<dd><p>Either <code>'union'</code> or <code>'intersection'</code>. This
tells how variants retrieved by different criteria  should be combined:
<code>'union'</code> binds together all results removing duplicates and
<code>'intersection'</code> only keeps same variants found with different
criteria.</p></dd>


<dt id="arg-interactive">interactive<a class="anchor" aria-label="anchor" href="#arg-interactive"></a></dt>
<dd><p>A logical. If all variants are requested, whether to ask
interactively if we really want to proceed.</p></dd>


<dt id="arg-std-chromosomes-only">std_chromosomes_only<a class="anchor" aria-label="anchor" href="#arg-std-chromosomes-only"></a></dt>
<dd><p>Whether to return only variants mapped to
standard chromosomes: 1 thru 22, X, Y, and MT.</p></dd>


<dt id="arg-verbose">verbose<a class="anchor" aria-label="anchor" href="#arg-verbose"></a></dt>
<dd><p>Whether the function should be
verbose about the different queries or not.</p></dd>


<dt id="arg-warnings">warnings<a class="anchor" aria-label="anchor" href="#arg-warnings"></a></dt>
<dd><p>Whether to print warnings.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A <a href="variants-class.html">variants</a> object.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>Please note that all search criteria are vectorised, thus allowing for batch
mode search, e.g., one can search by multiple variant identifiers at once by
passing a vector of identifiers to <code>variant_id</code>.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># Get variants by study identifier</span></span></span>
<span class="r-in"><span><span class="fu">get_variants</span><span class="op">(</span>study_id <span class="op">=</span> <span class="st">'GCST001085'</span>, warnings <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> An object of class "variants"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Slot "variants":</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 18 × 7</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    variant_id merged functional_class   chromosome_name chromosome_position</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                         <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 1</span> rs6469823       1 intergenic_variant 8                         119<span style="text-decoration: underline;">341</span>744</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 2</span> rs200752        0 intron_variant     20                         15<span style="text-decoration: underline;">618</span>886</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 3</span> rs1372662       0 intron_variant     8                         134<span style="text-decoration: underline;">554</span>803</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 4</span> rs2469997       0 intergenic_variant 8                         119<span style="text-decoration: underline;">341</span>027</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 5</span> rs12522034      0 intergenic_variant 5                          36<span style="text-decoration: underline;">425</span>491</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 6</span> rs7735940       0 intergenic_variant 5                          36<span style="text-decoration: underline;">423</span>829</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 7</span> rs7960483       0 intergenic_variant 12                         45<span style="text-decoration: underline;">531</span>972</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 8</span> rs6887846       0 intron_variant     5                          83<span style="text-decoration: underline;">140</span>542</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 9</span> rs3798440       0 intron_variant     6                          75<span style="text-decoration: underline;">846</span>902</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">10</span> rs6452524       0 intron_variant     5                          83<span style="text-decoration: underline;">137</span>962</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">11</span> rs7827545       0 intron_variant     8                         134<span style="text-decoration: underline;">554</span>324</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">12</span> rs10496288      0 intergenic_variant 2                          83<span style="text-decoration: underline;">065</span>441</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">13</span> rs13420028      0 intron_variant     2                         132<span style="text-decoration: underline;">430</span>533</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">14</span> rs10785581      0 intergenic_variant 12                         45<span style="text-decoration: underline;">537</span>190</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">15</span> rs10188442      0 intron_variant     2                         132<span style="text-decoration: underline;">431</span>666</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">16</span> rs10496289      0 intergenic_variant 2                          83<span style="text-decoration: underline;">066</span>256</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">17</span> rs9350602       0 intron_variant     6                          75<span style="text-decoration: underline;">850</span>781</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">18</span> rs200759        0 intron_variant     20                         15<span style="text-decoration: underline;">625</span>776</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 2 more variables: chromosome_region &lt;chr&gt;, last_update_date &lt;dttm&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Slot "genomic_contexts":</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 140 × 12</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    variant_id gene_name    chromosome_name chromosome_position distance</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                         <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 1</span> rs6469823  MAL2         8                         119<span style="text-decoration: underline;">341</span>744    <span style="text-decoration: underline;">96</span>071</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 2</span> rs6469823  CCN3         8                         119<span style="text-decoration: underline;">341</span>744    <span style="text-decoration: underline;">74</span>702</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 3</span> rs6469823  MAL2-AS1     8                         119<span style="text-decoration: underline;">341</span>744    <span style="text-decoration: underline;">94</span>887</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 4</span> rs6469823  MIR548AZ     8                         119<span style="text-decoration: underline;">341</span>744    <span style="text-decoration: underline;">16</span>479</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 5</span> rs6469823  LOC124900266 8                         119<span style="text-decoration: underline;">341</span>744    <span style="text-decoration: underline;">46</span>604</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 6</span> rs6469823  MAL2-AS1     8                         119<span style="text-decoration: underline;">341</span>744    <span style="text-decoration: underline;">94</span>887</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 7</span> rs6469823  MAL2         8                         119<span style="text-decoration: underline;">341</span>744    <span style="text-decoration: underline;">96</span>071</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 8</span> rs6469823  CCN3         8                         119<span style="text-decoration: underline;">341</span>744    <span style="text-decoration: underline;">74</span>702</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 9</span> rs6469823  MIR548AZ     8                         119<span style="text-decoration: underline;">341</span>744    <span style="text-decoration: underline;">16</span>479</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">10</span> rs6469823  LOC124902009 8                         119<span style="text-decoration: underline;">341</span>744     <span style="text-decoration: underline;">8</span>146</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 130 more rows</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 7 more variables: is_mapped_gene &lt;lgl&gt;, is_closest_gene &lt;lgl&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   is_intergenic &lt;lgl&gt;, is_upstream &lt;lgl&gt;, is_downstream &lt;lgl&gt;, source &lt;chr&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   mapping_method &lt;chr&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Slot "ensembl_ids":</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 62 × 3</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    variant_id gene_name ensembl_id     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 1</span> rs6469823  MAL2      ENSG00000147676</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 2</span> rs6469823  CCN3      ENSG00000136999</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 3</span> rs6469823  MAL2-AS1  ENSG00000253972</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 4</span> rs6469823  MIR548AZ  ENSG00000276479</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 5</span> rs200752   RNA5SP475 ENSG00000222500</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 6</span> rs200752   ENSAP1    ENSG00000224274</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 7</span> rs200752   MACROD2   ENSG00000172264</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 8</span> rs1372662  ZFAT      ENSG00000066827</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 9</span> rs1372662  ZFAT-AS1  ENSG00000248492</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">10</span> rs1372662  MTCO1P49  ENSG00000253916</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 52 more rows</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Slot "entrez_ids":</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 89 × 3</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    variant_id gene_name    entrez_id</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 1</span> rs6469823  MAL2         114569   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 2</span> rs6469823  CCN3         4856     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 3</span> rs6469823  MAL2-AS1     105375726</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 4</span> rs6469823  MIR548AZ     102466162</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 5</span> rs6469823  LOC124900266 124900266</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 6</span> rs6469823  LOC124902009 124902009</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 7</span> rs200752   RNA5SP475    100873717</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 8</span> rs200752   ENSAP1       170511   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 9</span> rs200752   MACROD2      140733   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">10</span> rs1372662  LOC124902070 124902070</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 79 more rows</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Get a variant by its identifier</span></span></span>
<span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span> <span class="co"># \dontrun{</span></span></span>
<span class="r-in"><span><span class="fu">get_variants</span><span class="op">(</span>variant_id <span class="op">=</span> <span class="st">'rs3798440'</span>, warnings <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span> <span class="co"># }</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://www.pattern.institute/team/rmagno/" class="external-link">Ramiro Magno</a>, Ana-Teresa Maia, CINTESIS, <a href="https://www.pattern.institute" class="external-link">Pattern Institute</a>.</p>
</div>

<div class="pkgdown-footer-right">

</div>

    </footer></div>





  </body></html>

