<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>An S4 class to represent a set of GWAS Catalog variants — variants-class • gwasrapidd</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



  <link href="../extra.css" rel="stylesheet">
  

<meta property="og:title" content="An S4 class to represent a set of GWAS Catalog variants — variants-class" />
<meta property="og:description" content="The variants object consists of four slots, each a table
(tibble), that combined form a relational database of a
subset of GWAS Catalog variants. Each variant is an observation (row) in the
variants table --- main table. All tables have the column
variant_id as primary key." />
<meta property="og:image" content="https://rmagno.eu/gwasrapidd/logo.svg" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gwasrapidd</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.12</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/gwasrapidd.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/bmi_variants.html">Variants associated with Body Mass Index (BMI)</a>
    </li>
    <li>
      <a href="../articles/faq.html">Frequently asked questions</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/ramiromagno/gwasrapidd/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>An S4 class to represent a set of GWAS Catalog variants</h1>
    <small class="dont-index">Source: <a href='https://github.com/ramiromagno/gwasrapidd/blob/master/R/class-variants.R'><code>R/class-variants.R</code></a></small>
    <div class="hidden name"><code>variants-class.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>The variants object consists of four slots, each a table
(<code><a href='https://tibble.tidyverse.org/reference/tibble.html'>tibble</a></code>), that combined form a relational database of a
subset of GWAS Catalog variants. Each variant is an observation (row) in the
<code>variants</code> table --- main table. All tables have the column
<code>variant_id</code> as primary key.</p>
    </div>



    <h2 class="hasAnchor" id="slots"><a class="anchor" href="#slots"></a>Slots</h2>

    

<dl>
<dt><code>variants</code></dt><dd><p>A <code><a href='https://tibble.tidyverse.org/reference/tibble.html'>tibble</a></code> listing variants. Columns:
<dl>
<dt>variant_id</dt><dd><p>Variant identifier, e.g., <code>'rs1333048'</code>.</p></dd>
<dt>merged</dt><dd><p>Whether this SNP has been merged with another SNP in a newer
genome build.</p></dd>
<dt>functional_class</dt><dd><p>Class according to Ensembl's predicted consequences
that each variant allele may have on transcripts. See
<a href='https://www.ensembl.org/info/genome/variation/prediction/predicted_data.html'>Ensembl
Variation - Calculated variant consequences</a>.</p></dd>
<dt>chromosome_name</dt><dd><p>Chromosome name.</p></dd>
<dt>chromosome_position</dt><dd><p>Chromosome position.</p></dd>
<dt>chromosome_region</dt><dd><p><a href='https://medlineplus.gov/genetics/understanding/howgeneswork/genelocation/'>Cytogenetic
location</a>.</p></dd>
<dt>last_update_date</dt><dd><p>Last time this variant was updated.</p></dd></dl></p></dd>

<dt><code>genomic_contexts</code></dt><dd><p>A <code><a href='https://tibble.tidyverse.org/reference/tibble.html'>tibble</a></code> listing genomic
  contexts associated with each variant. Columns:
<dl>
<dt>variant_id</dt><dd><p>Variant identifier.</p></dd>
<dt>gene_name</dt><dd><p>Gene symbol according to
<a href='https://www.genenames.org/'>HUGO Gene Nomenclature (HGNC)</a>.</p></dd>
<dt>chromosome_name</dt><dd><p>Chromosome name.</p></dd>
<dt>chromosome_position</dt><dd><p>Chromosome position.</p></dd>
<dt>distance</dt><dd><p>Genomic distance between the variant and the gene (in base
pairs).</p></dd>
<dt>is_mapped_gene</dt><dd><p>Whether this is a mapped gene to this variant. A mapped
gene is either an overlapping gene with the variant or the two closest genes
upstream and downstream of the variant. Moreover, only genes whose mapping
source is 'Ensembl' are considered.</p></dd>
<dt>is_closest_gene</dt><dd><p>Whether this is the closest gene to this variant.</p></dd>
<dt>is_intergenic</dt><dd><p>Whether this variant is intergenic, i.e, if there is no
gene up or downstream within 100kb.</p></dd>
<dt>is_upstream</dt><dd><p>Whether this variant is upstream of this gene.</p></dd>
<dt>is_downstream</dt><dd><p>Whether this variant is downstream of this gene.</p></dd>
<dt>source</dt><dd><p>Gene mapping source, either <code>Ensembl</code> or <code>NCBI</code>.</p></dd>
<dt>mapping_method</dt><dd><p>Gene mapping method.</p></dd></dl></p></dd>

<dt><code>ensembl_ids</code></dt><dd><p>A <code><a href='https://tibble.tidyverse.org/reference/tibble.html'>tibble</a></code> listing gene Ensembl
  identifiers associated with each genomic context. Columns:
<dl>
<dt>variant_id</dt><dd><p>Variant identifier.</p></dd>
<dt>gene_name</dt><dd><p>Gene symbol according to
<a href='https://www.genenames.org/'>HUGO Gene Nomenclature (HGNC)</a>.</p></dd>
<dt>ensembl_id</dt><dd><p>The Ensembl identifier of an Ensembl gene, see Section
<a href='https://www.ensembl.org/info/genome/genebuild/index.html'>Gene
annotation in Ensembl</a> for more information.</p></dd></dl></p></dd>

<dt><code>entrez_ids</code></dt><dd><p>A <code><a href='https://tibble.tidyverse.org/reference/tibble.html'>tibble</a></code> listing gene Entrez
  identifiers associated with each genomic context. Columns:
<dl>
<dt>variant_id</dt><dd><p>Variant identifier.</p></dd>
<dt>gene_name</dt><dd><p>Gene symbol according to
<a href='https://www.genenames.org/'>HUGO Gene Nomenclature (HGNC)</a>.</p></dd>
<dt>entrez_id</dt><dd><p>The Entrez identifier of a gene, see ref.
doi: <a href='https://doi.org/10.1093/nar/gkq1237'>10.1093/nar/gkq1237</a>
 for
more information.</p></dd></dl></p></dd>

</dl>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by <a href='https://rmagno.eu/'>Ramiro Magno</a>, Ana-Teresa Maia, <a href='https://cintesis.eu/'><img src='reference/figures/cintesis.svg' height=24></a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


