<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en-US"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>An S4 class to represent a set of GWAS Catalog variants — variants-class • gwasrapidd</title><!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png"><link rel="icon" type="”image/svg+xml”" href="../favicon.svg"><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png"><link rel="icon" sizes="any" href="../favicon.ico"><link rel="manifest" href="../site.webmanifest"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/Nunito-0.4.10/font.css" rel="stylesheet"><link href="../deps/Inconsolata-0.4.10/font.css" rel="stylesheet"><link href="../deps/Source_Sans_Pro-0.4.10/font.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="An S4 class to represent a set of GWAS Catalog variants — variants-class"><meta name="description" content="The variants object consists of four slots, each a table
(tibble), that combined form a relational database of a
subset of GWAS Catalog variants. Each variant is an observation (row) in the
variants table — main table. All tables have the column
variant_id as primary key."><meta property="og:description" content="The variants object consists of four slots, each a table
(tibble), that combined form a relational database of a
subset of GWAS Catalog variants. Each variant is an observation (row) in the
variants table — main table. All tables have the column
variant_id as primary key."><meta property="og:image" content="https://rmagno.eu/gwasrapidd/logo.svg"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-none" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">gwasrapidd</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.18</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/gwasrapidd.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/bmi_variants.html">Variants associated with Body Mass Index (BMI)</a></li>
    <li><a class="dropdown-item" href="../articles/faq.html">Frequently asked questions</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/ramiromagno/gwasrapidd/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>An S4 class to represent a set of GWAS Catalog variants</h1>
      <small class="dont-index">Source: <a href="https://github.com/ramiromagno/gwasrapidd/blob/master/R/class-variants.R" class="external-link"><code>R/class-variants.R</code></a></small>
      <div class="d-none name"><code>variants-class.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>The variants object consists of four slots, each a table
(<code><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></code>), that combined form a relational database of a
subset of GWAS Catalog variants. Each variant is an observation (row) in the
<code>variants</code> table — main table. All tables have the column
<code>variant_id</code> as primary key.</p>
    </div>


    <div class="section level2">
    <h2 id="slots">Slots<a class="anchor" aria-label="anchor" href="#slots"></a></h2>


<dl><dt><code>variants</code></dt>
<dd><p>A <code><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></code> listing variants. Columns:</p><dl><dt>variant_id</dt>
<dd><p>Variant identifier, e.g., <code>'rs1333048'</code>.</p></dd>

<dt>merged</dt>
<dd><p>Whether this SNP has been merged with another SNP in a newer
genome build.</p></dd>

<dt>functional_class</dt>
<dd><p>Class according to Ensembl's predicted consequences
that each variant allele may have on transcripts. See
<a href="https://www.ensembl.org/info/genome/variation/prediction/predicted_data.html" class="external-link">Ensembl
Variation - Calculated variant consequences</a>.</p></dd>

<dt>chromosome_name</dt>
<dd><p>Chromosome name.</p></dd>

<dt>chromosome_position</dt>
<dd><p>Chromosome position.</p></dd>

<dt>chromosome_region</dt>
<dd><p><a href="https://medlineplus.gov/genetics/understanding/howgeneswork/genelocation/" class="external-link">Cytogenetic
location</a>.</p></dd>

<dt>last_update_date</dt>
<dd><p>Last time this variant was updated.</p></dd>


</dl></dd>


<dt><code>genomic_contexts</code></dt>
<dd><p>A <code><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></code> listing genomic
  contexts associated with each variant. Columns:</p><dl><dt>variant_id</dt>
<dd><p>Variant identifier.</p></dd>

<dt>gene_name</dt>
<dd><p>Gene symbol according to
<a href="https://www.genenames.org/" class="external-link">HUGO Gene Nomenclature (HGNC)</a>.</p></dd>

<dt>chromosome_name</dt>
<dd><p>Chromosome name.</p></dd>

<dt>chromosome_position</dt>
<dd><p>Chromosome position.</p></dd>

<dt>distance</dt>
<dd><p>Genomic distance between the variant and the gene (in base
pairs).</p></dd>

<dt>is_mapped_gene</dt>
<dd><p>Whether this is a mapped gene to this variant. A mapped
gene is either an overlapping gene with the variant or the two closest genes
upstream and downstream of the variant. Moreover, only genes whose mapping
source is 'Ensembl' are considered.</p></dd>

<dt>is_closest_gene</dt>
<dd><p>Whether this is the closest gene to this variant.</p></dd>

<dt>is_intergenic</dt>
<dd><p>Whether this variant is intergenic, i.e, if there is no
gene up or downstream within 100kb.</p></dd>

<dt>is_upstream</dt>
<dd><p>Whether this variant is upstream of this gene.</p></dd>

<dt>is_downstream</dt>
<dd><p>Whether this variant is downstream of this gene.</p></dd>

<dt>source</dt>
<dd><p>Gene mapping source, either <code>Ensembl</code> or <code>NCBI</code>.</p></dd>

<dt>mapping_method</dt>
<dd><p>Gene mapping method.</p></dd>


</dl></dd>


<dt><code>ensembl_ids</code></dt>
<dd><p>A <code><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></code> listing gene Ensembl
  identifiers associated with each genomic context. Columns:</p><dl><dt>variant_id</dt>
<dd><p>Variant identifier.</p></dd>

<dt>gene_name</dt>
<dd><p>Gene symbol according to
<a href="https://www.genenames.org/" class="external-link">HUGO Gene Nomenclature (HGNC)</a>.</p></dd>

<dt>ensembl_id</dt>
<dd><p>The Ensembl identifier of an Ensembl gene, see Section
<a href="https://www.ensembl.org/info/genome/genebuild/index.html" class="external-link">Gene
annotation in Ensembl</a> for more information.</p></dd>


</dl></dd>


<dt><code>entrez_ids</code></dt>
<dd><p>A <code><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></code> listing gene Entrez
  identifiers associated with each genomic context. Columns:</p><dl><dt>variant_id</dt>
<dd><p>Variant identifier.</p></dd>

<dt>gene_name</dt>
<dd><p>Gene symbol according to
<a href="https://www.genenames.org/" class="external-link">HUGO Gene Nomenclature (HGNC)</a>.</p></dd>

<dt>entrez_id</dt>
<dd><p>The Entrez identifier of a gene, see ref.
<a href="https://doi.org/10.1093/nar/gkq1237" class="external-link">doi:10.1093/nar/gkq1237</a>
 for
more information.</p></dd>


</dl></dd>


</dl></div>

  </main></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://www.pattern.institute/team/rmagno/" class="external-link">Ramiro Magno</a>, Ana-Teresa Maia, CINTESIS, <a href="https://www.pattern.institute" class="external-link">Pattern Institute</a>.</p>
</div>

<div class="pkgdown-footer-right">

</div>

    </footer></div>





  </body></html>

