<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Set operations on GWAS Catalog objects. — setop • gwasrapidd</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />
<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



  <link href="../extra.css" rel="stylesheet">
  
<meta property="og:title" content="Set operations on GWAS Catalog objects. — setop" />

<meta property="og:description" content="Performs set union, intersection, and (asymmetric!) difference on two objects
of either class studies, associations,
variants, or traits. Note that union()
removes duplicated entities, whereas bind() does
not." />

<meta property="og:image" content="https://rmagno.eu/gwasrapidd/logo.png" />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gwasrapidd</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.5</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/gwasrapidd.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/faq.html">Frequently asked questions</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/ramiromagno/gwasrapidd">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Set operations on GWAS Catalog objects.</h1>
    <small class="dont-index">Source: <a href='https://github.com/ramiromagno/gwasrapidd/blob/master/R/generics.R'><code>R/generics.R</code></a></small>
    <div class="hidden name"><code>setop.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>Performs set union, intersection, and (asymmetric!) difference on two objects
of either class <a href='studies-class.html'>studies</a>, <a href='associations-class.html'>associations</a>,
<a href='variants-class.html'>variants</a>, or <a href='traits-class.html'>traits</a>. Note that <code>union()</code>
removes duplicated entities, whereas <code><a href='bind.html'>bind</a>()</code> does
not.</p>
    
    </div>

    <pre class="usage"><span class='fu'>union</span>(<span class='no'>x</span>, <span class='no'>y</span>, <span class='no'>...</span>)

<span class='fu'>intersect</span>(<span class='no'>x</span>, <span class='no'>y</span>, <span class='no'>...</span>)

<span class='fu'>setdiff</span>(<span class='no'>x</span>, <span class='no'>y</span>, <span class='no'>...</span>)

<span class='fu'>setequal</span>(<span class='no'>x</span>, <span class='no'>y</span>, <span class='no'>...</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>x, y</th>
      <td><p>Objects of either class <a href='studies-class.html'>studies</a>,
<a href='associations-class.html'>associations</a>, <a href='variants-class.html'>variants</a>, or
<a href='traits-class.html'>traits</a>.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>other arguments passed on to methods.</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>An object of the same class as <code>x</code> and <code>y</code>, i.e.,
  <a href='studies-class.html'>studies</a>, <a href='associations-class.html'>associations</a>, <a href='variants-class.html'>variants</a>,
  or <a href='traits-class.html'>traits</a>.</p>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'>#</span>
<span class='co'># union()</span>
<span class='co'>#</span>
<span class='co'># Combine studies and remove duplicates</span>
<span class='fu'>union</span>(<span class='no'>studies_ex01</span>, <span class='no'>studies_ex02</span>)</div><div class='output co'>#&gt; An object of class "studies"
#&gt; Slot "studies":
#&gt; <span style='color: #949494;'># A tibble: 3 x 13</span><span>
#&gt;   study_id reported_trait initial_sample_… replication_sam… gxe   gxg  
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>    </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>          </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>            </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>            </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span>
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> GCST001… Breast size    16,175 European… </span><span style='color: #BB0000;'>NA</span><span>               FALSE FALSE
#&gt; </span><span style='color: #BCBCBC;'>2</span><span> GCST003… Breast size    33,790 European… </span><span style='color: #BB0000;'>NA</span><span>               FALSE FALSE
#&gt; </span><span style='color: #BCBCBC;'>3</span><span> GCST006… Breast size    5,609 Japanese … </span><span style='color: #BB0000;'>NA</span><span>               FALSE FALSE
#&gt; </span><span style='color: #949494;'># … with 7 more variables: snp_count </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span style='color: #949494;'>, qualifier </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span style='color: #949494;'>, imputed </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span style='color: #949494;'>,
#&gt; #   pooled </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span style='color: #949494;'>, study_design_comment </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span style='color: #949494;'>, full_pvalue_set </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span style='color: #949494;'>,
#&gt; #   user_requested </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span>
#&gt; 
#&gt; Slot "genotyping_techs":
#&gt; </span><span style='color: #949494;'># A tibble: 3 x 2</span><span>
#&gt;   study_id   genotyping_technology       
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>      </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>                       
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> GCST001585 Genome-wide genotyping array
#&gt; </span><span style='color: #BCBCBC;'>2</span><span> GCST003985 Genome-wide genotyping array
#&gt; </span><span style='color: #BCBCBC;'>3</span><span> GCST006655 Genome-wide genotyping array
#&gt; 
#&gt; Slot "platforms":
#&gt; </span><span style='color: #949494;'># A tibble: 3 x 2</span><span>
#&gt;   study_id   manufacturer
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>      </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>       
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> GCST001585 Illumina    
#&gt; </span><span style='color: #BCBCBC;'>2</span><span> GCST003985 Illumina    
#&gt; </span><span style='color: #BCBCBC;'>3</span><span> GCST006655 Affymetrix  
#&gt; 
#&gt; Slot "ancestries":
#&gt; </span><span style='color: #949494;'># A tibble: 3 x 4</span><span>
#&gt;   study_id   ancestry_id type    number_of_individuals
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>            </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>                   </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span>
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> GCST001585           1 initial                 </span><span style='text-decoration: underline;'>16</span><span>175
#&gt; </span><span style='color: #BCBCBC;'>2</span><span> GCST003985           1 initial                 </span><span style='text-decoration: underline;'>33</span><span>790
#&gt; </span><span style='color: #BCBCBC;'>3</span><span> GCST006655           1 initial                  </span><span style='text-decoration: underline;'>5</span><span>609
#&gt; 
#&gt; Slot "ancestral_groups":
#&gt; </span><span style='color: #949494;'># A tibble: 3 x 3</span><span>
#&gt;   study_id   ancestry_id ancestral_group
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>            </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>          
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> GCST001585           1 European       
#&gt; </span><span style='color: #BCBCBC;'>2</span><span> GCST003985           1 European       
#&gt; </span><span style='color: #BCBCBC;'>3</span><span> GCST006655           1 East Asian     
#&gt; 
#&gt; Slot "countries_of_origin":
#&gt; </span><span style='color: #949494;'># A tibble: 3 x 5</span><span>
#&gt;   study_id   ancestry_id country_name major_area region
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>            </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>        </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>      </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span> 
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> GCST001585           1 </span><span style='color: #BB0000;'>NA</span><span>           </span><span style='color: #BB0000;'>NA</span><span>         </span><span style='color: #BB0000;'>NA</span><span>    
#&gt; </span><span style='color: #BCBCBC;'>2</span><span> GCST003985           1 </span><span style='color: #BB0000;'>NA</span><span>           </span><span style='color: #BB0000;'>NA</span><span>         </span><span style='color: #BB0000;'>NA</span><span>    
#&gt; </span><span style='color: #BCBCBC;'>3</span><span> GCST006655           1 </span><span style='color: #BB0000;'>NA</span><span>           </span><span style='color: #BB0000;'>NA</span><span>         </span><span style='color: #BB0000;'>NA</span><span>    
#&gt; 
#&gt; Slot "countries_of_recruitment":
#&gt; </span><span style='color: #949494;'># A tibble: 3 x 5</span><span>
#&gt;   study_id   ancestry_id country_name major_area region
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>            </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>        </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>      </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span> 
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> GCST001585           1 </span><span style='color: #BB0000;'>NA</span><span>           </span><span style='color: #BB0000;'>NA</span><span>         </span><span style='color: #BB0000;'>NA</span><span>    
#&gt; </span><span style='color: #BCBCBC;'>2</span><span> GCST003985           1 </span><span style='color: #BB0000;'>NA</span><span>           </span><span style='color: #BB0000;'>NA</span><span>         </span><span style='color: #BB0000;'>NA</span><span>    
#&gt; </span><span style='color: #BCBCBC;'>3</span><span> GCST006655           1 </span><span style='color: #BB0000;'>NA</span><span>           </span><span style='color: #BB0000;'>NA</span><span>         </span><span style='color: #BB0000;'>NA</span><span>    
#&gt; 
#&gt; Slot "publications":
#&gt; </span><span style='color: #949494;'># A tibble: 3 x 7</span><span>
#&gt;   study_id pubmed_id publication_date publication title author_fullname
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>        </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;date&gt;</span><span>           </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>       </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>          
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> GCST001…  22</span><span style='text-decoration: underline;'>747</span><span>683 2012-06-29       BMC Med Ge… Gene… Eriksson N     
#&gt; </span><span style='color: #BCBCBC;'>2</span><span> GCST003…  27</span><span style='text-decoration: underline;'>182</span><span>965 2016-05-15       Nat Genet   Dete… Pickrell JK    
#&gt; </span><span style='color: #BCBCBC;'>3</span><span> GCST006…  29</span><span style='text-decoration: underline;'>855</span><span>537 2018-05-30       Sci Rep     Japa… Hirata T       
#&gt; </span><span style='color: #949494;'># … with 1 more variable: author_orcid </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>
#&gt; </div><div class='input'>
<span class='co'># Combine associations and remove duplicates</span>
<span class='fu'>union</span>(<span class='no'>associations_ex01</span>, <span class='no'>associations_ex02</span>)</div><div class='output co'>#&gt; An object of class "associations"
#&gt; Slot "associations":
#&gt; </span><span style='color: #949494;'># A tibble: 6 x 17</span><span>
#&gt;   association_id   pvalue pvalue_descript… pvalue_mantissa pvalue_exponent
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>             </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>                      </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span>           </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span>
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> 22509          3.00</span><span style='color: #949494;'>e</span><span style='color: #BB0000;'>- 9</span><span> </span><span style='color: #BB0000;'>NA</span><span>                             3              -</span><span style='color: #BB0000;'>9</span><span>
#&gt; </span><span style='color: #BCBCBC;'>2</span><span> 22505          7.00</span><span style='color: #949494;'>e</span><span style='color: #BB0000;'>- 6</span><span> </span><span style='color: #BB0000;'>NA</span><span>                             7              -</span><span style='color: #BB0000;'>6</span><span>
#&gt; </span><span style='color: #BCBCBC;'>3</span><span> 19537565       7.00</span><span style='color: #949494;'>e</span><span style='color: #BB0000;'>-31</span><span> </span><span style='color: #BB0000;'>NA</span><span>                             7             -</span><span style='color: #BB0000;'>31</span><span>
#&gt; </span><span style='color: #BCBCBC;'>4</span><span> 19537593       1.00</span><span style='color: #949494;'>e</span><span style='color: #BB0000;'>-11</span><span> </span><span style='color: #BB0000;'>NA</span><span>                             1             -</span><span style='color: #BB0000;'>11</span><span>
#&gt; </span><span style='color: #BCBCBC;'>5</span><span> 31665940       8.00</span><span style='color: #949494;'>e</span><span style='color: #BB0000;'>- 6</span><span> </span><span style='color: #BB0000;'>NA</span><span>                             8              -</span><span style='color: #BB0000;'>6</span><span>
#&gt; </span><span style='color: #BCBCBC;'>6</span><span> 34944736       6.00</span><span style='color: #949494;'>e</span><span style='color: #BB0000;'>- 9</span><span> </span><span style='color: #BB0000;'>NA</span><span>                             6              -</span><span style='color: #BB0000;'>9</span><span>
#&gt; </span><span style='color: #949494;'># … with 12 more variables: multiple_snp_haplotype </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span style='color: #949494;'>,
#&gt; #   snp_interaction </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span style='color: #949494;'>, snp_type </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span style='color: #949494;'>, standard_error </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span style='color: #949494;'>, range </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span style='color: #949494;'>,
#&gt; #   or_per_copy_number </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span style='color: #949494;'>, beta_number </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span style='color: #949494;'>, beta_unit </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span style='color: #949494;'>,
#&gt; #   beta_direction </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span style='color: #949494;'>, beta_description </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span style='color: #949494;'>, last_mapping_date </span><span style='color: #949494;font-style: italic;'>&lt;dttm&gt;</span><span style='color: #949494;'>,
#&gt; #   last_update_date </span><span style='color: #949494;font-style: italic;'>&lt;dttm&gt;</span><span>
#&gt; 
#&gt; Slot "loci":
#&gt; </span><span style='color: #949494;'># A tibble: 6 x 4</span><span>
#&gt;   association_id locus_id haplotype_snp_count description   
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>             </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span>               </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>         
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> 22509                 1                  </span><span style='color: #BB0000;'>NA</span><span> Single variant
#&gt; </span><span style='color: #BCBCBC;'>2</span><span> 22505                 1                  </span><span style='color: #BB0000;'>NA</span><span> Single variant
#&gt; </span><span style='color: #BCBCBC;'>3</span><span> 19537565              1                  </span><span style='color: #BB0000;'>NA</span><span> Single variant
#&gt; </span><span style='color: #BCBCBC;'>4</span><span> 19537593              1                  </span><span style='color: #BB0000;'>NA</span><span> Single variant
#&gt; </span><span style='color: #BCBCBC;'>5</span><span> 31665940              1                  </span><span style='color: #BB0000;'>NA</span><span> Single variant
#&gt; </span><span style='color: #BCBCBC;'>6</span><span> 34944736              1                  </span><span style='color: #BB0000;'>NA</span><span> Single variant
#&gt; 
#&gt; Slot "risk_alleles":
#&gt; </span><span style='color: #949494;'># A tibble: 6 x 7</span><span>
#&gt;   association_id locus_id variant_id risk_allele risk_frequency genome_wide
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>             </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>      </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>                </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span>      
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> 22509                 1 rs7089814  C                   </span><span style='color: #BB0000;'>NA</span><span>     </span><span style='color: #BB0000;'>NA</span><span>         
#&gt; </span><span style='color: #BCBCBC;'>2</span><span> 22505                 1 chr22:458… T                   </span><span style='color: #BB0000;'>NA</span><span>     </span><span style='color: #BB0000;'>NA</span><span>         
#&gt; </span><span style='color: #BCBCBC;'>3</span><span> 19537565              1 rs10110651 </span><span style='color: #BB0000;'>NA</span><span>                  </span><span style='color: #BB0000;'>NA</span><span>     FALSE      
#&gt; </span><span style='color: #BCBCBC;'>4</span><span> 19537593              1 rs9397437  </span><span style='color: #BB0000;'>NA</span><span>                  </span><span style='color: #BB0000;'>NA</span><span>     FALSE      
#&gt; </span><span style='color: #BCBCBC;'>5</span><span> 31665940              1 rs985648   G                    0.248 FALSE      
#&gt; </span><span style='color: #BCBCBC;'>6</span><span> 34944736              1 rs1469924… TTCTTTCTTTC          0.381 FALSE      
#&gt; </span><span style='color: #949494;'># … with 1 more variable: limited_list </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span>
#&gt; 
#&gt; Slot "genes":
#&gt; </span><span style='color: #949494;'># A tibble: 11 x 3</span><span>
#&gt;    association_id locus_id gene_name   
#&gt;    </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>             </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>       
#&gt; </span><span style='color: #BCBCBC;'> 1</span><span> 22509                 1 ADO         
#&gt; </span><span style='color: #BCBCBC;'> 2</span><span> 22509                 1 ZNF365      
#&gt; </span><span style='color: #BCBCBC;'> 3</span><span> 22509                 1 ALDH7A1P4   
#&gt; </span><span style='color: #BCBCBC;'> 4</span><span> 22505                 1 TBC1D22A    
#&gt; </span><span style='color: #BCBCBC;'> 5</span><span> 22505                 1 LOC100128818
#&gt; </span><span style='color: #BCBCBC;'> 6</span><span> 19537565              1 KCNU1       
#&gt; </span><span style='color: #BCBCBC;'> 7</span><span> 19537565              1 ZNF703      
#&gt; </span><span style='color: #BCBCBC;'> 8</span><span> 19537593              1 CCDC170     
#&gt; </span><span style='color: #BCBCBC;'> 9</span><span> 19537593              1 ESR1        
#&gt; </span><span style='color: #BCBCBC;'>10</span><span> 31665940              1 </span><span style='color: #BB0000;'>NA</span><span>          
#&gt; </span><span style='color: #BCBCBC;'>11</span><span> 34944736              1 KCNU1       
#&gt; 
#&gt; Slot "ensembl_ids":
#&gt; </span><span style='color: #949494;'># A tibble: 11 x 4</span><span>
#&gt;    association_id locus_id gene_name    ensembl_id     
#&gt;    </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>             </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>        </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>          
#&gt; </span><span style='color: #BCBCBC;'> 1</span><span> 22509                 1 ADO          ENSG00000181915
#&gt; </span><span style='color: #BCBCBC;'> 2</span><span> 22509                 1 ZNF365       ENSG00000138311
#&gt; </span><span style='color: #BCBCBC;'> 3</span><span> 22509                 1 ALDH7A1P4    ENSG00000234489
#&gt; </span><span style='color: #BCBCBC;'> 4</span><span> 22505                 1 TBC1D22A     ENSG00000054611
#&gt; </span><span style='color: #BCBCBC;'> 5</span><span> 22505                 1 LOC100128818 </span><span style='color: #BB0000;'>NA</span><span>             
#&gt; </span><span style='color: #BCBCBC;'> 6</span><span> 19537565              1 KCNU1        ENSG00000215262
#&gt; </span><span style='color: #BCBCBC;'> 7</span><span> 19537565              1 ZNF703       ENSG00000183779
#&gt; </span><span style='color: #BCBCBC;'> 8</span><span> 19537593              1 CCDC170      ENSG00000120262
#&gt; </span><span style='color: #BCBCBC;'> 9</span><span> 19537593              1 ESR1         ENSG00000091831
#&gt; </span><span style='color: #BCBCBC;'>10</span><span> 31665940              1 </span><span style='color: #BB0000;'>NA</span><span>           </span><span style='color: #BB0000;'>NA</span><span>             
#&gt; </span><span style='color: #BCBCBC;'>11</span><span> 34944736              1 KCNU1        ENSG00000215262
#&gt; 
#&gt; Slot "entrez_ids":
#&gt; </span><span style='color: #949494;'># A tibble: 11 x 4</span><span>
#&gt;    association_id locus_id gene_name    entrez_id
#&gt;    </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>             </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>        </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>    
#&gt; </span><span style='color: #BCBCBC;'> 1</span><span> 22509                 1 ADO          84890    
#&gt; </span><span style='color: #BCBCBC;'> 2</span><span> 22509                 1 ZNF365       22891    
#&gt; </span><span style='color: #BCBCBC;'> 3</span><span> 22509                 1 ALDH7A1P4    544      
#&gt; </span><span style='color: #BCBCBC;'> 4</span><span> 22505                 1 TBC1D22A     25771    
#&gt; </span><span style='color: #BCBCBC;'> 5</span><span> 22505                 1 LOC100128818 </span><span style='color: #BB0000;'>NA</span><span>       
#&gt; </span><span style='color: #BCBCBC;'> 6</span><span> 19537565              1 KCNU1        157855   
#&gt; </span><span style='color: #BCBCBC;'> 7</span><span> 19537565              1 ZNF703       80139    
#&gt; </span><span style='color: #BCBCBC;'> 8</span><span> 19537593              1 CCDC170      80129    
#&gt; </span><span style='color: #BCBCBC;'> 9</span><span> 19537593              1 ESR1         2099     
#&gt; </span><span style='color: #BCBCBC;'>10</span><span> 31665940              1 </span><span style='color: #BB0000;'>NA</span><span>           </span><span style='color: #BB0000;'>NA</span><span>       
#&gt; </span><span style='color: #BCBCBC;'>11</span><span> 34944736              1 KCNU1        157855   
#&gt; </div><div class='input'>
<span class='co'># Combine variants and remove duplicates</span>
<span class='fu'>union</span>(<span class='no'>variants_ex01</span>, <span class='no'>variants_ex02</span>)</div><div class='output co'>#&gt; An object of class "variants"
#&gt; Slot "variants":
#&gt; </span><span style='color: #949494;'># A tibble: 5 x 7</span><span>
#&gt;   variant_id merged functional_class chromosome_name chromosome_posi…
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>       </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>            </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>                      </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span>
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> rs1469924…      0 regulatory_regi… 8                       36</span><span style='text-decoration: underline;'>984</span><span>538
#&gt; </span><span style='color: #BCBCBC;'>2</span><span> rs56261590      0 intron_variant   4                       73</span><span style='text-decoration: underline;'>189</span><span>781
#&gt; </span><span style='color: #BCBCBC;'>3</span><span> rs4725504       0 intergenic_vari… 7                      153</span><span style='text-decoration: underline;'>608</span><span>338
#&gt; </span><span style='color: #BCBCBC;'>4</span><span> rs11099757      0 intron_variant   4                      150</span><span style='text-decoration: underline;'>143</span><span>216
#&gt; </span><span style='color: #BCBCBC;'>5</span><span> rs16871509      0 intergenic_vari… 5                       71</span><span style='text-decoration: underline;'>726</span><span>395
#&gt; </span><span style='color: #949494;'># … with 2 more variables: chromosome_region </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span style='color: #949494;'>, last_update_date </span><span style='color: #949494;font-style: italic;'>&lt;dttm&gt;</span><span>
#&gt; 
#&gt; Slot "genomic_contexts":
#&gt; </span><span style='color: #949494;'># A tibble: 33 x 11</span><span>
#&gt;    variant_id gene_name chromosome_name chromosome_posi… distance
#&gt;    </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>      </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>     </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>                      </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span>    </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span>
#&gt; </span><span style='color: #BCBCBC;'> 1</span><span> rs1469924… RPL26P25  8                       36</span><span style='text-decoration: underline;'>984</span><span>538     </span><span style='text-decoration: underline;'>3</span><span>392
#&gt; </span><span style='color: #BCBCBC;'> 2</span><span> rs1469924… TPT1P8    8                       36</span><span style='text-decoration: underline;'>984</span><span>538    </span><span style='text-decoration: underline;'>95</span><span>008
#&gt; </span><span style='color: #BCBCBC;'> 3</span><span> rs1469924… LOC10537… 8                       36</span><span style='text-decoration: underline;'>984</span><span>538    </span><span style='text-decoration: underline;'>59</span><span>331
#&gt; </span><span style='color: #BCBCBC;'> 4</span><span> rs1469924… LOC10537… 8                       36</span><span style='text-decoration: underline;'>984</span><span>538    </span><span style='text-decoration: underline;'>81</span><span>555
#&gt; </span><span style='color: #BCBCBC;'> 5</span><span> rs1469924… LOC10537… 8                       36</span><span style='text-decoration: underline;'>984</span><span>538     </span><span style='text-decoration: underline;'>2</span><span>533
#&gt; </span><span style='color: #BCBCBC;'> 6</span><span> rs1469924… TPT1P8    8                       36</span><span style='text-decoration: underline;'>984</span><span>538    </span><span style='text-decoration: underline;'>95</span><span>231
#&gt; </span><span style='color: #BCBCBC;'> 7</span><span> rs1469924… AC092818… 8                       36</span><span style='text-decoration: underline;'>984</span><span>538    </span><span style='text-decoration: underline;'>82</span><span>903
#&gt; </span><span style='color: #BCBCBC;'> 8</span><span> rs1469924… KCNU1     8                       36</span><span style='text-decoration: underline;'>984</span><span>538    </span><span style='text-decoration: underline;'>48</span><span>413
#&gt; </span><span style='color: #BCBCBC;'> 9</span><span> rs1469924… AC090453… 8                       36</span><span style='text-decoration: underline;'>984</span><span>538     </span><span style='text-decoration: underline;'>3</span><span>439
#&gt; </span><span style='color: #BCBCBC;'>10</span><span> rs1469924… KCNU1     8                       36</span><span style='text-decoration: underline;'>984</span><span>538    </span><span style='text-decoration: underline;'>48</span><span>410
#&gt; </span><span style='color: #949494;'># … with 23 more rows, and 6 more variables: is_closest_gene </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span style='color: #949494;'>,
#&gt; #   is_intergenic </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span style='color: #949494;'>, is_upstream </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span style='color: #949494;'>, is_downstream </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span style='color: #949494;'>, source </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span style='color: #949494;'>,
#&gt; #   mapping_method </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>
#&gt; 
#&gt; Slot "ensembl_ids":
#&gt; </span><span style='color: #949494;'># A tibble: 17 x 3</span><span>
#&gt;    variant_id  gene_name  ensembl_id     
#&gt;    </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>       </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>      </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>          
#&gt; </span><span style='color: #BCBCBC;'> 1</span><span> rs146992477 TPT1P8     ENSG00000219491
#&gt; </span><span style='color: #BCBCBC;'> 2</span><span> rs146992477 AC092818.1 ENSG00000254038
#&gt; </span><span style='color: #BCBCBC;'> 3</span><span> rs146992477 KCNU1      ENSG00000215262
#&gt; </span><span style='color: #BCBCBC;'> 4</span><span> rs146992477 AC090453.1 ENSG00000243503
#&gt; </span><span style='color: #BCBCBC;'> 5</span><span> rs56261590  ANKRD17    ENSG00000132466
#&gt; </span><span style='color: #BCBCBC;'> 6</span><span> rs56261590  AC053527.1 ENSG00000250220
#&gt; </span><span style='color: #BCBCBC;'> 7</span><span> rs56261590  HMGA1P2    ENSG00000248641
#&gt; </span><span style='color: #BCBCBC;'> 8</span><span> rs56261590  RF00334    ENSG00000251778
#&gt; </span><span style='color: #BCBCBC;'> 9</span><span> rs4725504   DPP6       ENSG00000130226
#&gt; </span><span style='color: #BCBCBC;'>10</span><span> rs4725504   PAXBP1P1   ENSG00000233489
#&gt; </span><span style='color: #BCBCBC;'>11</span><span> rs11099757  DCLK2      ENSG00000170390
#&gt; </span><span style='color: #BCBCBC;'>12</span><span> rs11099757  RNU7-194P  ENSG00000238721
#&gt; </span><span style='color: #BCBCBC;'>13</span><span> rs11099757  RNU6-1230P ENSG00000207121
#&gt; </span><span style='color: #BCBCBC;'>14</span><span> rs11099757  AC105343.1 ENSG00000250704
#&gt; </span><span style='color: #BCBCBC;'>15</span><span> rs16871509  MCCC2      ENSG00000131844
#&gt; </span><span style='color: #BCBCBC;'>16</span><span> rs16871509  AC143336.1 ENSG00000278824
#&gt; </span><span style='color: #BCBCBC;'>17</span><span> rs16871509  CARTPT     ENSG00000164326
#&gt; 
#&gt; Slot "entrez_ids":
#&gt; </span><span style='color: #949494;'># A tibble: 18 x 3</span><span>
#&gt;    variant_id  gene_name    entrez_id
#&gt;    </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>       </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>        </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>    
#&gt; </span><span style='color: #BCBCBC;'> 1</span><span> rs146992477 RPL26P25     100270966
#&gt; </span><span style='color: #BCBCBC;'> 2</span><span> rs146992477 TPT1P8       59347    
#&gt; </span><span style='color: #BCBCBC;'> 3</span><span> rs146992477 LOC105379375 105379375
#&gt; </span><span style='color: #BCBCBC;'> 4</span><span> rs146992477 LOC105379377 105379377
#&gt; </span><span style='color: #BCBCBC;'> 5</span><span> rs146992477 LOC105379376 105379376
#&gt; </span><span style='color: #BCBCBC;'> 6</span><span> rs146992477 KCNU1        157855   
#&gt; </span><span style='color: #BCBCBC;'> 7</span><span> rs56261590  LOC102724832 102724832
#&gt; </span><span style='color: #BCBCBC;'> 8</span><span> rs56261590  ANKRD17      26057    
#&gt; </span><span style='color: #BCBCBC;'> 9</span><span> rs56261590  HMGA1P2      171559   
#&gt; </span><span style='color: #BCBCBC;'>10</span><span> rs4725504   DPP6         1804     
#&gt; </span><span style='color: #BCBCBC;'>11</span><span> rs4725504   LOC107986750 107986750
#&gt; </span><span style='color: #BCBCBC;'>12</span><span> rs4725504   PAXBP1P1     106481704
#&gt; </span><span style='color: #BCBCBC;'>13</span><span> rs11099757  DCLK2        166614   
#&gt; </span><span style='color: #BCBCBC;'>14</span><span> rs11099757  RNU7-194P    106480834
#&gt; </span><span style='color: #BCBCBC;'>15</span><span> rs11099757  RNU6-1230P   106481944
#&gt; </span><span style='color: #BCBCBC;'>16</span><span> rs16871509  MCCC2        64087    
#&gt; </span><span style='color: #BCBCBC;'>17</span><span> rs16871509  CARTPT       9607     
#&gt; </span><span style='color: #BCBCBC;'>18</span><span> rs16871509  LOC100422013 100422013
#&gt; </div><div class='input'>
<span class='co'># Combine traits and remove duplicates</span>
<span class='fu'>union</span>(<span class='no'>traits_ex01</span>, <span class='no'>traits_ex02</span>)</div><div class='output co'>#&gt; An object of class "traits"
#&gt; Slot "traits":
#&gt; </span><span style='color: #949494;'># A tibble: 6 x 3</span><span>
#&gt;   efo_id      trait                      uri                                 
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>       </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>                      </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>                               
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> EFO_0004884 breast size                http://www.ebi.ac.uk/efo/EFO_0004884
#&gt; </span><span style='color: #BCBCBC;'>2</span><span> EFO_0004343 waist-hip ratio            http://www.ebi.ac.uk/efo/EFO_0004343
#&gt; </span><span style='color: #BCBCBC;'>3</span><span> EFO_0005299 non-word reading           http://www.ebi.ac.uk/efo/EFO_0005299
#&gt; </span><span style='color: #BCBCBC;'>4</span><span> EFO_0007845 lip morphology measurement http://www.ebi.ac.uk/efo/EFO_0007845
#&gt; </span><span style='color: #BCBCBC;'>5</span><span> EFO_0004699 gambling                   http://www.ebi.ac.uk/efo/EFO_0004699
#&gt; </span><span style='color: #BCBCBC;'>6</span><span> EFO_0004875 mathematical ability       http://www.ebi.ac.uk/efo/EFO_0004875
#&gt; </div><div class='input'>
<span class='co'>#</span>
<span class='co'># intersect()</span>
<span class='co'>#</span>
<span class='co'># Intersect common studies</span>
<span class='fu'>intersect</span>(<span class='no'>studies_ex01</span>, <span class='no'>studies_ex02</span>)</div><div class='output co'>#&gt; An object of class "studies"
#&gt; Slot "studies":
#&gt; </span><span style='color: #949494;'># A tibble: 1 x 13</span><span>
#&gt;   study_id reported_trait initial_sample_… replication_sam… gxe   gxg  
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>    </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>          </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>            </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>            </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span>
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> GCST001… Breast size    16,175 European… </span><span style='color: #BB0000;'>NA</span><span>               FALSE FALSE
#&gt; </span><span style='color: #949494;'># … with 7 more variables: snp_count </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span style='color: #949494;'>, qualifier </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span style='color: #949494;'>, imputed </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span style='color: #949494;'>,
#&gt; #   pooled </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span style='color: #949494;'>, study_design_comment </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span style='color: #949494;'>, full_pvalue_set </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span style='color: #949494;'>,
#&gt; #   user_requested </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span>
#&gt; 
#&gt; Slot "genotyping_techs":
#&gt; </span><span style='color: #949494;'># A tibble: 1 x 2</span><span>
#&gt;   study_id   genotyping_technology       
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>      </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>                       
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> GCST001585 Genome-wide genotyping array
#&gt; 
#&gt; Slot "platforms":
#&gt; </span><span style='color: #949494;'># A tibble: 1 x 2</span><span>
#&gt;   study_id   manufacturer
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>      </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>       
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> GCST001585 Illumina    
#&gt; 
#&gt; Slot "ancestries":
#&gt; </span><span style='color: #949494;'># A tibble: 1 x 4</span><span>
#&gt;   study_id   ancestry_id type    number_of_individuals
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>            </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>                   </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span>
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> GCST001585           1 initial                 </span><span style='text-decoration: underline;'>16</span><span>175
#&gt; 
#&gt; Slot "ancestral_groups":
#&gt; </span><span style='color: #949494;'># A tibble: 1 x 3</span><span>
#&gt;   study_id   ancestry_id ancestral_group
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>            </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>          
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> GCST001585           1 European       
#&gt; 
#&gt; Slot "countries_of_origin":
#&gt; </span><span style='color: #949494;'># A tibble: 1 x 5</span><span>
#&gt;   study_id   ancestry_id country_name major_area region
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>            </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>        </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>      </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span> 
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> GCST001585           1 </span><span style='color: #BB0000;'>NA</span><span>           </span><span style='color: #BB0000;'>NA</span><span>         </span><span style='color: #BB0000;'>NA</span><span>    
#&gt; 
#&gt; Slot "countries_of_recruitment":
#&gt; </span><span style='color: #949494;'># A tibble: 1 x 5</span><span>
#&gt;   study_id   ancestry_id country_name major_area region
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>            </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>        </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>      </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span> 
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> GCST001585           1 </span><span style='color: #BB0000;'>NA</span><span>           </span><span style='color: #BB0000;'>NA</span><span>         </span><span style='color: #BB0000;'>NA</span><span>    
#&gt; 
#&gt; Slot "publications":
#&gt; </span><span style='color: #949494;'># A tibble: 1 x 7</span><span>
#&gt;   study_id pubmed_id publication_date publication title author_fullname
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>        </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;date&gt;</span><span>           </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>       </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>          
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> GCST001…  22</span><span style='text-decoration: underline;'>747</span><span>683 2012-06-29       BMC Med Ge… Gene… Eriksson N     
#&gt; </span><span style='color: #949494;'># … with 1 more variable: author_orcid </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>
#&gt; </div><div class='input'>
<span class='co'># Intersect common associations</span>
<span class='fu'>intersect</span>(<span class='no'>associations_ex01</span>, <span class='no'>associations_ex02</span>)</div><div class='output co'>#&gt; An object of class "associations"
#&gt; Slot "associations":
#&gt; </span><span style='color: #949494;'># A tibble: 1 x 17</span><span>
#&gt;   association_id   pvalue pvalue_descript… pvalue_mantissa pvalue_exponent
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>             </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>                      </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span>           </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span>
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> 19537593       1.00</span><span style='color: #949494;'>e</span><span style='color: #BB0000;'>-11</span><span> </span><span style='color: #BB0000;'>NA</span><span>                             1             -</span><span style='color: #BB0000;'>11</span><span>
#&gt; </span><span style='color: #949494;'># … with 12 more variables: multiple_snp_haplotype </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span style='color: #949494;'>,
#&gt; #   snp_interaction </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span style='color: #949494;'>, snp_type </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span style='color: #949494;'>, standard_error </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span style='color: #949494;'>, range </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span style='color: #949494;'>,
#&gt; #   or_per_copy_number </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span style='color: #949494;'>, beta_number </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span style='color: #949494;'>, beta_unit </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span style='color: #949494;'>,
#&gt; #   beta_direction </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span style='color: #949494;'>, beta_description </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span style='color: #949494;'>, last_mapping_date </span><span style='color: #949494;font-style: italic;'>&lt;dttm&gt;</span><span style='color: #949494;'>,
#&gt; #   last_update_date </span><span style='color: #949494;font-style: italic;'>&lt;dttm&gt;</span><span>
#&gt; 
#&gt; Slot "loci":
#&gt; </span><span style='color: #949494;'># A tibble: 1 x 4</span><span>
#&gt;   association_id locus_id haplotype_snp_count description   
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>             </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span>               </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>         
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> 19537593              1                  </span><span style='color: #BB0000;'>NA</span><span> Single variant
#&gt; 
#&gt; Slot "risk_alleles":
#&gt; </span><span style='color: #949494;'># A tibble: 1 x 7</span><span>
#&gt;   association_id locus_id variant_id risk_allele risk_frequency genome_wide
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>             </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>      </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>                </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span>      
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> 19537593              1 rs9397437  </span><span style='color: #BB0000;'>NA</span><span>                      </span><span style='color: #BB0000;'>NA</span><span> FALSE      
#&gt; </span><span style='color: #949494;'># … with 1 more variable: limited_list </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span>
#&gt; 
#&gt; Slot "genes":
#&gt; </span><span style='color: #949494;'># A tibble: 2 x 3</span><span>
#&gt;   association_id locus_id gene_name
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>             </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>    
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> 19537593              1 CCDC170  
#&gt; </span><span style='color: #BCBCBC;'>2</span><span> 19537593              1 ESR1     
#&gt; 
#&gt; Slot "ensembl_ids":
#&gt; </span><span style='color: #949494;'># A tibble: 2 x 4</span><span>
#&gt;   association_id locus_id gene_name ensembl_id     
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>             </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>     </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>          
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> 19537593              1 CCDC170   ENSG00000120262
#&gt; </span><span style='color: #BCBCBC;'>2</span><span> 19537593              1 ESR1      ENSG00000091831
#&gt; 
#&gt; Slot "entrez_ids":
#&gt; </span><span style='color: #949494;'># A tibble: 2 x 4</span><span>
#&gt;   association_id locus_id gene_name entrez_id
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>             </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>     </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>    
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> 19537593              1 CCDC170   80129    
#&gt; </span><span style='color: #BCBCBC;'>2</span><span> 19537593              1 ESR1      2099     
#&gt; </div><div class='input'>
<span class='co'># Intersect common variants</span>
<span class='fu'>intersect</span>(<span class='no'>variants_ex01</span>, <span class='no'>variants_ex02</span>)</div><div class='output co'>#&gt; An object of class "variants"
#&gt; Slot "variants":
#&gt; </span><span style='color: #949494;'># A tibble: 2 x 7</span><span>
#&gt;   variant_id merged functional_class chromosome_name chromosome_posi…
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>       </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>            </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>                      </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span>
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> rs56261590      0 intron_variant   4                       73</span><span style='text-decoration: underline;'>189</span><span>781
#&gt; </span><span style='color: #BCBCBC;'>2</span><span> rs4725504       0 intergenic_vari… 7                      153</span><span style='text-decoration: underline;'>608</span><span>338
#&gt; </span><span style='color: #949494;'># … with 2 more variables: chromosome_region </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span style='color: #949494;'>, last_update_date </span><span style='color: #949494;font-style: italic;'>&lt;dttm&gt;</span><span>
#&gt; 
#&gt; Slot "genomic_contexts":
#&gt; </span><span style='color: #949494;'># A tibble: 11 x 11</span><span>
#&gt;    variant_id gene_name chromosome_name chromosome_posi… distance
#&gt;    </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>      </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>     </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>                      </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span>    </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span>
#&gt; </span><span style='color: #BCBCBC;'> 1</span><span> rs56261590 LOC10272… 4                       73</span><span style='text-decoration: underline;'>189</span><span>781    </span><span style='text-decoration: underline;'>69</span><span>394
#&gt; </span><span style='color: #BCBCBC;'> 2</span><span> rs56261590 ANKRD17   4                       73</span><span style='text-decoration: underline;'>189</span><span>781        0
#&gt; </span><span style='color: #BCBCBC;'> 3</span><span> rs56261590 AC053527… 4                       73</span><span style='text-decoration: underline;'>189</span><span>781    </span><span style='text-decoration: underline;'>69</span><span>428
#&gt; </span><span style='color: #BCBCBC;'> 4</span><span> rs56261590 HMGA1P2   4                       73</span><span style='text-decoration: underline;'>189</span><span>781    </span><span style='text-decoration: underline;'>90</span><span>636
#&gt; </span><span style='color: #BCBCBC;'> 5</span><span> rs56261590 ANKRD17   4                       73</span><span style='text-decoration: underline;'>189</span><span>781        0
#&gt; </span><span style='color: #BCBCBC;'> 6</span><span> rs56261590 RF00334   4                       73</span><span style='text-decoration: underline;'>189</span><span>781    </span><span style='text-decoration: underline;'>74</span><span>179
#&gt; </span><span style='color: #BCBCBC;'> 7</span><span> rs56261590 HMGA1P2   4                       73</span><span style='text-decoration: underline;'>189</span><span>781    </span><span style='text-decoration: underline;'>89</span><span>275
#&gt; </span><span style='color: #BCBCBC;'> 8</span><span> rs4725504  DPP6      7                      153</span><span style='text-decoration: underline;'>608</span><span>338   </span><span style='text-decoration: underline;'>278</span><span>759
#&gt; </span><span style='color: #BCBCBC;'> 9</span><span> rs4725504  LOC10798… 7                      153</span><span style='text-decoration: underline;'>608</span><span>338    </span><span style='text-decoration: underline;'>84</span><span>107
#&gt; </span><span style='color: #BCBCBC;'>10</span><span> rs4725504  DPP6      7                      153</span><span style='text-decoration: underline;'>608</span><span>338   </span><span style='text-decoration: underline;'>139</span><span>648
#&gt; </span><span style='color: #BCBCBC;'>11</span><span> rs4725504  PAXBP1P1  7                      153</span><span style='text-decoration: underline;'>608</span><span>338   </span><span style='text-decoration: underline;'>158</span><span>463
#&gt; </span><span style='color: #949494;'># … with 6 more variables: is_closest_gene </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span style='color: #949494;'>, is_intergenic </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span style='color: #949494;'>,
#&gt; #   is_upstream </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span style='color: #949494;'>, is_downstream </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span style='color: #949494;'>, source </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span style='color: #949494;'>, mapping_method </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>
#&gt; 
#&gt; Slot "ensembl_ids":
#&gt; </span><span style='color: #949494;'># A tibble: 6 x 3</span><span>
#&gt;   variant_id gene_name  ensembl_id     
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>      </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>      </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>          
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> rs56261590 ANKRD17    ENSG00000132466
#&gt; </span><span style='color: #BCBCBC;'>2</span><span> rs56261590 AC053527.1 ENSG00000250220
#&gt; </span><span style='color: #BCBCBC;'>3</span><span> rs56261590 HMGA1P2    ENSG00000248641
#&gt; </span><span style='color: #BCBCBC;'>4</span><span> rs56261590 RF00334    ENSG00000251778
#&gt; </span><span style='color: #BCBCBC;'>5</span><span> rs4725504  DPP6       ENSG00000130226
#&gt; </span><span style='color: #BCBCBC;'>6</span><span> rs4725504  PAXBP1P1   ENSG00000233489
#&gt; 
#&gt; Slot "entrez_ids":
#&gt; </span><span style='color: #949494;'># A tibble: 6 x 3</span><span>
#&gt;   variant_id gene_name    entrez_id
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>      </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>        </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>    
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> rs56261590 LOC102724832 102724832
#&gt; </span><span style='color: #BCBCBC;'>2</span><span> rs56261590 ANKRD17      26057    
#&gt; </span><span style='color: #BCBCBC;'>3</span><span> rs56261590 HMGA1P2      171559   
#&gt; </span><span style='color: #BCBCBC;'>4</span><span> rs4725504  DPP6         1804     
#&gt; </span><span style='color: #BCBCBC;'>5</span><span> rs4725504  LOC107986750 107986750
#&gt; </span><span style='color: #BCBCBC;'>6</span><span> rs4725504  PAXBP1P1     106481704
#&gt; </div><div class='input'>
<span class='co'># Intersect common traits</span>
<span class='fu'>intersect</span>(<span class='no'>traits_ex01</span>, <span class='no'>traits_ex02</span>)</div><div class='output co'>#&gt; An object of class "traits"
#&gt; Slot "traits":
#&gt; </span><span style='color: #949494;'># A tibble: 1 x 3</span><span>
#&gt;   efo_id      trait       uri                                 
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>       </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>       </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>                               
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> EFO_0004884 breast size http://www.ebi.ac.uk/efo/EFO_0004884
#&gt; </div><div class='input'>
<span class='co'>#</span>
<span class='co'># setdiff()</span>
<span class='co'>#</span>
<span class='co'># Remove studies from ex01 that are also present in ex02</span>
<span class='fu'>setdiff</span>(<span class='no'>studies_ex01</span>, <span class='no'>studies_ex02</span>)</div><div class='output co'>#&gt; An object of class "studies"
#&gt; Slot "studies":
#&gt; </span><span style='color: #949494;'># A tibble: 1 x 13</span><span>
#&gt;   study_id reported_trait initial_sample_… replication_sam… gxe   gxg  
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>    </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>          </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>            </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>            </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span>
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> GCST003… Breast size    33,790 European… </span><span style='color: #BB0000;'>NA</span><span>               FALSE FALSE
#&gt; </span><span style='color: #949494;'># … with 7 more variables: snp_count </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span style='color: #949494;'>, qualifier </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span style='color: #949494;'>, imputed </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span style='color: #949494;'>,
#&gt; #   pooled </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span style='color: #949494;'>, study_design_comment </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span style='color: #949494;'>, full_pvalue_set </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span style='color: #949494;'>,
#&gt; #   user_requested </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span>
#&gt; 
#&gt; Slot "genotyping_techs":
#&gt; </span><span style='color: #949494;'># A tibble: 1 x 2</span><span>
#&gt;   study_id   genotyping_technology       
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>      </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>                       
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> GCST003985 Genome-wide genotyping array
#&gt; 
#&gt; Slot "platforms":
#&gt; </span><span style='color: #949494;'># A tibble: 1 x 2</span><span>
#&gt;   study_id   manufacturer
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>      </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>       
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> GCST003985 Illumina    
#&gt; 
#&gt; Slot "ancestries":
#&gt; </span><span style='color: #949494;'># A tibble: 1 x 4</span><span>
#&gt;   study_id   ancestry_id type    number_of_individuals
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>            </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>                   </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span>
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> GCST003985           1 initial                 </span><span style='text-decoration: underline;'>33</span><span>790
#&gt; 
#&gt; Slot "ancestral_groups":
#&gt; </span><span style='color: #949494;'># A tibble: 1 x 3</span><span>
#&gt;   study_id   ancestry_id ancestral_group
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>            </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>          
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> GCST003985           1 European       
#&gt; 
#&gt; Slot "countries_of_origin":
#&gt; </span><span style='color: #949494;'># A tibble: 1 x 5</span><span>
#&gt;   study_id   ancestry_id country_name major_area region
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>            </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>        </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>      </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span> 
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> GCST003985           1 </span><span style='color: #BB0000;'>NA</span><span>           </span><span style='color: #BB0000;'>NA</span><span>         </span><span style='color: #BB0000;'>NA</span><span>    
#&gt; 
#&gt; Slot "countries_of_recruitment":
#&gt; </span><span style='color: #949494;'># A tibble: 1 x 5</span><span>
#&gt;   study_id   ancestry_id country_name major_area region
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>            </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>        </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>      </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span> 
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> GCST003985           1 </span><span style='color: #BB0000;'>NA</span><span>           </span><span style='color: #BB0000;'>NA</span><span>         </span><span style='color: #BB0000;'>NA</span><span>    
#&gt; 
#&gt; Slot "publications":
#&gt; </span><span style='color: #949494;'># A tibble: 1 x 7</span><span>
#&gt;   study_id pubmed_id publication_date publication title author_fullname
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>        </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;date&gt;</span><span>           </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>       </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>          
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> GCST003…  27</span><span style='text-decoration: underline;'>182</span><span>965 2016-05-15       Nat Genet   Dete… Pickrell JK    
#&gt; </span><span style='color: #949494;'># … with 1 more variable: author_orcid </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>
#&gt; </div><div class='input'>
<span class='co'># Remove associations from ex01 that are also present in ex02</span>
<span class='fu'>setdiff</span>(<span class='no'>associations_ex01</span>, <span class='no'>associations_ex02</span>)</div><div class='output co'>#&gt; An object of class "associations"
#&gt; Slot "associations":
#&gt; </span><span style='color: #949494;'># A tibble: 3 x 17</span><span>
#&gt;   association_id   pvalue pvalue_descript… pvalue_mantissa pvalue_exponent
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>             </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>                      </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span>           </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span>
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> 22509          3.00</span><span style='color: #949494;'>e</span><span style='color: #BB0000;'>- 9</span><span> </span><span style='color: #BB0000;'>NA</span><span>                             3              -</span><span style='color: #BB0000;'>9</span><span>
#&gt; </span><span style='color: #BCBCBC;'>2</span><span> 22505          7.00</span><span style='color: #949494;'>e</span><span style='color: #BB0000;'>- 6</span><span> </span><span style='color: #BB0000;'>NA</span><span>                             7              -</span><span style='color: #BB0000;'>6</span><span>
#&gt; </span><span style='color: #BCBCBC;'>3</span><span> 19537565       7.00</span><span style='color: #949494;'>e</span><span style='color: #BB0000;'>-31</span><span> </span><span style='color: #BB0000;'>NA</span><span>                             7             -</span><span style='color: #BB0000;'>31</span><span>
#&gt; </span><span style='color: #949494;'># … with 12 more variables: multiple_snp_haplotype </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span style='color: #949494;'>,
#&gt; #   snp_interaction </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span style='color: #949494;'>, snp_type </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span style='color: #949494;'>, standard_error </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span style='color: #949494;'>, range </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span style='color: #949494;'>,
#&gt; #   or_per_copy_number </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span style='color: #949494;'>, beta_number </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span style='color: #949494;'>, beta_unit </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span style='color: #949494;'>,
#&gt; #   beta_direction </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span style='color: #949494;'>, beta_description </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span style='color: #949494;'>, last_mapping_date </span><span style='color: #949494;font-style: italic;'>&lt;dttm&gt;</span><span style='color: #949494;'>,
#&gt; #   last_update_date </span><span style='color: #949494;font-style: italic;'>&lt;dttm&gt;</span><span>
#&gt; 
#&gt; Slot "loci":
#&gt; </span><span style='color: #949494;'># A tibble: 3 x 4</span><span>
#&gt;   association_id locus_id haplotype_snp_count description   
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>             </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span>               </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>         
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> 22509                 1                  </span><span style='color: #BB0000;'>NA</span><span> Single variant
#&gt; </span><span style='color: #BCBCBC;'>2</span><span> 22505                 1                  </span><span style='color: #BB0000;'>NA</span><span> Single variant
#&gt; </span><span style='color: #BCBCBC;'>3</span><span> 19537565              1                  </span><span style='color: #BB0000;'>NA</span><span> Single variant
#&gt; 
#&gt; Slot "risk_alleles":
#&gt; </span><span style='color: #949494;'># A tibble: 3 x 7</span><span>
#&gt;   association_id locus_id variant_id risk_allele risk_frequency genome_wide
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>             </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>      </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>                </span><span style='color: #949494;font-style: italic;'>&lt;dbl&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span>      
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> 22509                 1 rs7089814  C                       </span><span style='color: #BB0000;'>NA</span><span> </span><span style='color: #BB0000;'>NA</span><span>         
#&gt; </span><span style='color: #BCBCBC;'>2</span><span> 22505                 1 chr22:458… T                       </span><span style='color: #BB0000;'>NA</span><span> </span><span style='color: #BB0000;'>NA</span><span>         
#&gt; </span><span style='color: #BCBCBC;'>3</span><span> 19537565              1 rs10110651 </span><span style='color: #BB0000;'>NA</span><span>                      </span><span style='color: #BB0000;'>NA</span><span> FALSE      
#&gt; </span><span style='color: #949494;'># … with 1 more variable: limited_list </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span>
#&gt; 
#&gt; Slot "genes":
#&gt; </span><span style='color: #949494;'># A tibble: 7 x 3</span><span>
#&gt;   association_id locus_id gene_name   
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>             </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>       
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> 22509                 1 ADO         
#&gt; </span><span style='color: #BCBCBC;'>2</span><span> 22509                 1 ZNF365      
#&gt; </span><span style='color: #BCBCBC;'>3</span><span> 22509                 1 ALDH7A1P4   
#&gt; </span><span style='color: #BCBCBC;'>4</span><span> 22505                 1 TBC1D22A    
#&gt; </span><span style='color: #BCBCBC;'>5</span><span> 22505                 1 LOC100128818
#&gt; </span><span style='color: #BCBCBC;'>6</span><span> 19537565              1 KCNU1       
#&gt; </span><span style='color: #BCBCBC;'>7</span><span> 19537565              1 ZNF703      
#&gt; 
#&gt; Slot "ensembl_ids":
#&gt; </span><span style='color: #949494;'># A tibble: 7 x 4</span><span>
#&gt;   association_id locus_id gene_name    ensembl_id     
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>             </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>        </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>          
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> 22509                 1 ADO          ENSG00000181915
#&gt; </span><span style='color: #BCBCBC;'>2</span><span> 22509                 1 ZNF365       ENSG00000138311
#&gt; </span><span style='color: #BCBCBC;'>3</span><span> 22509                 1 ALDH7A1P4    ENSG00000234489
#&gt; </span><span style='color: #BCBCBC;'>4</span><span> 22505                 1 TBC1D22A     ENSG00000054611
#&gt; </span><span style='color: #BCBCBC;'>5</span><span> 22505                 1 LOC100128818 </span><span style='color: #BB0000;'>NA</span><span>             
#&gt; </span><span style='color: #BCBCBC;'>6</span><span> 19537565              1 KCNU1        ENSG00000215262
#&gt; </span><span style='color: #BCBCBC;'>7</span><span> 19537565              1 ZNF703       ENSG00000183779
#&gt; 
#&gt; Slot "entrez_ids":
#&gt; </span><span style='color: #949494;'># A tibble: 7 x 4</span><span>
#&gt;   association_id locus_id gene_name    entrez_id
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>             </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>        </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>    
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> 22509                 1 ADO          84890    
#&gt; </span><span style='color: #BCBCBC;'>2</span><span> 22509                 1 ZNF365       22891    
#&gt; </span><span style='color: #BCBCBC;'>3</span><span> 22509                 1 ALDH7A1P4    544      
#&gt; </span><span style='color: #BCBCBC;'>4</span><span> 22505                 1 TBC1D22A     25771    
#&gt; </span><span style='color: #BCBCBC;'>5</span><span> 22505                 1 LOC100128818 </span><span style='color: #BB0000;'>NA</span><span>       
#&gt; </span><span style='color: #BCBCBC;'>6</span><span> 19537565              1 KCNU1        157855   
#&gt; </span><span style='color: #BCBCBC;'>7</span><span> 19537565              1 ZNF703       80139    
#&gt; </div><div class='input'>
<span class='co'># Remove variants from ex01 that are also present in ex02</span>
<span class='fu'>setdiff</span>(<span class='no'>variants_ex01</span>, <span class='no'>variants_ex02</span>)</div><div class='output co'>#&gt; An object of class "variants"
#&gt; Slot "variants":
#&gt; </span><span style='color: #949494;'># A tibble: 1 x 7</span><span>
#&gt;   variant_id merged functional_class chromosome_name chromosome_posi…
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>       </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span> </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>            </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>                      </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span>
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> rs1469924…      0 regulatory_regi… 8                       36</span><span style='text-decoration: underline;'>984</span><span>538
#&gt; </span><span style='color: #949494;'># … with 2 more variables: chromosome_region </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span style='color: #949494;'>, last_update_date </span><span style='color: #949494;font-style: italic;'>&lt;dttm&gt;</span><span>
#&gt; 
#&gt; Slot "genomic_contexts":
#&gt; </span><span style='color: #949494;'># A tibble: 10 x 11</span><span>
#&gt;    variant_id gene_name chromosome_name chromosome_posi… distance
#&gt;    </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>      </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>     </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>                      </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span>    </span><span style='color: #949494;font-style: italic;'>&lt;int&gt;</span><span>
#&gt; </span><span style='color: #BCBCBC;'> 1</span><span> rs1469924… RPL26P25  8                       36</span><span style='text-decoration: underline;'>984</span><span>538     </span><span style='text-decoration: underline;'>3</span><span>392
#&gt; </span><span style='color: #BCBCBC;'> 2</span><span> rs1469924… TPT1P8    8                       36</span><span style='text-decoration: underline;'>984</span><span>538    </span><span style='text-decoration: underline;'>95</span><span>008
#&gt; </span><span style='color: #BCBCBC;'> 3</span><span> rs1469924… LOC10537… 8                       36</span><span style='text-decoration: underline;'>984</span><span>538    </span><span style='text-decoration: underline;'>59</span><span>331
#&gt; </span><span style='color: #BCBCBC;'> 4</span><span> rs1469924… LOC10537… 8                       36</span><span style='text-decoration: underline;'>984</span><span>538    </span><span style='text-decoration: underline;'>81</span><span>555
#&gt; </span><span style='color: #BCBCBC;'> 5</span><span> rs1469924… LOC10537… 8                       36</span><span style='text-decoration: underline;'>984</span><span>538     </span><span style='text-decoration: underline;'>2</span><span>533
#&gt; </span><span style='color: #BCBCBC;'> 6</span><span> rs1469924… TPT1P8    8                       36</span><span style='text-decoration: underline;'>984</span><span>538    </span><span style='text-decoration: underline;'>95</span><span>231
#&gt; </span><span style='color: #BCBCBC;'> 7</span><span> rs1469924… AC092818… 8                       36</span><span style='text-decoration: underline;'>984</span><span>538    </span><span style='text-decoration: underline;'>82</span><span>903
#&gt; </span><span style='color: #BCBCBC;'> 8</span><span> rs1469924… KCNU1     8                       36</span><span style='text-decoration: underline;'>984</span><span>538    </span><span style='text-decoration: underline;'>48</span><span>413
#&gt; </span><span style='color: #BCBCBC;'> 9</span><span> rs1469924… AC090453… 8                       36</span><span style='text-decoration: underline;'>984</span><span>538     </span><span style='text-decoration: underline;'>3</span><span>439
#&gt; </span><span style='color: #BCBCBC;'>10</span><span> rs1469924… KCNU1     8                       36</span><span style='text-decoration: underline;'>984</span><span>538    </span><span style='text-decoration: underline;'>48</span><span>410
#&gt; </span><span style='color: #949494;'># … with 6 more variables: is_closest_gene </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span style='color: #949494;'>, is_intergenic </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span style='color: #949494;'>,
#&gt; #   is_upstream </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span style='color: #949494;'>, is_downstream </span><span style='color: #949494;font-style: italic;'>&lt;lgl&gt;</span><span style='color: #949494;'>, source </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span style='color: #949494;'>, mapping_method </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>
#&gt; 
#&gt; Slot "ensembl_ids":
#&gt; </span><span style='color: #949494;'># A tibble: 4 x 3</span><span>
#&gt;   variant_id  gene_name  ensembl_id     
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>       </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>      </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>          
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> rs146992477 TPT1P8     ENSG00000219491
#&gt; </span><span style='color: #BCBCBC;'>2</span><span> rs146992477 AC092818.1 ENSG00000254038
#&gt; </span><span style='color: #BCBCBC;'>3</span><span> rs146992477 KCNU1      ENSG00000215262
#&gt; </span><span style='color: #BCBCBC;'>4</span><span> rs146992477 AC090453.1 ENSG00000243503
#&gt; 
#&gt; Slot "entrez_ids":
#&gt; </span><span style='color: #949494;'># A tibble: 6 x 3</span><span>
#&gt;   variant_id  gene_name    entrez_id
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>       </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>        </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>    
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> rs146992477 RPL26P25     100270966
#&gt; </span><span style='color: #BCBCBC;'>2</span><span> rs146992477 TPT1P8       59347    
#&gt; </span><span style='color: #BCBCBC;'>3</span><span> rs146992477 LOC105379375 105379375
#&gt; </span><span style='color: #BCBCBC;'>4</span><span> rs146992477 LOC105379377 105379377
#&gt; </span><span style='color: #BCBCBC;'>5</span><span> rs146992477 LOC105379376 105379376
#&gt; </span><span style='color: #BCBCBC;'>6</span><span> rs146992477 KCNU1        157855   
#&gt; </div><div class='input'>
<span class='co'># Remove traits from ex01 that are also present in ex02</span>
<span class='fu'>setdiff</span>(<span class='no'>traits_ex01</span>, <span class='no'>traits_ex02</span>)</div><div class='output co'>#&gt; An object of class "traits"
#&gt; Slot "traits":
#&gt; </span><span style='color: #949494;'># A tibble: 2 x 3</span><span>
#&gt;   efo_id      trait            uri                                 
#&gt;   </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>       </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>            </span><span style='color: #949494;font-style: italic;'>&lt;chr&gt;</span><span>                               
#&gt; </span><span style='color: #BCBCBC;'>1</span><span> EFO_0004343 waist-hip ratio  http://www.ebi.ac.uk/efo/EFO_0004343
#&gt; </span><span style='color: #BCBCBC;'>2</span><span> EFO_0005299 non-word reading http://www.ebi.ac.uk/efo/EFO_0005299
#&gt; </div><div class='input'>
<span class='co'>#</span>
<span class='co'># setequal()</span>
<span class='co'>#</span>
<span class='co'># Compare two studies objects</span>
<span class='fu'>setequal</span>(<span class='no'>studies_ex01</span>, <span class='no'>studies_ex01</span>)</div><div class='output co'>#&gt; [1] TRUE</div><div class='input'><span class='fu'>setequal</span>(<span class='no'>studies_ex01</span>, <span class='no'>studies_ex02</span>)</div><div class='output co'>#&gt; [1] FALSE</div><div class='input'>
<span class='co'># Compare two associations objects</span>
<span class='fu'>setequal</span>(<span class='no'>associations_ex01</span>, <span class='no'>associations_ex01</span>)</div><div class='output co'>#&gt; [1] TRUE</div><div class='input'><span class='fu'>setequal</span>(<span class='no'>associations_ex01</span>, <span class='no'>associations_ex02</span>)</div><div class='output co'>#&gt; [1] FALSE</div><div class='input'>
<span class='co'># Compare two variants objects</span>
<span class='fu'>setequal</span>(<span class='no'>variants_ex01</span>, <span class='no'>variants_ex01</span>)</div><div class='output co'>#&gt; [1] TRUE</div><div class='input'><span class='fu'>setequal</span>(<span class='no'>variants_ex01</span>, <span class='no'>variants_ex02</span>)</div><div class='output co'>#&gt; [1] FALSE</div><div class='input'>
<span class='co'># Compare two traits objects</span>
<span class='fu'>setequal</span>(<span class='no'>traits_ex01</span>, <span class='no'>traits_ex01</span>)</div><div class='output co'>#&gt; [1] TRUE</div><div class='input'><span class='fu'>setequal</span>(<span class='no'>traits_ex01</span>, <span class='no'>traits_ex02</span>)</div><div class='output co'>#&gt; [1] FALSE</div><div class='input'>
</div></span></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Ramiro Magno, Ana-Teresa Maia, Centre for Biomedical Research, University of Algarve.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
   </div>

  

  </body>
</html>

