<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en-US">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Variants associated with Body Mass Index (BMI) • gwasrapidd</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/Nunito-0.4.10/font.css" rel="stylesheet">
<link href="../deps/Inconsolata-0.4.10/font.css" rel="stylesheet">
<link href="../deps/Source_Sans_Pro-0.4.10/font.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Variants associated with Body Mass Index (BMI)">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-none" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">gwasrapidd</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.18</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/gwasrapidd.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/bmi_variants.html">Variants associated with Body Mass Index (BMI)</a></li>
    <li><a class="dropdown-item" href="../articles/faq.html">Frequently asked questions</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/ramiromagno/gwasrapidd/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>Variants associated with Body Mass Index (BMI)</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/ramiromagno/gwasrapidd/blob/master/vignettes/bmi_variants.Rmd" class="external-link"><code>vignettes/bmi_variants.Rmd</code></a></small>
      <div class="d-none name"><code>bmi_variants.Rmd</code></div>
    </div>

    
    
<p>Start by loading <a href="https://github.com/ramiromagno/gwasrapidd" class="external-link">gwasrapidd</a>, and <a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a>
and <a href="https://tidyr.tidyverse.org" class="external-link">tidyr</a>:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyr.tidyverse.org" class="external-link">tidyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/ramiromagno/gwasrapidd" class="external-link">gwasrapidd</a></span><span class="op">)</span></span></code></pre></div>
<p>Let’s say you want to retrieve all variants associated with the
phenotype <em>Body Mass Index (BMI)</em>. Moreover, you want to sort
them by their risk allele (minor allele), as well as the effect size
(beta coefficient) and p-value.</p>
<p>First we start by finding the Experimental Factor Ontology (EFO)
identifier(s) corresponding to BMI. To do this, we start by downloading
all traits in the GWAS Catalog.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">all_traits</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_traits.html">get_traits</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>Then look for ‘BMI’ in the trait description column.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">all_traits</span><span class="op">@</span><span class="va">traits</span>, <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">'BMI'</span>, <span class="va">trait</span>, ignore.case <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; # A tibble: 10 × 3</span></span>
<span><span class="co">#&gt;    efo_id      trait                                          uri                                 </span></span>
<span><span class="co">#&gt;    &lt;chr&gt;       &lt;chr&gt;                                          &lt;chr&gt;                               </span></span>
<span><span class="co">#&gt;  1 EFO_0005937 longitudinal BMI measurement                   http://www.ebi.ac.uk/efo/EFO_0005937</span></span>
<span><span class="co">#&gt;  2 EFO_0007737 BMI-adjusted adiponectin measurement           http://www.ebi.ac.uk/efo/EFO_0007737</span></span>
<span><span class="co">#&gt;  3 EFO_0007788 BMI-adjusted waist-hip ratio                   http://www.ebi.ac.uk/efo/EFO_0007788</span></span>
<span><span class="co">#&gt;  4 EFO_0007789 BMI-adjusted waist circumference               http://www.ebi.ac.uk/efo/EFO_0007789</span></span>
<span><span class="co">#&gt;  5 EFO_0007793 BMI-adjusted leptin measurement                http://www.ebi.ac.uk/efo/EFO_0007793</span></span>
<span><span class="co">#&gt;  6 EFO_0008036 BMI-adjusted fasting blood glucose measurement http://www.ebi.ac.uk/efo/EFO_0008036</span></span>
<span><span class="co">#&gt;  7 EFO_0008037 BMI-adjusted fasting blood insulin measurement http://www.ebi.ac.uk/efo/EFO_0008037</span></span>
<span><span class="co">#&gt;  8 EFO_0008038 BMI-adjusted hip bone size                     http://www.ebi.ac.uk/efo/EFO_0008038</span></span>
<span><span class="co">#&gt;  9 EFO_0008039 BMI-adjusted hip circumference                 http://www.ebi.ac.uk/efo/EFO_0008039</span></span>
<span><span class="co">#&gt; 10 EFO_0011044 BMI-adjusted neck circumference                http://www.ebi.ac.uk/efo/EFO_0011044</span></span></code></pre></div>
<p>So there are several phenotypes whose description includes the
keyword ‘BMI’. However, only the EFO trait <code>'EFO_0005937'</code>
(‘longitudinal BMI measurement’) really corresponds to BMI as a
phenotypic trait. All other traits are adjusted for BMI but are not BMI
traits per se (you can further confirm this by looking at each trait
description, just by opening your web browser with each respective
URI).</p>
<p>To get statistical association data for the trait ‘longitudinal BMI
measurement’ (<code>'EFO_0005937'</code>), as well as associated
variants and effect sizes, we use the gwasrapidd
<code><a href="../reference/get_associations.html">get_associations()</a></code> function:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bmi_associations</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_associations.html">get_associations</a></span><span class="op">(</span>efo_id <span class="op">=</span> <span class="st">'EFO_0005937'</span><span class="op">)</span></span></code></pre></div>
<p>The S4 object <code>bmi_associations</code> contains several tables,
namely <code>'associations'</code>, <code>'loci'</code>,
<code>'risk_alleles'</code>, <code>'genes'</code>,
<code>'ensembl_ids'</code> and <code>'entrez_ids'</code>:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">slotNames</span><span class="op">(</span><span class="va">bmi_associations</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "associations" "loci"         "risk_alleles" "genes"        "ensembl_ids"  "entrez_ids"</span></span></code></pre></div>
<p>From table <code>'associations'</code> we can extract the
variables:</p>
<ul>
<li><code>'association_id'</code></li>
<li><code>'pvalue'</code></li>
<li><code>'beta_number'</code></li>
<li><code>'beta_unit'</code></li>
<li><code>'beta_direction'</code></li>
</ul>
<p>whereas from table <code>'risk_alleles'</code> we can obtain:</p>
<ul>
<li><code>'association_id'</code></li>
<li><code>'variant_id'</code></li>
<li>
<code>'risk_allele'</code>.</li>
</ul>
<p>We extract all these variables and combine them into one single
dataframe (<code>bmi_variants</code>), using
<code>'association_id'</code> as the matching key:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tbl01</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">bmi_associations</span><span class="op">@</span><span class="va">risk_alleles</span>, <span class="va">association_id</span>, <span class="va">variant_id</span>, <span class="va">risk_allele</span><span class="op">)</span></span>
<span><span class="va">tbl02</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">bmi_associations</span><span class="op">@</span><span class="va">associations</span>, <span class="va">association_id</span>, <span class="va">pvalue</span>, <span class="va">beta_number</span>, <span class="va">beta_unit</span>, <span class="va">beta_direction</span><span class="op">)</span></span>
<span></span>
<span><span class="va">bmi_variants</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">tbl01</span>, <span class="va">tbl02</span>, by <span class="op">=</span> <span class="st">'association_id'</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/drop_na.html" class="external-link">drop_na</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">variant_id</span>, <span class="va">risk_allele</span><span class="op">)</span></span></code></pre></div>
<p>The final results show 42 associations. Note that some variant/allele
combinations might be repeated as the same variant/allele combination
might have been assessed in more than one GWAS study.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bmi_variants</span></span>
<span><span class="co">#&gt; # A tibble: 42 × 7</span></span>
<span><span class="co">#&gt;    association_id variant_id  risk_allele     pvalue beta_number beta_unit beta_direction</span></span>
<span><span class="co">#&gt;    &lt;chr&gt;          &lt;chr&gt;       &lt;chr&gt;            &lt;dbl&gt;       &lt;dbl&gt; &lt;chr&gt;     &lt;chr&gt;         </span></span>
<span><span class="co">#&gt;  1 10066323       rs10041997  A           0.0000004        0.138 unit      increase      </span></span>
<span><span class="co">#&gt;  2 61729722       rs10070777  A           0.000004         0.879 unit      increase      </span></span>
<span><span class="co">#&gt;  3 61729714       rs10278819  C           0.000002         0.873 unit      increase      </span></span>
<span><span class="co">#&gt;  4 61729787       rs10426669  G           0.000006         0.765 unit      increase      </span></span>
<span><span class="co">#&gt;  5 61729731       rs1048163   G           0.000005         0.450 unit      increase      </span></span>
<span><span class="co">#&gt;  6 61729771       rs1048164   A           0.000009         0.436 unit      increase      </span></span>
<span><span class="co">#&gt;  7 55309232       rs10515235  A           0.000002         0.05  unit      increase      </span></span>
<span><span class="co">#&gt;  8 55309249       rs10938397  G           0.00000003       0.06  unit      increase      </span></span>
<span><span class="co">#&gt;  9 61729763       rs112045010 C           0.000005         0.906 unit      increase      </span></span>
<span><span class="co">#&gt; 10 61729759       rs11979775  C           0.000005         0.501 unit      increase      </span></span>
<span><span class="co">#&gt; # … with 32 more rows</span></span></code></pre></div>
  </main>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://www.pattern.institute/team/rmagno/" class="external-link">Ramiro Magno</a>, Ana-Teresa Maia, CINTESIS, <a href="https://www.pattern.institute" class="external-link">Pattern Institute</a>.</p>
</div>

<div class="pkgdown-footer-right">

</div>

    </footer>
</div>





  </body>
</html>
