<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en-US">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Getting started with gwasrapidd • gwasrapidd</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/Nunito-0.4.10/font.css" rel="stylesheet">
<link href="../deps/Inconsolata-0.4.10/font.css" rel="stylesheet">
<link href="../deps/Source_Sans_Pro-0.4.10/font.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Getting started with gwasrapidd">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-none" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">gwasrapidd</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.18</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/gwasrapidd.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/bmi_variants.html">Variants associated with Body Mass Index (BMI)</a></li>
    <li><a class="dropdown-item" href="../articles/faq.html">Frequently asked questions</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/ramiromagno/gwasrapidd/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">


<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>Getting started with gwasrapidd</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/ramiromagno/gwasrapidd/blob/master/vignettes/gwasrapidd.Rmd" class="external-link"><code>vignettes/gwasrapidd.Rmd</code></a></small>
      <div class="d-none name"><code>gwasrapidd.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="the-gwas-catalog">The GWAS Catalog<a class="anchor" aria-label="anchor" href="#the-gwas-catalog"></a>
</h2>
<p>The GWAS Catalog is a service provided by the EMBL-EBI and NHGRI that offers a
manually curated and freely available database of published genome-wide
association studies (GWAS). The Catalog website and infrastructure is hosted by
the <a href="https://www.ebi.ac.uk" class="external-link">EMBL-EBI</a>.</p>
<p><a href="https://github.com/ramiromagno/gwasrapidd" class="external-link">gwasrapidd</a> facilitates the access to the Catalog via the REST API,
allowing you to programmatically retrieve data directly into R.</p>
</div>
<div class="section level2">
<h2 id="gwas-catalog-entities">GWAS Catalog Entities<a class="anchor" aria-label="anchor" href="#gwas-catalog-entities"></a>
</h2>
<p>The Catalog REST API is organized around four core entities:</p>
<ul>
<li>studies</li>
<li>associations</li>
<li>variants</li>
<li>traits</li>
</ul>
<p><a href="https://github.com/ramiromagno/gwasrapidd" class="external-link">gwasrapidd</a> provides four corresponding functions to get each of the
entities: <code><a href="../reference/get_studies.html">get_studies()</a></code>, <code><a href="../reference/get_associations.html">get_associations()</a></code>, <code><a href="../reference/get_variants.html">get_variants()</a></code>, and
<code><a href="../reference/get_traits.html">get_traits()</a></code>.</p>
<p>Each function maps to an appropriately named S4 classed object: <a href="https://rmagno.eu/gwasrapidd/reference/studies-class.html">studies</a>, <a href="https://rmagno.eu/gwasrapidd/reference/associations-class.html">associations</a>, <a href="https://rmagno.eu/gwasrapidd/reference/variants-class.html">variants</a>, and <a href="https://rmagno.eu/gwasrapidd/reference/traits-class.html">traits</a> (see Figure <a href="#fig:fns">1</a>).</p>
<div class="figure">
<span style="display:block;" id="fig:fns"></span>
<img src="../reference/figures/get_fns.png" alt="`{gwasrapidd}` retrieval functions." width="70%"><p class="caption">
Figure 1: <a href="https://github.com/ramiromagno/gwasrapidd" class="external-link">gwasrapidd</a> retrieval functions.
</p>
</div>
<p>You can use a combination of several search criteria with each retrieval
function as shown in Figure <a href="#fig:criteria">2</a>. For example, if you want to get studies using
either one of these two criteria:</p>
<ul>
<li>study accession identifier (<code>study_id</code>)</li>
<li>variant identifier (<code>variant_id</code>)</li>
</ul>
<p>You could run the following code:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/ramiromagno/gwasrapidd" class="external-link">gwasrapidd</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">my_studies</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_studies.html">get_studies</a></span><span class="op">(</span>study_id <span class="op">=</span> <span class="st">'GCST000858'</span>, variant_id <span class="op">=</span> <span class="st">'rs12752552'</span><span class="op">)</span></span>
<span><span class="va">my_studies</span><span class="op">@</span><span class="va">studies</span></span></code></pre></div>
<p>This command returns all studies that match either <code>'GCST000858'</code> or
<code>'rs12752552'</code>. This is equivalent to running <code>get_studies</code> separately on each
criteria, and combining the results afterwards:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">s1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_studies.html">get_studies</a></span><span class="op">(</span>study_id <span class="op">=</span> <span class="st">'GCST000858'</span><span class="op">)</span></span>
<span><span class="va">s2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_studies.html">get_studies</a></span><span class="op">(</span>variant_id <span class="op">=</span> <span class="st">'rs12752552'</span><span class="op">)</span></span>
<span><span class="va">my_studies</span> <span class="op">&lt;-</span> <span class="fu">gwasrapidd</span><span class="fu">::</span><span class="fu"><a href="../reference/setop.html">union</a></span><span class="op">(</span><span class="va">s1</span>, <span class="va">s2</span><span class="op">)</span></span></code></pre></div>
<p>All four retrieval functions accept the <code>set_operation</code> parameter which defines the way the results obtained with each criterion are combined. The two options for this parameter are <code>'union'</code> (default) or <code>'intersection'</code>, resulting, respectively, in an OR or AND operation.</p>
<div class="figure">
<span style="display:block;" id="fig:criteria"></span>
<img src="../reference/figures/get_criteria.png" alt="`{gwasrapidd}` arguments for retrieval functions. Colors indicate the criteria that can be used for retrieving GWAS Catalog entities: studies (green), associations (red), variants (purple), and traits (orange)." width="70%"><p class="caption">
Figure 2: <a href="https://github.com/ramiromagno/gwasrapidd" class="external-link">gwasrapidd</a> arguments for retrieval functions. Colors indicate the criteria that can be used for retrieving GWAS Catalog entities: studies (green), associations (red), variants (purple), and traits (orange).
</p>
</div>
</div>
<div class="section level2">
<h2 id="finding-risk-alleles-associated-with-autoimmune-disease">Finding Risk Alleles Associated with Autoimmune Disease<a class="anchor" aria-label="anchor" href="#finding-risk-alleles-associated-with-autoimmune-disease"></a>
</h2>
<p>As a first example, take the work by <span class="citation">Light et al. (2014)</span>. In this work the authors focused on variants that had been previously reported in genome-wide association studies (GWAS) for autoimmune disease.</p>
<p>With <strong>gwasrapidd</strong> we can interrogate the GWAS Catalog for the study/studies by searching by <em>autoimmune disease</em> (an EFO trait). To do that let’s load gwasrapidd first:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/ramiromagno/gwasrapidd" class="external-link">gwasrapidd</a></span><span class="op">)</span></span></code></pre></div>
<p>Then query the GWAS Catalog by EFO trait:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">my_studies</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_studies.html">get_studies</a></span><span class="op">(</span>efo_trait <span class="op">=</span> <span class="st">'autoimmune disease'</span><span class="op">)</span></span></code></pre></div>
<p>We can now check how many GWAS studies we got back:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">gwasrapidd</span><span class="fu">::</span><span class="fu"><a href="../reference/n.html">n</a></span><span class="op">(</span><span class="va">my_studies</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 9</span></span>
<span><span class="va">my_studies</span><span class="op">@</span><span class="va">studies</span><span class="op">$</span><span class="va">study_id</span></span>
<span><span class="co">#&gt; [1] "GCST003097"   "GCST011008"   "GCST007071"   "GCST009873"   "GCST011005"   "GCST011009"   "GCST90029015"</span></span>
<span><span class="co">#&gt; [8] "GCST90029016" "GCST90012738"</span></span></code></pre></div>
<p>Apparently only 9 studies: GCST003097, GCST011008, GCST007071, GCST009873, GCST011005, GCST011009, GCST90029015, GCST90029016, GCST90012738. Let’s see the associated publication titles:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">my_studies</span><span class="op">@</span><span class="va">publications</span><span class="op">$</span><span class="va">title</span></span>
<span><span class="co">#&gt; [1] "Meta-analysis of shared genetic architecture across ten pediatric autoimmune diseases."                                                            </span></span>
<span><span class="co">#&gt; [2] "Genetic factors underlying the bidirectional relationship between autoimmune and mental disorders - findings from a Danish population-based study."</span></span>
<span><span class="co">#&gt; [3] "Leveraging Polygenic Functional Enrichment to Improve GWAS Power."                                                                                 </span></span>
<span><span class="co">#&gt; [4] "Meta-analysis of Immunochip data of four autoimmune diseases reveals novel single-disease and cross-phenotype associations."                       </span></span>
<span><span class="co">#&gt; [5] "Genetic factors underlying the bidirectional relationship between autoimmune and mental disorders - findings from a Danish population-based study."</span></span>
<span><span class="co">#&gt; [6] "Genetic factors underlying the bidirectional relationship between autoimmune and mental disorders - findings from a Danish population-based study."</span></span>
<span><span class="co">#&gt; [7] "Mixed-model association for biobank-scale datasets."                                                                                               </span></span>
<span><span class="co">#&gt; [8] "Mixed-model association for biobank-scale datasets."                                                                                               </span></span>
<span><span class="co">#&gt; [9] "Clinical laboratory test-wide association scan of polygenic scores identifies biomarkers of complex disease."</span></span></code></pre></div>
<p>If you want to further inspect these publications, you can quickly browse the respective PubMed entries:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># This launches your web browser at https://www.ncbi.nlm.nih.gov/pubmed/26301688</span></span>
<span><span class="fu"><a href="../reference/open_in_pubmed.html">open_in_pubmed</a></span><span class="op">(</span><span class="va">my_studies</span><span class="op">@</span><span class="va">publications</span><span class="op">$</span><span class="va">pubmed_id</span><span class="op">)</span></span></code></pre></div>
<p>Now if we want to know the variants previously associated with autoimmune disease, as used by <span class="citation">Light et al. (2014)</span>, we need to retrieve statistical association information on these variants, and then filter them based on the same level of significance <span class="math inline">\(P &lt; 1\times 10^{-6}\)</span> <span class="citation">(Light et al. 2014)</span>.</p>
<p>So let’s start by getting the associations by <code>study_id</code>:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># You could have also used get_associations(efo_trait = 'autoimmune disease')</span></span>
<span><span class="va">my_associations</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_associations.html">get_associations</a></span><span class="op">(</span>study_id <span class="op">=</span> <span class="va">my_studies</span><span class="op">@</span><span class="va">studies</span><span class="op">$</span><span class="va">study_id</span><span class="op">)</span></span></code></pre></div>
<p>Seemingly, there are 182 associations.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">gwasrapidd</span><span class="fu">::</span><span class="fu"><a href="../reference/n.html">n</a></span><span class="op">(</span><span class="va">my_associations</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 182</span></span></code></pre></div>
<p>However, not all variants meet the level of significance, as required by <span class="citation">Light et al. (2014)</span>:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get association ids for which pvalue is less than 1e-6.</span></span>
<span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">my_associations</span><span class="op">@</span><span class="va">associations</span>, <span class="va">pvalue</span> <span class="op">&lt;</span> <span class="fl">1e-6</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="co"># Filter by p-value</span></span>
<span>  <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/drop_na.html" class="external-link">drop_na</a></span><span class="op">(</span><span class="va">pvalue</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">association_id</span><span class="op">)</span> <span class="op">-&gt;</span> <span class="va">association_ids</span> <span class="co"># Extract column association_id</span></span></code></pre></div>
<p>Here we subset the <code>my_associations</code> object by a vector of association identifiers (<code>association_ids</code>) into a smaller object, <code>my_associations2</code>:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Extract associations by association id</span></span>
<span><span class="va">my_associations2</span> <span class="op">&lt;-</span> <span class="va">my_associations</span><span class="op">[</span><span class="va">association_ids</span><span class="op">]</span></span>
<span><span class="fu">gwasrapidd</span><span class="fu">::</span><span class="fu"><a href="../reference/n.html">n</a></span><span class="op">(</span><span class="va">my_associations2</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 180</span></span></code></pre></div>
<p>Of the 182 associations found in GWAS Catalog, 180 meet the p-value threshold of <span class="math inline">\(1\times 10^{-6}\)</span>. Here are the variants, and their respective risk allele and risk frequency:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">my_associations2</span><span class="op">@</span><span class="va">risk_alleles</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'variant_id'</span>, <span class="st">'risk_allele'</span>, <span class="st">'risk_frequency'</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="co">#&gt; # A tibble: 180 × 3</span></span>
<span><span class="co">#&gt;    variant_id risk_allele risk_frequency</span></span>
<span><span class="co">#&gt;    &lt;chr&gt;      &lt;chr&gt;                &lt;dbl&gt;</span></span>
<span><span class="co">#&gt;  1 rs11580078 G                     0.43</span></span>
<span><span class="co">#&gt;  2 rs6679677  A                     0.09</span></span>
<span><span class="co">#&gt;  3 rs34884278 C                     0.3 </span></span>
<span><span class="co">#&gt;  4 rs6689858  C                     0.29</span></span>
<span><span class="co">#&gt;  5 rs2075184  T                     0.23</span></span>
<span><span class="co">#&gt;  6 rs36001488 C                     0.48</span></span>
<span><span class="co">#&gt;  7 rs4676410  A                     0.19</span></span>
<span><span class="co">#&gt;  8 rs4625     G                     0.31</span></span>
<span><span class="co">#&gt;  9 rs62324212 A                     0.42</span></span>
<span><span class="co">#&gt; 10 rs7725052  C                     0.43</span></span>
<span><span class="co">#&gt; # … with 170 more rows</span></span></code></pre></div>
</div>
<div class="section level2 unnumbered">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-Light2014" class="csl-entry">
Light, Nicholas, Véronique Adoue, Bing Ge, Shu-Huang Chen, Tony Kwan, and Tomi Pastinen. 2014. <span>“Interrogation of Allelic Chromatin States in Human Cells by High-Density ChIP-Genotyping.”</span> <em>Epigenetics</em> 9 (9): 1238–51.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://www.pattern.institute/team/rmagno/" class="external-link">Ramiro Magno</a>, Ana-Teresa Maia, CINTESIS, <a href="https://www.pattern.institute" class="external-link">Pattern Institute</a>.</p>
</div>

<div class="pkgdown-footer-right">

</div>

    </footer>
</div>





  </body>
</html>
